c_files/pr30457.c:17:16: warning: passing a parameter declared with the 'register' keyword to 'va_start' has undefined behavior [-Wvarargs]
  va_start(ap, paramN); /* { dg-warning "undefined behavior when second parameter of 'va_start' is declared with 'register' storage" } */
               ^
c_files/pr30457.c:13:25: note: parameter of type 'short' is declared here
void foo(register short paramN, ...)
                        ^
c_files/pr30457.c:20:21: warning: second argument to 'va_arg' is of promotable type 'char'; this va_arg has undefined behavior because arguments will be promoted to 'int' [-Wvarargs]
  (void) va_arg(ap, char); /* { dg-warning "'char' is promoted to 'int' when passed through '...'" "promoted" } */
                    ^~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/9.1.0/include/stdarg.h:35:50: note: expanded from macro 'va_arg'
#define va_arg(ap, type)    __builtin_va_arg(ap, type)
                                                 ^~~~
2 warnings generated.
TranslationUnitDecl 0x7f8e868264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7f8e86826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7f8e86826740 '__int128'
|-TypedefDecl 0x7f8e86826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7f8e86826760 'unsigned __int128'
|-TypedefDecl 0x7f8e86826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7f8e86826b70 'struct __NSConstantString_tag'
|   `-Record 0x7f8e86826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7f8e86826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7f8e86826dc0 'char *'
|   `-BuiltinType 0x7f8e86826560 'char'
|-TypedefDecl 0x7f8e868270c8 <<invalid sloc>> <invalid sloc> implicit referenced __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7f8e86827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7f8e86826ee0 'struct __va_list_tag'
|     `-Record 0x7f8e86826e58 '__va_list_tag'
|-TypedefDecl 0x7f8e86827150 </Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/9.1.0/include/stdarg.h:30:1, col:27> col:27 referenced va_list '__builtin_va_list':'struct __va_list_tag [1]'
| `-TypedefType 0x7f8e86827120 '__builtin_va_list' sugar
|   |-Typedef 0x7f8e868270c8 '__builtin_va_list'
|   `-ConstantArrayType 0x7f8e86827070 'struct __va_list_tag [1]' 1 
|     `-RecordType 0x7f8e86826ee0 'struct __va_list_tag'
|       `-Record 0x7f8e86826e58 '__va_list_tag'
|-TypedefDecl 0x7f8e87027a00 <line:48:1, col:27> col:27 __gnuc_va_list '__builtin_va_list':'struct __va_list_tag [1]'
| `-TypedefType 0x7f8e86827120 '__builtin_va_list' sugar
|   |-Typedef 0x7f8e868270c8 '__builtin_va_list'
|   `-ConstantArrayType 0x7f8e86827070 'struct __va_list_tag [1]' 1 
|     `-RecordType 0x7f8e86826ee0 'struct __va_list_tag'
|       `-Record 0x7f8e86826e58 '__va_list_tag'
|-FunctionDecl 0x7f8e87027b30 <c_files/pr30457.c:13:1, line:25:1> line:13:6 foo 'void (short, ...)'
| |-ParmVarDecl 0x7f8e87027a70 <col:10, col:25> col:25 used paramN 'short' register
| `-CompoundStmt 0x7f8e87028618 <line:14:1, line:25:1>
|   |-DeclStmt 0x7f8e87027ca0 <line:15:3, col:13>
|   | `-VarDecl 0x7f8e87027c40 <col:3, col:11> col:11 used ap 'va_list':'struct __va_list_tag [1]'
|   |-CallExpr 0x7f8e87027f30 </Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/9.1.0/include/stdarg.h:33:29, col:57> 'void'
|   | |-ImplicitCastExpr 0x7f8e87027f18 <col:29> 'void (*)(struct __va_list_tag *, ...)' <BuiltinFnToFnPtr>
|   | | `-DeclRefExpr 0x7f8e87027e70 <col:29> '<builtin fn type>' Function 0x7f8e87027d28 '__builtin_va_start' 'void (struct __va_list_tag *, ...)'
|   | |-ImplicitCastExpr 0x7f8e87027f68 <c_files/pr30457.c:17:12> 'struct __va_list_tag *' <ArrayToPointerDecay>
|   | | `-DeclRefExpr 0x7f8e87027e98 <col:12> 'va_list':'struct __va_list_tag [1]' lvalue Var 0x7f8e87027c40 'ap' 'va_list':'struct __va_list_tag [1]'
|   | `-DeclRefExpr 0x7f8e87027ec0 <col:16> 'short' lvalue ParmVar 0x7f8e87027a70 'paramN' 'short'
|   |-CStyleCastExpr 0x7f8e87028398 <line:20:3, /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/9.1.0/include/stdarg.h:35:54> 'void' <ToVoid>
|   | `-VAArgExpr 0x7f8e87028360 <col:29, col:54> 'char'
|   |   `-ImplicitCastExpr 0x7f8e87028348 <c_files/pr30457.c:20:17> 'struct __va_list_tag *' <ArrayToPointerDecay>
|   |     `-DeclRefExpr 0x7f8e87028310 <col:17> 'va_list':'struct __va_list_tag [1]' lvalue Var 0x7f8e87027c40 'ap' 'va_list':'struct __va_list_tag [1]'
|   `-CallExpr 0x7f8e870285d0 </Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/9.1.0/include/stdarg.h:34:29, col:48> 'void'
|     |-ImplicitCastExpr 0x7f8e870285b8 <col:29> 'void (*)(struct __va_list_tag *)' <BuiltinFnToFnPtr>
|     | `-DeclRefExpr 0x7f8e87028540 <col:29> '<builtin fn type>' Function 0x7f8e870283f8 '__builtin_va_end' 'void (struct __va_list_tag *)'
|     `-ImplicitCastExpr 0x7f8e87028600 <c_files/pr30457.c:24:10> 'struct __va_list_tag *' <ArrayToPointerDecay>
|       `-DeclRefExpr 0x7f8e87028568 <col:10> 'va_list':'struct __va_list_tag [1]' lvalue Var 0x7f8e87027c40 'ap' 'va_list':'struct __va_list_tag [1]'
|-FunctionDecl 0x7f8e87027d28 </Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/9.1.0/include/stdarg.h:33:29> col:29 implicit used __builtin_va_start 'void (struct __va_list_tag *, ...)' extern
| |-ParmVarDecl 0x7f8e87027dc0 <<invalid sloc>> <invalid sloc> 'struct __va_list_tag *'
| `-NoThrowAttr 0x7f8e87027e28 <col:29> Implicit
`-FunctionDecl 0x7f8e870283f8 <line:34:29> col:29 implicit used __builtin_va_end 'void (struct __va_list_tag *)' extern
  |-ParmVarDecl 0x7f8e87028490 <<invalid sloc>> <invalid sloc> 'struct __va_list_tag *'
  `-NoThrowAttr 0x7f8e870284f8 <col:29> Implicit
