c_files/pr35738.c:13:5: warning: incompatible pointer to integer conversion assigning to 'int' from 'char *' [-Wint-conversion]
  i += q;               /* { dg-warning "makes integer from pointer without a cast" } */
    ^  ~
c_files/pr35738.c:15:5: warning: incompatible pointer to integer conversion assigning to 'int' from 'void (*)(void)' [-Wint-conversion]
  i += foo;             /* { dg-warning "makes integer from pointer without a cast" } */
    ^  ~~~
c_files/pr35738.c:17:5: warning: incompatible pointer to integer conversion assigning to 'int' from 'void *' [-Wint-conversion]
  i += p;               /* { dg-warning "makes integer from pointer without a cast" } */
    ^  ~
3 warnings generated.
TranslationUnitDecl 0x7fb7008264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fb700826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fb700826740 '__int128'
|-TypedefDecl 0x7fb700826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fb700826760 'unsigned __int128'
|-TypedefDecl 0x7fb700826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fb700826b70 'struct __NSConstantString_tag'
|   `-Record 0x7fb700826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fb700826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fb700826dc0 'char *'
|   `-BuiltinType 0x7fb700826560 'char'
|-TypedefDecl 0x7fb7008270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fb700827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fb700826ee0 'struct __va_list_tag'
|     `-Record 0x7fb700826e58 '__va_list_tag'
|-FunctionDecl 0x7fb701025a00 <c_files/pr35738.c:5:1, col:15> col:6 used foo 'void (void)'
`-FunctionDecl 0x7fb701025bb0 <line:7:1, line:18:1> line:8:1 bar 'void (void *)'
  |-ParmVarDecl 0x7fb701025af0 <col:6, col:12> col:12 used p 'void *'
  `-CompoundStmt 0x7fb7010263b8 <line:9:1, line:18:1>
    |-DeclStmt 0x7fb701025ce8 <line:10:3, col:12>
    | `-VarDecl 0x7fb701025c68 <col:3, col:11> col:7 used i 'int' cinit
    |   `-IntegerLiteral 0x7fb701025cc8 <col:11> 'int' 0
    |-DeclStmt 0x7fb701025dd8 <line:11:3, col:13>
    | `-VarDecl 0x7fb701025d78 <col:3, col:12> col:8 used q 'char [10]'
    |-CompoundAssignOperator 0x7fb701026210 <line:13:3, col:8> 'int' '+=' ComputeLHSTy='int' ComputeResultTy='char *'
    | |-DeclRefExpr 0x7fb701025df0 <col:3> 'int' lvalue Var 0x7fb701025c68 'i' 'int'
    | `-ImplicitCastExpr 0x7fb701025e40 <col:8> 'char *' <ArrayToPointerDecay>
    |   `-DeclRefExpr 0x7fb701025e18 <col:8> 'char [10]' lvalue Var 0x7fb701025d78 'q' 'char [10]'
    |-CompoundAssignOperator 0x7fb7010262e0 <line:15:3, col:8> 'int' '+=' ComputeLHSTy='int' ComputeResultTy='void (*)(void)'
    | |-DeclRefExpr 0x7fb701026248 <col:3> 'int' lvalue Var 0x7fb701025c68 'i' 'int'
    | `-ImplicitCastExpr 0x7fb7010262c8 <col:8> 'void (*)(void)' <FunctionToPointerDecay>
    |   `-DeclRefExpr 0x7fb701026270 <col:8> 'void (void)' Function 0x7fb701025a00 'foo' 'void (void)'
    `-CompoundAssignOperator 0x7fb701026380 <line:17:3, col:8> 'int' '+=' ComputeLHSTy='int' ComputeResultTy='void *'
      |-DeclRefExpr 0x7fb701026318 <col:3> 'int' lvalue Var 0x7fb701025c68 'i' 'int'
      `-ImplicitCastExpr 0x7fb701026368 <col:8> 'void *' <LValueToRValue>
        `-DeclRefExpr 0x7fb701026340 <col:8> 'void *' lvalue ParmVar 0x7fb701025af0 'p' 'void *'
