c_files/pr38957.c:8:10: warning: address of stack memory associated with local variable 'a' returned [-Wreturn-stack-address]
  return a+1;   /* { dg-warning "returns address of local variable" } */
         ^
c_files/pr38957.c:15:10: warning: address of stack memory associated with local variable 'a' returned [-Wreturn-stack-address]
  return a;     /* { dg-warning "returns address of local variable" } */
         ^
2 warnings generated.
TranslationUnitDecl 0x7f85988264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7f8598826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7f8598826740 '__int128'
|-TypedefDecl 0x7f8598826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7f8598826760 'unsigned __int128'
|-TypedefDecl 0x7f8598826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7f8598826b70 'struct __NSConstantString_tag'
|   `-Record 0x7f8598826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7f8598826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7f8598826dc0 'char *'
|   `-BuiltinType 0x7f8598826560 'char'
|-TypedefDecl 0x7f85988270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7f8598827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7f8598826ee0 'struct __va_list_tag'
|     `-Record 0x7f8598826e58 '__va_list_tag'
|-FunctionDecl 0x7f8599025a00 <c_files/pr38957.c:4:1, line:9:1> line:5:1 foo 'char *(void)'
| `-CompoundStmt 0x7f8599025c70 <line:6:1, line:9:1>
|   |-DeclStmt 0x7f8599025bb8 <line:7:3, col:12>
|   | `-VarDecl 0x7f8599025b58 <col:3, col:11> col:8 used a 'char [2]'
|   `-ReturnStmt 0x7f8599025c58 <line:8:3, col:12>
|     `-BinaryOperator 0x7f8599025c30 <col:10, col:12> 'char *' '+'
|       |-ImplicitCastExpr 0x7f8599025c18 <col:10> 'char *' <ArrayToPointerDecay>
|       | `-DeclRefExpr 0x7f8599025bd0 <col:10> 'char [2]' lvalue Var 0x7f8599025b58 'a' 'char [2]'
|       `-IntegerLiteral 0x7f8599025bf8 <col:12> 'int' 1
`-FunctionDecl 0x7f8599025d30 <line:11:1, line:16:1> line:12:1 bar 'char *(void)'
  `-CompoundStmt 0x7f8599025ed8 <line:13:1, line:16:1>
    |-DeclStmt 0x7f8599025e68 <line:14:3, col:12>
    | `-VarDecl 0x7f8599025e08 <col:3, col:11> col:8 used a 'char [2]'
    `-ReturnStmt 0x7f8599025ec0 <line:15:3, col:10>
      `-ImplicitCastExpr 0x7f8599025ea8 <col:10> 'char *' <ArrayToPointerDecay>
        `-DeclRefExpr 0x7f8599025e80 <col:10> 'char [2]' lvalue Var 0x7f8599025e08 'a' 'char [2]'
