c_files/pr59963-2.c:15:13: warning: implicit conversion from 'long' to 'short' changes value from 4294967295 to -1 [-Wconstant-conversion]
  return f (0xffffffffL, /* { dg-warning "13:-Woverflow" } */
         ~  ^~~~~~~~~~~
c_files/pr59963-2.c:16:13: warning: implicit conversion from 'long' to 'short' changes value from 4294967295 to -1 [-Wconstant-conversion]
            0xffffffffL) /* { dg-warning "13:-Woverflow" } */
            ^~~~~~~~~~~
c_files/pr59963-2.c:17:16: warning: implicit conversion from 'long' to 'short' changes value from 4294967295 to -1 [-Wconstant-conversion]
         && f (0xffffffffL, /* { dg-warning "16:-Woverflow" } */
            ~  ^~~~~~~~~~~
c_files/pr59963-2.c:18:16: warning: implicit conversion from 'long' to 'short' changes value from 4294967295 to -1 [-Wconstant-conversion]
               0xffffffffL); /* { dg-warning "16:-Woverflow" } */
               ^~~~~~~~~~~
c_files/pr59963-2.c:25:8: warning: implicit conversion from 'float' to 'unsigned char' changes value from 6.6599998 to 6 [-Wliteral-conversion]
  bar (6.66f); /* { dg-warning "8:conversion" } */
  ~~~  ^~~~~
c_files/pr59963-2.c:24:8: warning: implicit conversion from 'int' to 'unsigned char' changes value from 256 to 0 [-Wconstant-conversion]
  bar (256); /* { dg-warning "8:-Woverflow" } */
  ~~~  ^~~
6 warnings generated.
TranslationUnitDecl 0x7fd52b8264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fd52b826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fd52b826740 '__int128'
|-TypedefDecl 0x7fd52b826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fd52b826760 'unsigned __int128'
|-TypedefDecl 0x7fd52b826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fd52b826b70 'struct __NSConstantString_tag'
|   `-Record 0x7fd52b826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fd52b826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fd52b826dc0 'char *'
|   `-BuiltinType 0x7fd52b826560 'char'
|-TypedefDecl 0x7fd52b8270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fd52b827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fd52b826ee0 'struct __va_list_tag'
|     `-Record 0x7fd52b826e58 '__va_list_tag'
|-FunctionDecl 0x7fd52c025a00 <c_files/pr59963-2.c:5:1, col:31> col:13 used bar 'void (unsigned char)' extern
| `-ParmVarDecl 0x7fd52b827138 <col:18, col:27> col:31 'unsigned char'
|-FunctionDecl 0x7fd52c025f70 <line:6:1, line:7:63> line:6:13 used bar8 'void (unsigned char, unsigned char, unsigned char, unsigned char, unsigned char, unsigned char, unsigned char, unsigned char)' extern
| |-ParmVarDecl 0x7fd52c025af8 <col:19, col:28> col:32 'unsigned char'
| |-ParmVarDecl 0x7fd52c025b70 <col:34, col:43> col:47 'unsigned char'
| |-ParmVarDecl 0x7fd52c025be8 <col:49, col:58> col:62 'unsigned char'
| |-ParmVarDecl 0x7fd52c025c60 <col:64, col:73> col:77 'unsigned char'
| |-ParmVarDecl 0x7fd52c025cd8 <line:7:5, col:14> col:18 'unsigned char'
| |-ParmVarDecl 0x7fd52c025d50 <col:20, col:29> col:33 'unsigned char'
| |-ParmVarDecl 0x7fd52c025dc8 <col:35, col:44> col:48 'unsigned char'
| `-ParmVarDecl 0x7fd52c025e40 <col:50, col:59> col:63 'unsigned char'
|-FunctionDecl 0x7fd52c026120 <line:8:1, col:36> col:13 used bazu 'void (unsigned int, ...)' extern
| `-ParmVarDecl 0x7fd52c026060 <col:19, col:28> col:31 'unsigned int'
|-FunctionDecl 0x7fd52c026320 <line:9:1, col:28> col:13 used bazi 'void (char, int)' extern
| |-ParmVarDecl 0x7fd52c0261d0 <col:19> col:23 'char'
| `-ParmVarDecl 0x7fd52c026248 <col:25> col:28 'int'
|-FunctionDecl 0x7fd52c026538 <line:10:1, col:31> col:12 used f 'int (short, short)' extern
| |-ParmVarDecl 0x7fd52c0263e0 <col:15, col:21> col:21 a 'short'
| `-ParmVarDecl 0x7fd52c026458 <col:24, col:30> col:30 b 'short'
|-FunctionDecl 0x7fd52c0266a8 <line:12:1, line:19:1> line:13:1 g 'int (void)'
| `-CompoundStmt 0x7fd52c026980 <line:14:1, line:19:1>
|   `-ReturnStmt 0x7fd52c026968 <line:15:3, line:18:27>
|     `-BinaryOperator 0x7fd52c026940 <line:15:10, line:18:27> 'int' '&&'
|       |-CallExpr 0x7fd52c0267f0 <line:15:10, line:16:24> 'int'
|       | |-ImplicitCastExpr 0x7fd52c0267d8 <line:15:10> 'int (*)(short, short)' <FunctionToPointerDecay>
|       | | `-DeclRefExpr 0x7fd52c026740 <col:10> 'int (short, short)' Function 0x7fd52c026538 'f' 'int (short, short)'
|       | |-ImplicitCastExpr 0x7fd52c026828 <col:13> 'short' <IntegralCast>
|       | | `-IntegerLiteral 0x7fd52c026768 <col:13> 'long' 4294967295
|       | `-ImplicitCastExpr 0x7fd52c026840 <line:16:13> 'short' <IntegralCast>
|       |   `-IntegerLiteral 0x7fd52c026788 <col:13> 'long' 4294967295
|       `-CallExpr 0x7fd52c0268d8 <line:17:13, line:18:27> 'int'
|         |-ImplicitCastExpr 0x7fd52c0268c0 <line:17:13> 'int (*)(short, short)' <FunctionToPointerDecay>
|         | `-DeclRefExpr 0x7fd52c026858 <col:13> 'int (short, short)' Function 0x7fd52c026538 'f' 'int (short, short)'
|         |-ImplicitCastExpr 0x7fd52c026910 <col:16> 'short' <IntegralCast>
|         | `-IntegerLiteral 0x7fd52c026880 <col:16> 'long' 4294967295
|         `-ImplicitCastExpr 0x7fd52c026928 <line:18:16> 'short' <IntegralCast>
|           `-IntegerLiteral 0x7fd52c0268a0 <col:16> 'long' 4294967295
`-FunctionDecl 0x7fd52c027e40 <line:21:1, line:36:1> line:22:1 foo 'void (int)'
  |-ParmVarDecl 0x7fd52c027d78 <col:6, col:10> col:10 used i 'int'
  `-CompoundStmt 0x7fd52c028648 <line:23:1, line:36:1>
    |-CallExpr 0x7fd52c027f70 <line:24:3, col:11> 'void'
    | |-ImplicitCastExpr 0x7fd52c027f58 <col:3> 'void (*)(unsigned char)' <FunctionToPointerDecay>
    | | `-DeclRefExpr 0x7fd52c027ee0 <col:3> 'void (unsigned char)' Function 0x7fd52c025a00 'bar' 'void (unsigned char)'
    | `-ImplicitCastExpr 0x7fd52c027fa0 <col:8> 'unsigned char' <IntegralCast>
    |   `-IntegerLiteral 0x7fd52c027f08 <col:8> 'int' 256
    |-CallExpr 0x7fd52c028018 <line:25:3, col:13> 'void'
    | |-ImplicitCastExpr 0x7fd52c028000 <col:3> 'void (*)(unsigned char)' <FunctionToPointerDecay>
    | | `-DeclRefExpr 0x7fd52c027fb8 <col:3> 'void (unsigned char)' Function 0x7fd52c025a00 'bar' 'void (unsigned char)'
    | `-ImplicitCastExpr 0x7fd52c028048 <col:8> 'unsigned char' <FloatingToIntegral>
    |   `-FloatingLiteral 0x7fd52c027fe0 <col:8> 'float' 6.660000e+00
    |-CallExpr 0x7fd52c0282d0 <line:26:3, line:33:18> 'void'
    | |-ImplicitCastExpr 0x7fd52c0282b8 <line:26:3> 'void (*)(unsigned char, unsigned char, unsigned char, unsigned char, unsigned char, unsigned char, unsigned char, unsigned char)' <FunctionToPointerDecay>
    | | `-DeclRefExpr 0x7fd52c028060 <col:3> 'void (unsigned char, unsigned char, unsigned char, unsigned char, unsigned char, unsigned char, unsigned char, unsigned char)' Function 0x7fd52c025f70 'bar8' 'void (unsigned char, unsigned char, unsigned char, unsigned char, unsigned char, unsigned char, unsigned char, unsigned char)'
    | |-ImplicitCastExpr 0x7fd52c028338 <col:9, col:10> 'unsigned char' <IntegralCast>
    | | `-UnaryOperator 0x7fd52c0280a8 <col:9, col:10> 'int' prefix '-'
    | |   `-IntegerLiteral 0x7fd52c028088 <col:10> 'int' 1
    | |-ImplicitCastExpr 0x7fd52c028350 <line:27:10, col:11> 'unsigned char' <IntegralCast>
    | | `-UnaryOperator 0x7fd52c0280e8 <col:10, col:11> 'int' prefix '-'
    | |   `-IntegerLiteral 0x7fd52c0280c8 <col:11> 'int' 2
    | |-ImplicitCastExpr 0x7fd52c028368 <line:28:11, col:12> 'unsigned char' <IntegralCast>
    | | `-UnaryOperator 0x7fd52c028128 <col:11, col:12> 'int' prefix '-'
    | |   `-IntegerLiteral 0x7fd52c028108 <col:12> 'int' 3
    | |-ImplicitCastExpr 0x7fd52c028380 <line:29:12, col:13> 'unsigned char' <IntegralCast>
    | | `-UnaryOperator 0x7fd52c028168 <col:12, col:13> 'int' prefix '-'
    | |   `-IntegerLiteral 0x7fd52c028148 <col:13> 'int' 4
    | |-ImplicitCastExpr 0x7fd52c028398 <line:30:13, col:14> 'unsigned char' <IntegralCast>
    | | `-UnaryOperator 0x7fd52c0281a8 <col:13, col:14> 'int' prefix '-'
    | |   `-IntegerLiteral 0x7fd52c028188 <col:14> 'int' 5
    | |-ImplicitCastExpr 0x7fd52c0283b0 <line:31:14, col:15> 'unsigned char' <IntegralCast>
    | | `-UnaryOperator 0x7fd52c0281e8 <col:14, col:15> 'int' prefix '-'
    | |   `-IntegerLiteral 0x7fd52c0281c8 <col:15> 'int' 6
    | |-ImplicitCastExpr 0x7fd52c0283c8 <line:32:15, col:16> 'unsigned char' <IntegralCast>
    | | `-UnaryOperator 0x7fd52c028228 <col:15, col:16> 'int' prefix '-'
    | |   `-IntegerLiteral 0x7fd52c028208 <col:16> 'int' 7
    | `-ImplicitCastExpr 0x7fd52c0283e0 <line:33:16, col:17> 'unsigned char' <IntegralCast>
    |   `-UnaryOperator 0x7fd52c028268 <col:16, col:17> 'int' prefix '-'
    |     `-IntegerLiteral 0x7fd52c028248 <col:17> 'int' 8
    |-CallExpr 0x7fd52c0284b0 <line:34:3, col:13> 'void'
    | |-ImplicitCastExpr 0x7fd52c028498 <col:3> 'void (*)(unsigned int, ...)' <FunctionToPointerDecay>
    | | `-DeclRefExpr 0x7fd52c0283f8 <col:3> 'void (unsigned int, ...)' Function 0x7fd52c026120 'bazu' 'void (unsigned int, ...)'
    | |-ImplicitCastExpr 0x7fd52c028500 <col:9> 'unsigned int' <IntegralCast>
    | | `-ImplicitCastExpr 0x7fd52c0284e8 <col:9> 'int' <LValueToRValue>
    | |   `-DeclRefExpr 0x7fd52c028420 <col:9> 'int' lvalue ParmVar 0x7fd52c027d78 'i' 'int'
    | `-ImplicitCastExpr 0x7fd52c028518 <col:12> 'int' <LValueToRValue>
    |   `-DeclRefExpr 0x7fd52c028448 <col:12> 'int' lvalue ParmVar 0x7fd52c027d78 'i' 'int'
    `-CallExpr 0x7fd52c0285e0 <line:35:3, col:24> 'void'
      |-ImplicitCastExpr 0x7fd52c0285c8 <col:3> 'void (*)(char, int)' <FunctionToPointerDecay>
      | `-DeclRefExpr 0x7fd52c028530 <col:3> 'void (char, int)' Function 0x7fd52c026320 'bazi' 'void (char, int)'
      |-ImplicitCastExpr 0x7fd52c028618 <col:9> 'char' <IntegralCast>
      | `-IntegerLiteral 0x7fd52c028558 <col:9> 'int' 8
      `-ImplicitCastExpr 0x7fd52c028630 <col:14> 'int' <IntegralCast>
        `-IntegerLiteral 0x7fd52c028578 <col:14> 'unsigned int' 2147483648
