c_files/pr66066-2.c:6:13: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
int a1 = -1 << 0;               /* { dg-warning "initializer element is not a constant expression" } */
         ~~ ^
c_files/pr66066-2.c:7:13: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
int a2 = -1 << 0 | 0;           /* { dg-warning "initializer element is not a constant expression" } */
         ~~ ^
c_files/pr66066-2.c:8:13: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
int a3 = -1 << 0 & 1;           /* { dg-warning "initializer element is not a constant expression" } */
         ~~ ^
c_files/pr66066-2.c:9:13: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
int a4 = -1 << 2 ^ 1;           /* { dg-warning "initializer element is not a constant expression" } */
         ~~ ^
c_files/pr66066-2.c:10:17: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
int a5 = 4 & -1 << 2;           /* { dg-warning "initializer element is not a constant expression" } */
             ~~ ^
c_files/pr66066-2.c:11:14: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
int a6 = (-1 << 2) ^ (1 >> 1);  /* { dg-warning "initializer element is not a constant expression" } */
          ~~ ^
c_files/pr66066-2.c:12:19: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
int a7 = 0 || (-1 << 1);        /* { dg-warning "initializer element is not a constant expression" } */
               ~~ ^
c_files/pr66066-2.c:12:12: warning: use of logical '||' with constant operand [-Wconstant-logical-operand]
int a7 = 0 || (-1 << 1);        /* { dg-warning "initializer element is not a constant expression" } */
           ^  ~~~~~~~~~
c_files/pr66066-2.c:12:12: note: use '|' for a bitwise operation
int a7 = 0 || (-1 << 1);        /* { dg-warning "initializer element is not a constant expression" } */
           ^~
           |
c_files/pr66066-2.c:13:22: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
int a8 = 0 ? 2 : (-1 << 1);     /* { dg-warning "initializer element is not a constant expression" } */
                  ~~ ^
c_files/pr66066-2.c:14:18: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
int a9 = 1 && -1 << 0;          /* { dg-warning "initializer element is not a constant expression" } */
              ~~ ^
c_files/pr66066-2.c:14:12: warning: use of logical '&&' with constant operand [-Wconstant-logical-operand]
int a9 = 1 && -1 << 0;          /* { dg-warning "initializer element is not a constant expression" } */
           ^  ~~~~~~~
c_files/pr66066-2.c:14:12: note: use '&' for a bitwise operation
int a9 = 1 && -1 << 0;          /* { dg-warning "initializer element is not a constant expression" } */
           ^~
           &
c_files/pr66066-2.c:14:12: note: remove constant to silence this warning
int a9 = 1 && -1 << 0;          /* { dg-warning "initializer element is not a constant expression" } */
          ~^~~~~~~~~~
c_files/pr66066-2.c:15:16: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
int a10 = !(-1 << 0);           /* { dg-warning "initializer element is not a constant expression" } */
            ~~ ^
c_files/pr66066-2.c:18:12: warning: division by zero is undefined [-Wdivision-by-zero]
int b1 = 1 / 0;         /* { dg-error "initializer element is not constant" } */
           ^ ~
c_files/pr66066-2.c:18:12: error: initializer element is not a compile-time constant
int b1 = 1 / 0;         /* { dg-error "initializer element is not constant" } */
         ~~^~~
c_files/pr66066-2.c:19:17: warning: division by zero is undefined [-Wdivision-by-zero]
int b2 = 1 / (1 / 0);   /* { dg-error "initializer element is not constant" } */
                ^ ~
c_files/pr66066-2.c:19:12: error: initializer element is not a compile-time constant
int b2 = 1 / (1 / 0);   /* { dg-error "initializer element is not constant" } */
         ~~^~~~~~~~~
c_files/pr66066-2.c:20:20: warning: division by zero is undefined [-Wdivision-by-zero]
int b3 = 0 ? 2 : 1 / 0; /* { dg-error "initializer element is not constant" } */
                   ^ ~
c_files/pr66066-2.c:20:10: error: initializer element is not a compile-time constant
int b3 = 0 ? 2 : 1 / 0; /* { dg-error "initializer element is not constant" } */
         ^~~~~~~~~~~~~
c_files/pr66066-2.c:21:17: warning: division by zero is undefined [-Wdivision-by-zero]
int b4 = 0 || 1 / 0;    /* { dg-error "initializer element is not constant" } */
                ^ ~
c_files/pr66066-2.c:21:12: error: initializer element is not a compile-time constant
int b4 = 0 || 1 / 0;    /* { dg-error "initializer element is not constant" } */
         ~~^~~~~~~~
c_files/pr66066-2.c:22:17: warning: division by zero is undefined [-Wdivision-by-zero]
int b5 = 0 * (1 / 0);   /* { dg-error "initializer element is not constant" } */
                ^ ~
c_files/pr66066-2.c:22:12: error: initializer element is not a compile-time constant
int b5 = 0 * (1 / 0);   /* { dg-error "initializer element is not constant" } */
         ~~^~~~~~~~~
c_files/pr66066-2.c:23:17: warning: division by zero is undefined [-Wdivision-by-zero]
int b6 = 1 * (1 / 0);   /* { dg-error "initializer element is not constant" } */
                ^ ~
c_files/pr66066-2.c:23:12: error: initializer element is not a compile-time constant
int b6 = 1 * (1 / 0);   /* { dg-error "initializer element is not constant" } */
         ~~^~~~~~~~~
c_files/pr66066-2.c:24:13: warning: division by zero is undefined [-Wdivision-by-zero]
int b7 = (1 / 0) * 0;   /* { dg-error "initializer element is not constant" } */
            ^ ~
c_files/pr66066-2.c:24:18: error: initializer element is not a compile-time constant
int b7 = (1 / 0) * 0;   /* { dg-error "initializer element is not constant" } */
         ~~~~~~~~^~~
c_files/pr66066-2.c:25:13: warning: division by zero is undefined [-Wdivision-by-zero]
int b8 = (1 / 0) * 1;   /* { dg-error "initializer element is not constant" } */
            ^ ~
c_files/pr66066-2.c:25:18: error: initializer element is not a compile-time constant
int b8 = (1 / 0) * 1;   /* { dg-error "initializer element is not constant" } */
         ~~~~~~~~^~~
c_files/pr66066-2.c:26:17: warning: division by zero is undefined [-Wdivision-by-zero]
int b9 = 1 && 1 / 0;    /* { dg-error "initializer element is not constant" } */
                ^ ~
c_files/pr66066-2.c:26:12: error: initializer element is not a compile-time constant
int b9 = 1 && 1 / 0;    /* { dg-error "initializer element is not constant" } */
         ~~^~~~~~~~
c_files/pr66066-2.c:27:15: warning: division by zero is undefined [-Wdivision-by-zero]
int b10 = !(1 / 0);     /* { dg-error "initializer element is not constant" } */
              ^ ~
c_files/pr66066-2.c:27:11: error: initializer element is not a compile-time constant
int b10 = !(1 / 0);     /* { dg-error "initializer element is not constant" } */
          ^~~~~~~~
c_files/pr66066-2.c:28:12: warning: remainder by zero is undefined [-Wdivision-by-zero]
int c1 = 1 % 0;         /* { dg-error "initializer element is not constant" } */
           ^ ~
c_files/pr66066-2.c:28:12: error: initializer element is not a compile-time constant
int c1 = 1 % 0;         /* { dg-error "initializer element is not constant" } */
         ~~^~~
c_files/pr66066-2.c:29:17: warning: remainder by zero is undefined [-Wdivision-by-zero]
int c2 = 1 / (1 % 0);   /* { dg-error "initializer element is not constant" } */
                ^ ~
c_files/pr66066-2.c:29:12: error: initializer element is not a compile-time constant
int c2 = 1 / (1 % 0);   /* { dg-error "initializer element is not constant" } */
         ~~^~~~~~~~~
c_files/pr66066-2.c:30:20: warning: remainder by zero is undefined [-Wdivision-by-zero]
int c3 = 0 ? 2 : 1 % 0; /* { dg-error "initializer element is not constant" } */
                   ^ ~
c_files/pr66066-2.c:30:10: error: initializer element is not a compile-time constant
int c3 = 0 ? 2 : 1 % 0; /* { dg-error "initializer element is not constant" } */
         ^~~~~~~~~~~~~
c_files/pr66066-2.c:31:17: warning: remainder by zero is undefined [-Wdivision-by-zero]
int c4 = 0 || 1 % 0;    /* { dg-error "initializer element is not constant" } */
                ^ ~
c_files/pr66066-2.c:31:12: error: initializer element is not a compile-time constant
int c4 = 0 || 1 % 0;    /* { dg-error "initializer element is not constant" } */
         ~~^~~~~~~~
c_files/pr66066-2.c:32:17: warning: remainder by zero is undefined [-Wdivision-by-zero]
int c5 = 0 * (1 % 0);   /* { dg-error "initializer element is not constant" } */
                ^ ~
c_files/pr66066-2.c:32:12: error: initializer element is not a compile-time constant
int c5 = 0 * (1 % 0);   /* { dg-error "initializer element is not constant" } */
         ~~^~~~~~~~~
c_files/pr66066-2.c:33:17: warning: remainder by zero is undefined [-Wdivision-by-zero]
int c6 = 1 * (1 % 0);   /* { dg-error "initializer element is not constant" } */
                ^ ~
c_files/pr66066-2.c:33:12: error: initializer element is not a compile-time constant
int c6 = 1 * (1 % 0);   /* { dg-error "initializer element is not constant" } */
         ~~^~~~~~~~~
c_files/pr66066-2.c:34:13: warning: remainder by zero is undefined [-Wdivision-by-zero]
int c7 = (1 % 0) * 0;   /* { dg-error "initializer element is not constant" } */
            ^ ~
c_files/pr66066-2.c:34:18: error: initializer element is not a compile-time constant
int c7 = (1 % 0) * 0;   /* { dg-error "initializer element is not constant" } */
         ~~~~~~~~^~~
c_files/pr66066-2.c:35:13: warning: remainder by zero is undefined [-Wdivision-by-zero]
int c8 = (1 % 0) * 1;   /* { dg-error "initializer element is not constant" } */
            ^ ~
c_files/pr66066-2.c:35:18: error: initializer element is not a compile-time constant
int c8 = (1 % 0) * 1;   /* { dg-error "initializer element is not constant" } */
         ~~~~~~~~^~~
c_files/pr66066-2.c:36:17: warning: remainder by zero is undefined [-Wdivision-by-zero]
int c9 = 1 && 1 % 0;    /* { dg-error "initializer element is not constant" } */
                ^ ~
c_files/pr66066-2.c:36:12: error: initializer element is not a compile-time constant
int c9 = 1 && 1 % 0;    /* { dg-error "initializer element is not constant" } */
         ~~^~~~~~~~
c_files/pr66066-2.c:37:15: warning: remainder by zero is undefined [-Wdivision-by-zero]
int c10 = !(1 % 0);     /* { dg-error "initializer element is not constant" } */
              ^ ~
fatal error: too many errors emitted, stopping now [-ferror-limit=]
32 warnings and 20 errors generated.
TranslationUnitDecl 0x7ffa3a8264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7ffa3a826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7ffa3a826740 '__int128'
|-TypedefDecl 0x7ffa3a826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7ffa3a826760 'unsigned __int128'
|-TypedefDecl 0x7ffa3a826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7ffa3a826b70 'struct __NSConstantString_tag'
|   `-Record 0x7ffa3a826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7ffa3a826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7ffa3a826dc0 'char *'
|   `-BuiltinType 0x7ffa3a826560 'char'
|-TypedefDecl 0x7ffa3a8270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7ffa3a827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7ffa3a826ee0 'struct __va_list_tag'
|     `-Record 0x7ffa3a826e58 '__va_list_tag'
|-VarDecl 0x7ffa3a827138 <c_files/pr66066-2.c:6:1, col:16> col:5 a1 'int' cinit
| `-BinaryOperator 0x7ffa3b025a40 <col:10, col:16> 'int' '<<'
|   |-UnaryOperator 0x7ffa3b025a00 <col:10, col:11> 'int' prefix '-'
|   | `-IntegerLiteral 0x7ffa3a8271d8 <col:11> 'int' 1
|   `-IntegerLiteral 0x7ffa3b025a20 <col:16> 'int' 0
|-VarDecl 0x7ffa3b025a80 <line:7:1, col:20> col:5 a2 'int' cinit
| `-BinaryOperator 0x7ffa3b025b88 <col:10, col:20> 'int' '|'
|   |-BinaryOperator 0x7ffa3b025b40 <col:10, col:16> 'int' '<<'
|   | |-UnaryOperator 0x7ffa3b025b00 <col:10, col:11> 'int' prefix '-'
|   | | `-IntegerLiteral 0x7ffa3b025ae0 <col:11> 'int' 1
|   | `-IntegerLiteral 0x7ffa3b025b20 <col:16> 'int' 0
|   `-IntegerLiteral 0x7ffa3b025b68 <col:20> 'int' 0
|-VarDecl 0x7ffa3b025bc8 <line:8:1, col:20> col:5 a3 'int' cinit
| `-BinaryOperator 0x7ffa3b025cd0 <col:10, col:20> 'int' '&'
|   |-BinaryOperator 0x7ffa3b025c88 <col:10, col:16> 'int' '<<'
|   | |-UnaryOperator 0x7ffa3b025c48 <col:10, col:11> 'int' prefix '-'
|   | | `-IntegerLiteral 0x7ffa3b025c28 <col:11> 'int' 1
|   | `-IntegerLiteral 0x7ffa3b025c68 <col:16> 'int' 0
|   `-IntegerLiteral 0x7ffa3b025cb0 <col:20> 'int' 1
|-VarDecl 0x7ffa3b025d10 <line:9:1, col:20> col:5 a4 'int' cinit
| `-BinaryOperator 0x7ffa3b025e18 <col:10, col:20> 'int' '^'
|   |-BinaryOperator 0x7ffa3b025dd0 <col:10, col:16> 'int' '<<'
|   | |-UnaryOperator 0x7ffa3b025d90 <col:10, col:11> 'int' prefix '-'
|   | | `-IntegerLiteral 0x7ffa3b025d70 <col:11> 'int' 1
|   | `-IntegerLiteral 0x7ffa3b025db0 <col:16> 'int' 2
|   `-IntegerLiteral 0x7ffa3b025df8 <col:20> 'int' 1
|-VarDecl 0x7ffa3b025e58 <line:10:1, col:20> col:5 a5 'int' cinit
| `-BinaryOperator 0x7ffa3b025f60 <col:10, col:20> 'int' '&'
|   |-IntegerLiteral 0x7ffa3b025eb8 <col:10> 'int' 4
|   `-BinaryOperator 0x7ffa3b025f38 <col:14, col:20> 'int' '<<'
|     |-UnaryOperator 0x7ffa3b025ef8 <col:14, col:15> 'int' prefix '-'
|     | `-IntegerLiteral 0x7ffa3b025ed8 <col:15> 'int' 1
|     `-IntegerLiteral 0x7ffa3b025f18 <col:20> 'int' 2
|-VarDecl 0x7ffa3b025fa0 <line:11:1, col:29> col:5 a6 'int' cinit
| `-BinaryOperator 0x7ffa3b026130 <col:10, col:29> 'int' '^'
|   |-ParenExpr 0x7ffa3b026088 <col:10, col:18> 'int'
|   | `-BinaryOperator 0x7ffa3b026060 <col:11, col:17> 'int' '<<'
|   |   |-UnaryOperator 0x7ffa3b026020 <col:11, col:12> 'int' prefix '-'
|   |   | `-IntegerLiteral 0x7ffa3b026000 <col:12> 'int' 1
|   |   `-IntegerLiteral 0x7ffa3b026040 <col:17> 'int' 2
|   `-ParenExpr 0x7ffa3b026110 <col:22, col:29> 'int'
|     `-BinaryOperator 0x7ffa3b0260e8 <col:23, col:28> 'int' '>>'
|       |-IntegerLiteral 0x7ffa3b0260a8 <col:23> 'int' 1
|       `-IntegerLiteral 0x7ffa3b0260c8 <col:28> 'int' 1
|-VarDecl 0x7ffa3b026170 <line:12:1, col:23> col:5 a7 'int' cinit
| `-BinaryOperator 0x7ffa3b026298 <col:10, col:23> 'int' '||'
|   |-IntegerLiteral 0x7ffa3b0261d0 <col:10> 'int' 0
|   `-ParenExpr 0x7ffa3b026278 <col:15, col:23> 'int'
|     `-BinaryOperator 0x7ffa3b026250 <col:16, col:22> 'int' '<<'
|       |-UnaryOperator 0x7ffa3b026210 <col:16, col:17> 'int' prefix '-'
|       | `-IntegerLiteral 0x7ffa3b0261f0 <col:17> 'int' 1
|       `-IntegerLiteral 0x7ffa3b026230 <col:22> 'int' 1
|-VarDecl 0x7ffa3b0262d8 <line:13:1, col:26> col:5 a8 'int' cinit
| `-ConditionalOperator 0x7ffa3b026420 <col:10, col:26> 'int'
|   |-IntegerLiteral 0x7ffa3b026338 <col:10> 'int' 0
|   |-IntegerLiteral 0x7ffa3b026358 <col:14> 'int' 2
|   `-ParenExpr 0x7ffa3b026400 <col:18, col:26> 'int'
|     `-BinaryOperator 0x7ffa3b0263d8 <col:19, col:25> 'int' '<<'
|       |-UnaryOperator 0x7ffa3b026398 <col:19, col:20> 'int' prefix '-'
|       | `-IntegerLiteral 0x7ffa3b026378 <col:20> 'int' 1
|       `-IntegerLiteral 0x7ffa3b0263b8 <col:25> 'int' 1
|-VarDecl 0x7ffa3b026468 <line:14:1, col:21> col:5 a9 'int' cinit
| `-BinaryOperator 0x7ffa3b026570 <col:10, col:21> 'int' '&&'
|   |-IntegerLiteral 0x7ffa3b0264c8 <col:10> 'int' 1
|   `-BinaryOperator 0x7ffa3b026548 <col:15, col:21> 'int' '<<'
|     |-UnaryOperator 0x7ffa3b026508 <col:15, col:16> 'int' prefix '-'
|     | `-IntegerLiteral 0x7ffa3b0264e8 <col:16> 'int' 1
|     `-IntegerLiteral 0x7ffa3b026528 <col:21> 'int' 0
|-VarDecl 0x7ffa3b0265b0 <line:15:1, col:20> col:5 a10 'int' cinit
| `-UnaryOperator 0x7ffa3b0266b8 <col:11, col:20> 'int' prefix '!'
|   `-ParenExpr 0x7ffa3b026698 <col:12, col:20> 'int'
|     `-BinaryOperator 0x7ffa3b026670 <col:13, col:19> 'int' '<<'
|       |-UnaryOperator 0x7ffa3b026630 <col:13, col:14> 'int' prefix '-'
|       | `-IntegerLiteral 0x7ffa3b026610 <col:14> 'int' 1
|       `-IntegerLiteral 0x7ffa3b026650 <col:19> 'int' 0
|-VarDecl 0x7ffa3b0266f0 <line:18:1, col:14> col:5 b1 'int' cinit
| `-BinaryOperator 0x7ffa3b026790 <col:10, col:14> 'int' '/'
|   |-IntegerLiteral 0x7ffa3b026750 <col:10> 'int' 1
|   `-IntegerLiteral 0x7ffa3b026770 <col:14> 'int' 0
|-VarDecl 0x7ffa3b0267d0 <line:19:1, col:20> col:5 b2 'int' cinit
| `-BinaryOperator 0x7ffa3b0268d8 <col:10, col:20> 'int' '/'
|   |-IntegerLiteral 0x7ffa3b026830 <col:10> 'int' 1
|   `-ParenExpr 0x7ffa3b0268b8 <col:14, col:20> 'int'
|     `-BinaryOperator 0x7ffa3b026890 <col:15, col:19> 'int' '/'
|       |-IntegerLiteral 0x7ffa3b026850 <col:15> 'int' 1
|       `-IntegerLiteral 0x7ffa3b026870 <col:19> 'int' 0
|-VarDecl 0x7ffa3b026918 <line:20:1, col:22> col:5 b3 'int' cinit
| `-ConditionalOperator 0x7ffa3b027a28 <col:10, col:22> 'int'
|   |-IntegerLiteral 0x7ffa3b026978 <col:10> 'int' 0
|   |-IntegerLiteral 0x7ffa3b026998 <col:14> 'int' 2
|   `-BinaryOperator 0x7ffa3b027a00 <col:18, col:22> 'int' '/'
|     |-IntegerLiteral 0x7ffa3b0269b8 <col:18> 'int' 1
|     `-IntegerLiteral 0x7ffa3b0269d8 <col:22> 'int' 0
|-VarDecl 0x7ffa3b027a70 <line:21:1, col:19> col:5 b4 'int' cinit
| `-BinaryOperator 0x7ffa3b027b58 <col:10, col:19> 'int' '||'
|   |-IntegerLiteral 0x7ffa3b027ad0 <col:10> 'int' 0
|   `-BinaryOperator 0x7ffa3b027b30 <col:15, col:19> 'int' '/'
|     |-IntegerLiteral 0x7ffa3b027af0 <col:15> 'int' 1
|     `-IntegerLiteral 0x7ffa3b027b10 <col:19> 'int' 0
|-VarDecl 0x7ffa3b027b98 <line:22:1, col:20> col:5 b5 'int' cinit
| `-BinaryOperator 0x7ffa3b027ca0 <col:10, col:20> 'int' '*'
|   |-IntegerLiteral 0x7ffa3b027bf8 <col:10> 'int' 0
|   `-ParenExpr 0x7ffa3b027c80 <col:14, col:20> 'int'
|     `-BinaryOperator 0x7ffa3b027c58 <col:15, col:19> 'int' '/'
|       |-IntegerLiteral 0x7ffa3b027c18 <col:15> 'int' 1
|       `-IntegerLiteral 0x7ffa3b027c38 <col:19> 'int' 0
|-VarDecl 0x7ffa3b027ce0 <line:23:1, col:20> col:5 b6 'int' cinit
| `-BinaryOperator 0x7ffa3b027de8 <col:10, col:20> 'int' '*'
|   |-IntegerLiteral 0x7ffa3b027d40 <col:10> 'int' 1
|   `-ParenExpr 0x7ffa3b027dc8 <col:14, col:20> 'int'
|     `-BinaryOperator 0x7ffa3b027da0 <col:15, col:19> 'int' '/'
|       |-IntegerLiteral 0x7ffa3b027d60 <col:15> 'int' 1
|       `-IntegerLiteral 0x7ffa3b027d80 <col:19> 'int' 0
|-VarDecl 0x7ffa3b027e28 <line:24:1, col:20> col:5 b7 'int' cinit
| `-BinaryOperator 0x7ffa3b027f30 <col:10, col:20> 'int' '*'
|   |-ParenExpr 0x7ffa3b027ef0 <col:10, col:16> 'int'
|   | `-BinaryOperator 0x7ffa3b027ec8 <col:11, col:15> 'int' '/'
|   |   |-IntegerLiteral 0x7ffa3b027e88 <col:11> 'int' 1
|   |   `-IntegerLiteral 0x7ffa3b027ea8 <col:15> 'int' 0
|   `-IntegerLiteral 0x7ffa3b027f10 <col:20> 'int' 0
|-VarDecl 0x7ffa3b027f70 <line:25:1, col:20> col:5 b8 'int' cinit
| `-BinaryOperator 0x7ffa3b028078 <col:10, col:20> 'int' '*'
|   |-ParenExpr 0x7ffa3b028038 <col:10, col:16> 'int'
|   | `-BinaryOperator 0x7ffa3b028010 <col:11, col:15> 'int' '/'
|   |   |-IntegerLiteral 0x7ffa3b027fd0 <col:11> 'int' 1
|   |   `-IntegerLiteral 0x7ffa3b027ff0 <col:15> 'int' 0
|   `-IntegerLiteral 0x7ffa3b028058 <col:20> 'int' 1
|-VarDecl 0x7ffa3b0280b8 <line:26:1, col:19> col:5 b9 'int' cinit
| `-BinaryOperator 0x7ffa3b0281a0 <col:10, col:19> 'int' '&&'
|   |-IntegerLiteral 0x7ffa3b028118 <col:10> 'int' 1
|   `-BinaryOperator 0x7ffa3b028178 <col:15, col:19> 'int' '/'
|     |-IntegerLiteral 0x7ffa3b028138 <col:15> 'int' 1
|     `-IntegerLiteral 0x7ffa3b028158 <col:19> 'int' 0
|-VarDecl 0x7ffa3b0281e0 <line:27:1, col:18> col:5 b10 'int' cinit
| `-UnaryOperator 0x7ffa3b0282c8 <col:11, col:18> 'int' prefix '!'
|   `-ParenExpr 0x7ffa3b0282a8 <col:12, col:18> 'int'
|     `-BinaryOperator 0x7ffa3b028280 <col:13, col:17> 'int' '/'
|       |-IntegerLiteral 0x7ffa3b028240 <col:13> 'int' 1
|       `-IntegerLiteral 0x7ffa3b028260 <col:17> 'int' 0
|-VarDecl 0x7ffa3b028300 <line:28:1, col:14> col:5 c1 'int' cinit
| `-BinaryOperator 0x7ffa3b0283a0 <col:10, col:14> 'int' '%'
|   |-IntegerLiteral 0x7ffa3b028360 <col:10> 'int' 1
|   `-IntegerLiteral 0x7ffa3b028380 <col:14> 'int' 0
|-VarDecl 0x7ffa3b0283e0 <line:29:1, col:20> col:5 c2 'int' cinit
| `-BinaryOperator 0x7ffa3b0284e8 <col:10, col:20> 'int' '/'
|   |-IntegerLiteral 0x7ffa3b028440 <col:10> 'int' 1
|   `-ParenExpr 0x7ffa3b0284c8 <col:14, col:20> 'int'
|     `-BinaryOperator 0x7ffa3b0284a0 <col:15, col:19> 'int' '%'
|       |-IntegerLiteral 0x7ffa3b028460 <col:15> 'int' 1
|       `-IntegerLiteral 0x7ffa3b028480 <col:19> 'int' 0
|-VarDecl 0x7ffa3b028528 <line:30:1, col:22> col:5 c3 'int' cinit
| `-ConditionalOperator 0x7ffa3b028630 <col:10, col:22> 'int'
|   |-IntegerLiteral 0x7ffa3b028588 <col:10> 'int' 0
|   |-IntegerLiteral 0x7ffa3b0285a8 <col:14> 'int' 2
|   `-BinaryOperator 0x7ffa3b028608 <col:18, col:22> 'int' '%'
|     |-IntegerLiteral 0x7ffa3b0285c8 <col:18> 'int' 1
|     `-IntegerLiteral 0x7ffa3b0285e8 <col:22> 'int' 0
|-VarDecl 0x7ffa3b028678 <line:31:1, col:19> col:5 c4 'int' cinit
| `-BinaryOperator 0x7ffa3b028760 <col:10, col:19> 'int' '||'
|   |-IntegerLiteral 0x7ffa3b0286d8 <col:10> 'int' 0
|   `-BinaryOperator 0x7ffa3b028738 <col:15, col:19> 'int' '%'
|     |-IntegerLiteral 0x7ffa3b0286f8 <col:15> 'int' 1
|     `-IntegerLiteral 0x7ffa3b028718 <col:19> 'int' 0
|-VarDecl 0x7ffa3b0287a0 <line:32:1, col:20> col:5 c5 'int' cinit
| `-BinaryOperator 0x7ffa3b0288a8 <col:10, col:20> 'int' '*'
|   |-IntegerLiteral 0x7ffa3b028800 <col:10> 'int' 0
|   `-ParenExpr 0x7ffa3b028888 <col:14, col:20> 'int'
|     `-BinaryOperator 0x7ffa3b028860 <col:15, col:19> 'int' '%'
|       |-IntegerLiteral 0x7ffa3b028820 <col:15> 'int' 1
|       `-IntegerLiteral 0x7ffa3b028840 <col:19> 'int' 0
|-VarDecl 0x7ffa3b0288e8 <line:33:1, col:20> col:5 c6 'int' cinit
| `-BinaryOperator 0x7ffa3b026a00 <col:10, col:20> 'int' '*'
|   |-IntegerLiteral 0x7ffa3b028948 <col:10> 'int' 1
|   `-ParenExpr 0x7ffa3b0289d0 <col:14, col:20> 'int'
|     `-BinaryOperator 0x7ffa3b0289a8 <col:15, col:19> 'int' '%'
|       |-IntegerLiteral 0x7ffa3b028968 <col:15> 'int' 1
|       `-IntegerLiteral 0x7ffa3b028988 <col:19> 'int' 0
|-VarDecl 0x7ffa3b026a40 <line:34:1, col:20> col:5 c7 'int' cinit
| `-BinaryOperator 0x7ffa3b026b48 <col:10, col:20> 'int' '*'
|   |-ParenExpr 0x7ffa3b026b08 <col:10, col:16> 'int'
|   | `-BinaryOperator 0x7ffa3b026ae0 <col:11, col:15> 'int' '%'
|   |   |-IntegerLiteral 0x7ffa3b026aa0 <col:11> 'int' 1
|   |   `-IntegerLiteral 0x7ffa3b026ac0 <col:15> 'int' 0
|   `-IntegerLiteral 0x7ffa3b026b28 <col:20> 'int' 0
|-VarDecl 0x7ffa3b026b88 <line:35:1, col:20> col:5 c8 'int' cinit
| `-BinaryOperator 0x7ffa3b026c90 <col:10, col:20> 'int' '*'
|   |-ParenExpr 0x7ffa3b026c50 <col:10, col:16> 'int'
|   | `-BinaryOperator 0x7ffa3b026c28 <col:11, col:15> 'int' '%'
|   |   |-IntegerLiteral 0x7ffa3b026be8 <col:11> 'int' 1
|   |   `-IntegerLiteral 0x7ffa3b026c08 <col:15> 'int' 0
|   `-IntegerLiteral 0x7ffa3b026c70 <col:20> 'int' 1
|-VarDecl 0x7ffa3b026cd0 <line:36:1, col:19> col:5 c9 'int' cinit
| `-BinaryOperator 0x7ffa3b026db8 <col:10, col:19> 'int' '&&'
|   |-IntegerLiteral 0x7ffa3b026d30 <col:10> 'int' 1
|   `-BinaryOperator 0x7ffa3b026d90 <col:15, col:19> 'int' '%'
|     |-IntegerLiteral 0x7ffa3b026d50 <col:15> 'int' 1
|     `-IntegerLiteral 0x7ffa3b026d70 <col:19> 'int' 0
`-VarDecl 0x7ffa3b026df8 <line:37:1, col:18> col:5 c10 'int' cinit
  `-UnaryOperator 0x7ffa3b026ee0 <col:11, col:18> 'int' prefix '!'
    `-ParenExpr 0x7ffa3b026ec0 <col:12, col:18> 'int'
      `-BinaryOperator 0x7ffa3b026e98 <col:13, col:17> 'int' '%'
        |-IntegerLiteral 0x7ffa3b026e58 <col:13> 'int' 1
        `-IntegerLiteral 0x7ffa3b026e78 <col:17> 'int' 0
