c_files/pr68533.c:9:10: warning: declaration of 'struct S' will not be visible outside of this function [-Wvisibility]
  struct S *    /* { dg-warning "declared inside parameter list will not be visible outside of this definition or declaration" } */
         ^
c_files/pr68533.c:30:4: error: 'void' as parameter must not have type qualifiers
f3 (
   ^
c_files/pr68533.c:37:4: error: 'void' must be the first and only parameter if specified
f4 (
   ^
c_files/pr68533.c:38:8: error: parameter name omitted
   void,        /* { dg-error "'void' must be the only parameter" } */
       ^
c_files/pr68533.c:47:5: error: expected ')'
   x;           /* { dg-error "parameter 'x' has just a forward declaration" } */
    ^
c_files/pr68533.c:45:4: note: to match this '('
f5 (
   ^
c_files/pr68533.c:48:9: error: expected ';' after top level declarator
   int y
        ^
        ;
1 warning and 5 errors generated.
TranslationUnitDecl 0x7fe6920264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fe692026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fe692026740 '__int128'
|-TypedefDecl 0x7fe692026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fe692026760 'unsigned __int128'
|-TypedefDecl 0x7fe692026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fe692026b70 'struct __NSConstantString_tag'
|   `-Record 0x7fe692026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fe692026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fe692026dc0 'char *'
|   `-BuiltinType 0x7fe692026560 'char'
|-TypedefDecl 0x7fe6920270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fe692027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fe692026ee0 'struct __va_list_tag'
|     `-Record 0x7fe692026e58 '__va_list_tag'
|-RecordDecl 0x7fe692027120 <c_files/pr68533.c:5:1, col:19> col:8 struct T definition
| `-FieldDecl 0x7fe692825a00 <col:12, col:16> col:16 referenced t 'int'
|-FunctionDecl 0x7fe692826170 <line:7:1, line:16:1> line:8:1 f1 'void (struct S *, struct T *)'
| |-ParmVarDecl 0x7fe692825f50 <line:9:3, line:10:3> col:3 x 'struct S *'
| |-ParmVarDecl 0x7fe692826060 <line:11:3, line:12:3> col:3 used y 'struct T *'
| `-CompoundStmt 0x7fe692826318 <line:14:1, line:16:1>
|   `-BinaryOperator 0x7fe6928262f0 <line:15:3, col:10> 'int' '='
|     |-MemberExpr 0x7fe692826298 <col:3, col:6> 'int' lvalue ->t 0x7fe692825a00
|     | `-ImplicitCastExpr 0x7fe692826280 <col:3> 'struct T *' <LValueToRValue>
|     |   `-DeclRefExpr 0x7fe692826258 <col:3> 'struct T *' lvalue ParmVar 0x7fe692826060 'y' 'struct T *'
|     `-IntegerLiteral 0x7fe6928262d0 <col:10> 'int' 4
|-RecordDecl 0x7fe692826338 <line:20:3, col:17> col:3 struct definition
| `-FieldDecl 0x7fe6928263f8 <col:11, col:15> col:15 s 'int'
|-FunctionDecl 0x7fe692826690 <line:18:1, line:27:1> line:19:1 f2 'void (struct (anonymous struct at c_files/pr68533.c:20:3) *, struct T *)'
| |-ParmVarDecl 0x7fe6928264f0 <line:20:3, line:21:3> col:3 x 'struct (anonymous struct at c_files/pr68533.c:20:3) *'
| |-ParmVarDecl 0x7fe692826578 <line:22:3, line:23:3> col:3 used y 'struct T *'
| `-CompoundStmt 0x7fe6928267f8 <line:25:1, line:27:1>
|   `-BinaryOperator 0x7fe6928267d0 <line:26:3, col:10> 'int' '='
|     |-MemberExpr 0x7fe692826778 <col:3, col:6> 'int' lvalue ->t 0x7fe692825a00
|     | `-ImplicitCastExpr 0x7fe692826760 <col:3> 'struct T *' <LValueToRValue>
|     |   `-DeclRefExpr 0x7fe692826738 <col:3> 'struct T *' lvalue ParmVar 0x7fe692826578 'y' 'struct T *'
|     `-IntegerLiteral 0x7fe6928267b0 <col:10> 'int' 5
|-FunctionDecl 0x7fe6928268e0 <line:29:1, line:34:1> line:30:1 f3 'void (void)'
| `-CompoundStmt 0x7fe692826978 <line:33:1, line:34:1>
|-FunctionDecl 0x7fe692826a60 <line:36:1, line:42:1> line:37:1 f4 'void (int, ...)'
| |-ParmVarDecl 0x7fe6928269a0 <line:38:4> col:8 'int'
| `-CompoundStmt 0x7fe692826b00 <line:41:1, line:42:1>
|-FunctionDecl 0x7fe692826bf0 <line:44:1, line:45:1> col:1 f5 'void (int)'
| `-ParmVarDecl 0x7fe692826b30 <line:46:4, line:47:4> col:4 x 'int'
`-VarDecl 0x7fe692826ca8 <line:48:4, col:8> col:8 y 'int'
