c_files/pr79677.c:13:21: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
  __builtin_printf (s); /* { dg-error "format not a string literal and no format arguments" } */
                    ^
c_files/pr79677.c:13:21: note: treat the string as an argument to avoid this
  __builtin_printf (s); /* { dg-error "format not a string literal and no format arguments" } */
                    ^
                    "%s", 
1 warning generated.
TranslationUnitDecl 0x7f94760264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7f9476026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7f9476026740 '__int128'
|-TypedefDecl 0x7f9476026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7f9476026760 'unsigned __int128'
|-TypedefDecl 0x7f9476026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7f9476026b70 'struct __NSConstantString_tag'
|   `-Record 0x7f9476026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7f9476026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7f9476026dc0 'char *'
|   `-BuiltinType 0x7f9476026560 'char'
|-TypedefDecl 0x7f94760270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7f9476027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7f9476026ee0 'struct __va_list_tag'
|     `-Record 0x7f9476026e58 '__va_list_tag'
|-FunctionDecl 0x7f947601b000 <c_files/pr79677.c:6:1, col:17> col:6 used foo 'void (char *)'
| `-ParmVarDecl 0x7f9476027138 <col:11, col:16> col:17 'char *'
|-FunctionDecl 0x7f947601b130 <line:8:1, line:15:1> line:9:1 main 'int ()'
| `-CompoundStmt 0x7f947601b6c8 <line:10:1, line:15:1>
|   |-DeclStmt 0x7f947601b318 <line:11:3, col:20>
|   | `-VarDecl 0x7f947601b248 <col:3, col:16> col:8 used s 'char [10]' cinit
|   |   `-StringLiteral 0x7f947601b2e8 <col:16> 'char [10]' lvalue "%s"
|   |-CallExpr 0x7f947601b3c0 <line:12:3, col:9> 'void'
|   | |-ImplicitCastExpr 0x7f947601b3a8 <col:3> 'void (*)(char *)' <FunctionToPointerDecay>
|   | | `-DeclRefExpr 0x7f947601b330 <col:3> 'void (char *)' Function 0x7f947601b000 'foo' 'void (char *)'
|   | `-ImplicitCastExpr 0x7f947601b3f0 <col:8> 'char *' <ArrayToPointerDecay>
|   |   `-DeclRefExpr 0x7f947601b358 <col:8> 'char [10]' lvalue Var 0x7f947601b248 's' 'char [10]'
|   |-CallExpr 0x7f947601b630 <line:13:3, col:22> 'int'
|   | |-ImplicitCastExpr 0x7f947601b618 <col:3> 'int (*)(const char *, ...)' <BuiltinFnToFnPtr>
|   | | `-DeclRefExpr 0x7f947601b5a0 <col:3> '<builtin fn type>' Function 0x7f947601b448 '__builtin_printf' 'int (const char *, ...)'
|   | `-ImplicitCastExpr 0x7f947601b678 <col:21> 'const char *' <BitCast>
|   |   `-ImplicitCastExpr 0x7f947601b660 <col:21> 'char *' <ArrayToPointerDecay>
|   |     `-DeclRefExpr 0x7f947601b5c8 <col:21> 'char [10]' lvalue Var 0x7f947601b248 's' 'char [10]'
|   `-ReturnStmt 0x7f947601b6b0 <line:14:3, col:10>
|     `-IntegerLiteral 0x7f947601b690 <col:10> 'int' 0
`-FunctionDecl 0x7f947601b448 <line:13:3> col:3 implicit used __builtin_printf 'int (const char *, ...)' extern
  |-ParmVarDecl 0x7f947601b4e0 <<invalid sloc>> <invalid sloc> 'const char *'
  `-FormatAttr 0x7f947601b548 <col:3> Implicit printf 1 2
