c_files/pragma-float-const-decimal64-6.c:14:14: warning: unknown pragma in STDC namespace [-Wunknown-pragmas]
#pragma STDC FLOAT_CONST_DECIMAL64 ON           /* { dg-error "ISO C" } */
             ^
c_files/pragma-float-const-decimal64-6.c:20:14: warning: unknown pragma in STDC namespace [-Wunknown-pragmas]
#pragma STDC FLOAT_CONST_DECIMAL64 OFF          /* { dg-error "ISO C" } */
             ^
c_files/pragma-float-const-decimal64-6.c:26:14: warning: unknown pragma in STDC namespace [-Wunknown-pragmas]
#pragma STDC FLOAT_CONST_DECIMAL64 DEFAULT      /* { dg-error "ISO C" } */
             ^
c_files/pragma-float-const-decimal64-6.c:32:3: warning: unknown pragma in STDC namespace [-Wunknown-pragmas]
  _Pragma ("STDC FLOAT_CONST_DECIMAL64 ON")     /* { dg-error "ISO C" } */
  ^
<scratch space>:2:7: note: expanded from here
 STDC FLOAT_CONST_DECIMAL64 ON
      ^
c_files/pragma-float-const-decimal64-6.c:38:3: warning: unknown pragma in STDC namespace [-Wunknown-pragmas]
  _Pragma ("STDC FLOAT_CONST_DECIMAL64 OFF")    /* { dg-error "ISO C" } */
  ^
<scratch space>:4:7: note: expanded from here
 STDC FLOAT_CONST_DECIMAL64 OFF
      ^
c_files/pragma-float-const-decimal64-6.c:44:3: warning: unknown pragma in STDC namespace [-Wunknown-pragmas]
  _Pragma ("STDC FLOAT_CONST_DECIMAL64 DEFAULT") /* { dg-error "ISO C" } */
  ^
<scratch space>:6:7: note: expanded from here
 STDC FLOAT_CONST_DECIMAL64 DEFAULT
      ^
6 warnings generated.
TranslationUnitDecl 0x7ff76d0264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7ff76d026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7ff76d026740 '__int128'
|-TypedefDecl 0x7ff76d026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7ff76d026760 'unsigned __int128'
|-TypedefDecl 0x7ff76d026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7ff76d026b70 'struct __NSConstantString_tag'
|   `-Record 0x7ff76d026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7ff76d026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7ff76d026dc0 'char *'
|   `-BuiltinType 0x7ff76d026560 'char'
|-TypedefDecl 0x7ff76d0270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7ff76d027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7ff76d026ee0 'struct __va_list_tag'
|     `-Record 0x7ff76d026e58 '__va_list_tag'
|-VarDecl 0x7ff76d027138 <c_files/pragma-float-const-decimal64-6.c:10:1, col:8> col:8 used a 'double'
|-FunctionDecl 0x7ff76e025ab0 <line:12:1, line:16:1> line:12:6 f1 'void (void)'
| `-CompoundStmt 0x7ff76e025bb8 <line:13:1, line:16:1>
|   `-BinaryOperator 0x7ff76e025b90 <line:15:3, col:7> 'double' '='
|     |-DeclRefExpr 0x7ff76e025b48 <col:3> 'double' lvalue Var 0x7ff76d027138 'a' 'double'
|     `-FloatingLiteral 0x7ff76e025b70 <col:7> 'double' 1.000000e+00
|-FunctionDecl 0x7ff76e025c68 <line:18:1, line:22:1> line:18:6 f2 'void (void)'
| `-CompoundStmt 0x7ff76e025d70 <line:19:1, line:22:1>
|   `-BinaryOperator 0x7ff76e025d48 <line:21:3, col:7> 'double' '='
|     |-DeclRefExpr 0x7ff76e025d00 <col:3> 'double' lvalue Var 0x7ff76d027138 'a' 'double'
|     `-FloatingLiteral 0x7ff76e025d28 <col:7> 'double' 2.000000e+00
|-FunctionDecl 0x7ff76e025e20 <line:24:1, line:28:1> line:24:6 f3 'void (void)'
| `-CompoundStmt 0x7ff76e025f28 <line:25:1, line:28:1>
|   `-BinaryOperator 0x7ff76e025f00 <line:27:3, col:7> 'double' '='
|     |-DeclRefExpr 0x7ff76e025eb8 <col:3> 'double' lvalue Var 0x7ff76d027138 'a' 'double'
|     `-FloatingLiteral 0x7ff76e025ee0 <col:7> 'double' 3.000000e+00
|-FunctionDecl 0x7ff76e025fd8 <line:30:1, line:34:1> line:30:6 f4 'void (void)'
| `-CompoundStmt 0x7ff76e0260e0 <line:31:1, line:34:1>
|   `-BinaryOperator 0x7ff76e0260b8 <line:33:3, col:7> 'double' '='
|     |-DeclRefExpr 0x7ff76e026070 <col:3> 'double' lvalue Var 0x7ff76d027138 'a' 'double'
|     `-FloatingLiteral 0x7ff76e026098 <col:7> 'double' 1.000000e+00
|-FunctionDecl 0x7ff76e026190 <line:36:1, line:40:1> line:36:6 f5 'void (void)'
| `-CompoundStmt 0x7ff76e026298 <line:37:1, line:40:1>
|   `-BinaryOperator 0x7ff76e026270 <line:39:3, col:7> 'double' '='
|     |-DeclRefExpr 0x7ff76e026228 <col:3> 'double' lvalue Var 0x7ff76d027138 'a' 'double'
|     `-FloatingLiteral 0x7ff76e026250 <col:7> 'double' 2.000000e+00
`-FunctionDecl 0x7ff76e026348 <line:42:1, line:46:1> line:42:6 f6 'void (void)'
  `-CompoundStmt 0x7ff76e026450 <line:43:1, line:46:1>
    `-BinaryOperator 0x7ff76e026428 <line:45:3, col:7> 'double' '='
      |-DeclRefExpr 0x7ff76e0263e0 <col:3> 'double' lvalue Var 0x7ff76d027138 'a' 'double'
      `-FloatingLiteral 0x7ff76e026408 <col:7> 'double' 3.000000e+00
