c_files/pragma-re-2.c:2:59: warning: expected identifier in '#pragma redefine_extname' - ignored [-Wignored-pragmas]
#pragma redefine_extname                /* { dg-warning "malformed" } */
                                                                        ^
c_files/pragma-re-2.c:3:63: warning: expected identifier in '#pragma redefine_extname' - ignored [-Wignored-pragmas]
#pragma redefine_extname foo            /* { dg-warning "malformed" } */
                                                                        ^
c_files/pragma-re-2.c:4:30: warning: expected identifier in '#pragma redefine_extname' - ignored [-Wignored-pragmas]
#pragma redefine_extname foo 1          /* { dg-warning "malformed" } */
                             ^
c_files/pragma-re-2.c:5:34: warning: extra tokens at end of '#pragma redefine_extname' - ignored [-Wignored-pragmas]
#pragma redefine_extname foo bar 2      /* { dg-warning "junk" } */
                                 ^
4 warnings generated.
TranslationUnitDecl 0x7f9e248264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7f9e24826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7f9e24826740 '__int128'
|-TypedefDecl 0x7f9e24826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7f9e24826760 'unsigned __int128'
|-TypedefDecl 0x7f9e24826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7f9e24826b70 'struct __NSConstantString_tag'
|   `-Record 0x7f9e24826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7f9e24826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7f9e24826dc0 'char *'
|   `-BuiltinType 0x7f9e24826560 'char'
|-TypedefDecl 0x7f9e248270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7f9e24827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7f9e24826ee0 'struct __va_list_tag'
|     `-Record 0x7f9e24826e58 '__va_list_tag'
`-VarDecl 0x7f9e24827138 <c_files/pragma-re-2.c:7:1, col:5> col:5 bar 'int'
