c_files/pr17188-1.c:10:8: error: redefinition of 's0'
struct s0 { }; /* { dg-error "redefinition of 'struct s0'" } */
       ^
c_files/pr17188-1.c:8:8: note: previous definition is here
struct s0 { }; /* { dg-message "note: originally defined here" } */
       ^
c_files/pr17188-1.c:13:8: error: redefinition of 's1'
struct s1 { }; /* { dg-error "redefinition of 'struct s1'" } */
       ^
c_files/pr17188-1.c:12:8: note: previous definition is here
struct s1 { }; /* { dg-message "note: originally defined here" } */
       ^
c_files/pr17188-1.c:16:8: error: redefinition of 's2'
struct s2 { int a : 1; }; /* { dg-error "redefinition of 'struct s2'" } */
       ^
c_files/pr17188-1.c:15:8: note: previous definition is here
struct s2 { int a : 1; }; /* { dg-message "note: originally defined here" } */
       ^
c_files/pr17188-1.c:19:8: error: redefinition of 's3'
struct s3 { int a : 1; }; /* { dg-error "redefinition of 'struct s3'" } */
       ^
c_files/pr17188-1.c:18:8: note: previous definition is here
struct s3 { }; /* { dg-message "note: originally defined here" } */
       ^
c_files/pr17188-1.c:22:8: error: redefinition of 's4'
struct s4 { }; /* { dg-error "redefinition of 'struct s4'" } */
       ^
c_files/pr17188-1.c:21:8: note: previous definition is here
struct s4 { int a : 1; }; /* { dg-message "note: originally defined here" } */
       ^
5 errors generated.
TranslationUnitDecl 0x7fbc7b8264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fbc7b826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fbc7b826740 '__int128'
|-TypedefDecl 0x7fbc7b826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fbc7b826760 'unsigned __int128'
|-TypedefDecl 0x7fbc7b826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fbc7b826b70 'struct __NSConstantString_tag'
|   `-Record 0x7fbc7b826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fbc7b826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fbc7b826dc0 'char *'
|   `-BuiltinType 0x7fbc7b826560 'char'
|-TypedefDecl 0x7fbc7b8270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fbc7b827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fbc7b826ee0 'struct __va_list_tag'
|     `-Record 0x7fbc7b826e58 '__va_list_tag'
|-RecordDecl 0x7fbc7b827120 <c_files/pr17188-1.c:8:1, col:13> col:8 struct s0 definition
|-RecordDecl 0x7fbc7c025a00 prev 0x7fbc7b827120 <line:9:1, col:8> col:8 struct s0
|-RecordDecl 0x7fbc7c025a88 <line:10:1, col:13> col:8 invalid struct definition
|-RecordDecl 0x7fbc7c025b30 <line:12:1, col:13> col:8 struct s1 definition
|-RecordDecl 0x7fbc7c025be0 <line:13:1, col:13> col:8 invalid struct definition
|-RecordDecl 0x7fbc7c025c90 <line:15:1, col:24> col:8 struct s2 definition
| `-FieldDecl 0x7fbc7c025d78 <col:13, col:21> col:17 a 'int'
|   `-IntegerLiteral 0x7fbc7c025d40 <col:21> 'int' 1
|-RecordDecl 0x7fbc7c025dc0 <line:16:1, col:24> col:8 invalid struct definition
| `-FieldDecl 0x7fbc7c025ea8 <col:13, col:21> col:17 a 'int'
|   `-IntegerLiteral 0x7fbc7c025e70 <col:21> 'int' 1
|-RecordDecl 0x7fbc7c025ef0 <line:18:1, col:13> col:8 struct s3 definition
|-RecordDecl 0x7fbc7c025fa0 <line:19:1, col:24> col:8 invalid struct definition
| `-FieldDecl 0x7fbc7c026088 <col:13, col:21> col:17 a 'int'
|   `-IntegerLiteral 0x7fbc7c026050 <col:21> 'int' 1
|-RecordDecl 0x7fbc7c0260d0 <line:21:1, col:24> col:8 struct s4 definition
| `-FieldDecl 0x7fbc7c0261b8 <col:13, col:21> col:17 a 'int'
|   `-IntegerLiteral 0x7fbc7c026180 <col:21> 'int' 1
|-RecordDecl 0x7fbc7c026200 <line:22:1, col:13> col:8 invalid struct definition
|-RecordDecl 0x7fbc7c0262b0 <line:24:1, col:24> col:8 struct s5 definition
| `-FieldDecl 0x7fbc7c026398 <col:13, col:21> col:17 a 'int'
|   `-IntegerLiteral 0x7fbc7c026360 <col:21> 'int' 1
|-RecordDecl 0x7fbc7c0263e0 prev 0x7fbc7c0262b0 <line:25:1, col:8> col:8 struct s5
|-RecordDecl 0x7fbc7c026468 <line:27:1, col:8> col:8 struct s6
|-RecordDecl 0x7fbc7c026510 prev 0x7fbc7c026468 <line:28:1, col:24> col:8 struct s6 definition
| `-FieldDecl 0x7fbc7c0265d0 <col:13, col:21> col:17 a 'int'
|   `-IntegerLiteral 0x7fbc7c026598 <col:21> 'int' 1
|-RecordDecl 0x7fbc7c026618 <line:30:1, col:8> col:8 struct s7
`-RecordDecl 0x7fbc7c0266c0 prev 0x7fbc7c026618 <line:31:1, col:13> col:8 struct s7 definition
