c_files/pr23576.c:1:27: error: array has incomplete element type 'struct ipr_path_entry'
struct ipr_path_entry path[0]; /* { dg-error "array type has incomplete element type" } */
                          ^
c_files/pr23576.c:1:8: note: forward declaration of 'struct ipr_path_entry'
struct ipr_path_entry path[0]; /* { dg-error "array type has incomplete element type" } */
       ^
1 error generated.
TranslationUnitDecl 0x7ff72f0264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7ff72f026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7ff72f026740 '__int128'
|-TypedefDecl 0x7ff72f026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7ff72f026760 'unsigned __int128'
|-TypedefDecl 0x7ff72f026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7ff72f026b70 'struct __NSConstantString_tag'
|   `-Record 0x7ff72f026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7ff72f026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7ff72f026dc0 'char *'
|   `-BuiltinType 0x7ff72f026560 'char'
|-TypedefDecl 0x7ff72f0270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7ff72f027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7ff72f026ee0 'struct __va_list_tag'
|     `-Record 0x7ff72f026e58 '__va_list_tag'
|-RecordDecl 0x7ff72f027120 <c_files/pr23576.c:1:1, col:8> col:8 struct ipr_path_entry
`-VarDecl 0x7ff72f0777f8 <col:1, col:23> col:23 invalid path 'int'
