c_files/pr30552-3.c:12:10: error: expected ';' at end of declaration
  int g()
         ^
         ;
c_files/pr30552-3.c:13:19: error: expected ';' at end of declaration
    int a[( {int b} )]; /* { dg-error "braced-group within expression allowed only inside a function|declaration for parameter" } */
                  ^
                  ;
c_files/pr30552-3.c:13:11: error: size of array has non-integer type 'void'
    int a[( {int b} )]; /* { dg-error "braced-group within expression allowed only inside a function|declaration for parameter" } */
          ^~~~~~~~~~~
3 errors generated.
TranslationUnitDecl 0x7f96af0264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7f96af026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7f96af026740 '__int128'
|-TypedefDecl 0x7f96af026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7f96af026760 'unsigned __int128'
|-TypedefDecl 0x7f96af026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7f96af026b70 'struct __NSConstantString_tag'
|   `-Record 0x7f96af026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7f96af026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7f96af026dc0 'char *'
|   `-BuiltinType 0x7f96af026560 'char'
|-TypedefDecl 0x7f96af0270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7f96af027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7f96af026ee0 'struct __va_list_tag'
|     `-Record 0x7f96af026e58 '__va_list_tag'
`-FunctionDecl 0x7f96af077400 <c_files/pr30552-3.c:10:1, line:15:1> line:10:5 main 'int ()'
  `-CompoundStmt 0x7f96af077b08 <line:11:1, line:15:1>
    |-DeclStmt 0x7f96af0775a8 <line:12:3, line:13:5>
    | `-FunctionDecl 0x7f96af077500 parent 0x7f96af0264d0 <line:12:3, col:9> col:7 g 'int ()'
    |-DeclStmt 0x7f96af077ab8 <line:13:5, col:23>
    | `-VarDecl 0x7f96af077a58 <col:5, col:9> col:9 invalid a 'int'
    `-ReturnStmt 0x7f96af077af0 <line:14:3, col:10>
      `-IntegerLiteral 0x7f96af077ad0 <col:10> 'int' 0
