c_files/pr35441.c:8:10: error: called object type 'long' is not a function or function pointer
  (p - q)();                    /* { dg-error "is not a function" } */
  ~~~~~~~^
c_files/pr35441.c:13:8: error: called object type 'double' is not a function or function pointer
  (x/y)();                      /* { dg-error "is not a function" } */
  ~~~~~^
c_files/pr35441.c:18:27: error: called object type 'unsigned int' is not a function or function pointer
  (i << j | i >> (32 - j))();   /* { dg-error "is not a function" } */
  ~~~~~~~~~~~~~~~~~~~~~~~~^
c_files/pr35441.c:19:27: error: called object type 'unsigned int' is not a function or function pointer
  (i >> j | i << (32 - j))();   /* { dg-error "is not a function" } */
  ~~~~~~~~~~~~~~~~~~~~~~~~^
c_files/pr35441.c:24:18: error: called object type 'char *' is not a function or function pointer
  (p < q ? p : q)();            /* { dg-error "is not a function" } */
  ~~~~~~~~~~~~~~~^
c_files/pr35441.c:25:18: error: called object type 'char *' is not a function or function pointer
  (p > q ? p : q)();            /* { dg-error "is not a function" } */
  ~~~~~~~~~~~~~~~^
6 errors generated.
TranslationUnitDecl 0x7fc0c18264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fc0c1826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fc0c1826740 '__int128'
|-TypedefDecl 0x7fc0c1826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fc0c1826760 'unsigned __int128'
|-TypedefDecl 0x7fc0c1826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fc0c1826b70 'struct __NSConstantString_tag'
|   `-Record 0x7fc0c1826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fc0c1826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fc0c1826dc0 'char *'
|   `-BuiltinType 0x7fc0c1826560 'char'
|-TypedefDecl 0x7fc0c18270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fc0c1827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fc0c1826ee0 'struct __va_list_tag'
|     `-Record 0x7fc0c1826e58 '__va_list_tag'
|-FunctionDecl 0x7fc0c2025ad0 <c_files/pr35441.c:6:1, line:9:1> line:6:6 foo1 'void (char **, char **)'
| |-ParmVarDecl 0x7fc0c1827160 <col:11, col:18> col:18 used p 'char **'
| |-ParmVarDecl 0x7fc0c2025a00 <col:21, col:28> col:28 used q 'char **'
| `-CompoundStmt 0x7fc0c2026060 <line:7:1, line:9:1>
|-FunctionDecl 0x7fc0c20261e0 <line:11:1, line:14:1> line:11:6 foo2 'void (double, double)'
| |-ParmVarDecl 0x7fc0c2026090 <col:11, col:18> col:18 used x 'double'
| |-ParmVarDecl 0x7fc0c2026108 <col:21, col:28> col:28 used y 'double'
| `-CompoundStmt 0x7fc0c2026378 <line:12:1, line:14:1>
|-FunctionDecl 0x7fc0c20264f0 <line:16:1, line:20:1> line:16:6 foo3 'void (unsigned int, int)'
| |-ParmVarDecl 0x7fc0c20263a8 <col:11, col:20> col:20 used i 'unsigned int'
| |-ParmVarDecl 0x7fc0c2026420 <col:23, col:27> col:27 used j 'int'
| `-CompoundStmt 0x7fc0c20269e8 <line:17:1, line:20:1>
`-FunctionDecl 0x7fc0c2026b60 <line:22:1, line:26:1> line:22:6 foo4 'void (char *, char *)'
  |-ParmVarDecl 0x7fc0c2026a18 <col:11, col:17> col:17 used p 'char *'
  |-ParmVarDecl 0x7fc0c2026a90 <col:20, col:26> col:26 used q 'char *'
  `-CompoundStmt 0x7fc0c2026f48 <line:23:1, line:26:1>
