c_files/pr35444-2.c:3:31: error: expected ')'
void foo(int n, int a[n], ... 0); /* { dg-error "expected" } */
                              ^
c_files/pr35444-2.c:3:9: note: to match this '('
void foo(int n, int a[n], ... 0); /* { dg-error "expected" } */
        ^
1 error generated.
TranslationUnitDecl 0x7f832c0264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7f832c026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7f832c026740 '__int128'
|-TypedefDecl 0x7f832c026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7f832c026760 'unsigned __int128'
|-TypedefDecl 0x7f832c026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7f832c026b70 'struct __NSConstantString_tag'
|   `-Record 0x7f832c026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7f832c026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7f832c026dc0 'char *'
|   `-BuiltinType 0x7f832c026560 'char'
|-TypedefDecl 0x7f832c0270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7f832c027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7f832c026ee0 'struct __va_list_tag'
|     `-Record 0x7f832c026e58 '__va_list_tag'
|-FunctionDecl 0x7f832c0775d0 <c_files/pr35444-2.c:3:1, col:32> col:6 foo 'void (int, int *, ...)'
| |-ParmVarDecl 0x7f832c027138 <col:10, col:14> col:14 used n 'int'
| `-ParmVarDecl 0x7f832c0774c0 <col:17, col:24> col:21 a 'int *':'int *'
`-FunctionDecl 0x7f832c077708 <line:4:1, col:13> col:6 bar 'void ()'
  `-CompoundStmt 0x7f832c0777a0 <col:12, col:13>
