c_files/pr37772.c:9:8: error: expected string literal in 'asm'
  asm ();                 /* { dg-error "expected string literal before" } */
       ^
c_files/pr37772.c:10:8: error: expected string literal in 'asm'
  asm (1);                /* { dg-error "expected string literal before" } */
       ^
c_files/pr37772.c:11:8: error: expected string literal in 'asm'
  asm (int);              /* { dg-error "expected string literal before" } */
       ^
c_files/pr37772.c:12:8: error: expected string literal in 'asm'
  asm (: "=r" (i));       /* { dg-error "expected string literal before" } */
       ^
c_files/pr37772.c:13:8: error: expected string literal in 'asm'
  asm (1 : "=r" (i));     /* { dg-error "expected string literal before" } */
       ^
c_files/pr37772.c:14:8: error: expected string literal in 'asm'
  asm (int : "=r" (i));   /* { dg-error "expected string literal before" } */
       ^
c_files/pr37772.c:15:8: error: expected string literal in 'asm'
  asm (: : "r" (i));      /* { dg-error "expected string literal before" } */
       ^
c_files/pr37772.c:16:8: error: expected string literal in 'asm'
  asm (1 : : "r" (i));    /* { dg-error "expected string literal before" } */
       ^
c_files/pr37772.c:17:8: error: expected string literal in 'asm'
  asm (int : : "r" (i));  /* { dg-error "expected string literal before" } */
       ^
c_files/pr37772.c:18:8: error: expected string literal in 'asm'
  asm (: : : "memory");   /* { dg-error "expected string literal before" } */
       ^
c_files/pr37772.c:19:8: error: expected string literal in 'asm'
  asm (1 : : : "memory"); /* { dg-error "expected string literal before" } */
       ^
11 errors generated.
TranslationUnitDecl 0x7fadc08264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fadc0826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fadc0826740 '__int128'
|-TypedefDecl 0x7fadc0826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fadc0826760 'unsigned __int128'
|-TypedefDecl 0x7fadc0826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fadc0826b70 'struct __NSConstantString_tag'
|   `-Record 0x7fadc0826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fadc0826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fadc0826dc0 'char *'
|   `-BuiltinType 0x7fadc0826560 'char'
|-TypedefDecl 0x7fadc08270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fadc0827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fadc0826ee0 'struct __va_list_tag'
|     `-Record 0x7fadc0826e58 '__va_list_tag'
`-FunctionDecl 0x7fadc0827168 <c_files/pr37772.c:5:1, line:20:1> line:6:1 foo 'void ()'
  `-CompoundStmt 0x7fadc1025ad0 <line:7:1, line:20:1>
    `-DeclStmt 0x7fadc1025ab8 <line:8:3, col:8>
      `-VarDecl 0x7fadc1025a58 <col:3, col:7> col:7 i 'int'
