c_files/pr37561.c:10:4: error: assignment to cast is illegal, lvalue casts are not supported
  ((char *) p)++;       /* { dg-error "lvalue" } */
  ~^~~~~~~~~~~~~
c_files/pr37561.c:12:4: warning: cast to 'char *' from smaller integer type 'char' [-Wint-to-pointer-cast]
  ((char *) q)++;       /* { dg-error "lvalue" } */
   ^
c_files/pr37561.c:12:4: error: assignment to cast is illegal, lvalue casts are not supported
  ((char *) q)++;       /* { dg-error "lvalue" } */
  ~^~~~~~~~~~~~~
c_files/pr37561.c:15:4: error: assignment to cast is illegal, lvalue casts are not supported
  ((char *) p)--;       /* { dg-error "lvalue" } */
  ~^~~~~~~~~~~~~
c_files/pr37561.c:17:4: warning: cast to 'char *' from smaller integer type 'char' [-Wint-to-pointer-cast]
  ((char *) q)--;       /* { dg-error "lvalue" } */
   ^
c_files/pr37561.c:17:4: error: assignment to cast is illegal, lvalue casts are not supported
  ((char *) q)--;       /* { dg-error "lvalue" } */
  ~^~~~~~~~~~~~~
c_files/pr37561.c:20:14: error: expected '{' in compound literal
  ++(char *) p;         /* { dg-error "lvalue" } */
             ^
c_files/pr37561.c:22:14: error: expected '{' in compound literal
  ++(char *) q;         /* { dg-error "lvalue" } */
             ^
c_files/pr37561.c:25:14: error: expected '{' in compound literal
  --(char *) p;         /* { dg-error "lvalue" } */
             ^
c_files/pr37561.c:27:14: error: expected '{' in compound literal
  --(char *) q;         /* { dg-error "lvalue" } */
             ^
2 warnings and 8 errors generated.
TranslationUnitDecl 0x7fbf4f8264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fbf4f826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fbf4f826740 '__int128'
|-TypedefDecl 0x7fbf4f826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fbf4f826760 'unsigned __int128'
|-TypedefDecl 0x7fbf4f826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fbf4f826b70 'struct __NSConstantString_tag'
|   `-Record 0x7fbf4f826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fbf4f826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fbf4f826dc0 'char *'
|   `-BuiltinType 0x7fbf4f826560 'char'
|-TypedefDecl 0x7fbf4f8270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fbf4f827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fbf4f826ee0 'struct __va_list_tag'
|     `-Record 0x7fbf4f826e58 '__va_list_tag'
|-VarDecl 0x7fbf4f827138 <<built-in>:71:25, c_files/pr37561.c:4:31> col:31 used p 'long'
|-VarDecl 0x7fbf50025a00 <line:5:1, col:6> col:6 used q 'char'
`-FunctionDecl 0x7fbf50025aa8 <line:7:1, line:29:1> line:8:1 foo 'void ()'
  `-CompoundStmt 0x7fbf50026178 <line:9:1, line:29:1>
