c_files/pr39323-1.c:4:25: error: requested alignment must be 268435456 bytes or smaller
int foo __attribute__ ((aligned(1 << 29))) =  20; /* { dg-error "requested alignment is too large" } */
                        ^       ~~~~~~~
c_files/pr39323-1.c:5:29: error: requested alignment must be 268435456 bytes or smaller
typedef int __attribute__ ((aligned(1 << 29))) int29; /* { dg-error "requested alignment is too large" } */
                            ^       ~~~~~~~
2 errors generated.
TranslationUnitDecl 0x7fd2810264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fd281026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fd281026740 '__int128'
|-TypedefDecl 0x7fd281026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fd281026760 'unsigned __int128'
|-TypedefDecl 0x7fd281026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fd281026b70 'struct __NSConstantString_tag'
|   `-Record 0x7fd281026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fd281026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fd281026dc0 'char *'
|   `-BuiltinType 0x7fd281026560 'char'
|-TypedefDecl 0x7fd2810270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fd281027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fd281026ee0 'struct __va_list_tag'
|     `-Record 0x7fd281026e58 '__va_list_tag'
|-VarDecl 0x7fd2810271a0 <c_files/pr39323-1.c:4:1, col:47> col:5 foo 'int' cinit
| `-IntegerLiteral 0x7fd281826a40 <col:47> 'int' 20
`-TypedefDecl 0x7fd281826ae0 <line:5:1, col:48> col:48 int29 'int'
  `-BuiltinType 0x7fd2810265c0 'int'
