c_files/pr53532.c:11:8: error: variable has incomplete type 'struct T'
  foo ((struct T) {});  /* { dg-error "invalid use of undefined type" } */
       ^~~~~~~~~~~~~
c_files/pr53532.c:11:16: note: forward declaration of 'struct T'
  foo ((struct T) {});  /* { dg-error "invalid use of undefined type" } */
               ^
1 error generated.
TranslationUnitDecl 0x7fa24b0264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fa24b026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fa24b026740 '__int128'
|-TypedefDecl 0x7fa24b026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fa24b026760 'unsigned __int128'
|-TypedefDecl 0x7fa24b026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fa24b026b70 'struct __NSConstantString_tag'
|   `-Record 0x7fa24b026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fa24b026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fa24b026dc0 'char *'
|   `-BuiltinType 0x7fa24b026560 'char'
|-TypedefDecl 0x7fa24b0270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fa24b027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fa24b026ee0 'struct __va_list_tag'
|     `-Record 0x7fa24b026e58 '__va_list_tag'
|-RecordDecl 0x7fa24b027120 <c_files/pr53532.c:5:1, col:11> col:8 struct S definition
|-FunctionDecl 0x7fa24b825b28 <line:6:1, col:25> col:12 used foo 'int (struct S)' extern
| `-ParmVarDecl 0x7fa24b825a20 <col:17, col:24> col:25 'struct S':'struct S'
`-FunctionDecl 0x7fa24b825c60 <line:8:1, line:13:1> line:9:1 main 'int ()'
  `-CompoundStmt 0x7fa24b826248 <line:10:1, line:13:1>
    `-ReturnStmt 0x7fa24b826230 <line:12:3, col:10>
      `-IntegerLiteral 0x7fa24b826210 <col:10> 'int' 0
