c_files/pr56980.c:18:10: error: passing 'B *' (aka 'struct A *') to parameter of incompatible type 'struct A'; dereference with *
  foo_s (b); /* { dg-error "incompatible" } */
         ^
         *
c_files/pr56980.c:8:21: note: passing argument to parameter here
void foo_s (struct A); /* { dg-message "expected .struct A. but argument is of type 'B \\*' {aka 'struct A \\*'}" } */
                    ^
c_files/pr56980.c:19:10: error: passing 'V *' (aka 'union U *') to parameter of incompatible type 'union U'; dereference with *
  foo_u (v); /* { dg-error "incompatible" } */
         ^
         *
c_files/pr56980.c:9:20: note: passing argument to parameter here
void foo_u (union U); /* { dg-message "expected .union U. but argument is of type 'V \\*' {aka 'union U \\*'}" } */
                   ^
c_files/pr56980.c:20:10: warning: incompatible pointer to integer conversion passing 'F *' (aka 'enum E *') to parameter of type 'enum E'; dereference with * [-Wint-conversion]
  foo_e (f); /* { dg-error "incompatible" } */
         ^
         *
c_files/pr56980.c:10:19: note: passing argument to parameter here
void foo_e (enum E); /* { dg-message "expected .enum E. but argument is of type 'F \\*' {aka 'enum E \\*'}" } */
                  ^
c_files/pr56980.c:21:11: warning: incompatible pointer types passing 'struct B *' to parameter of type 'B *' (aka 'struct A *') [-Wincompatible-pointer-types]
  foo_sp ((struct B *) b); /* { dg-error "passing argument" } */
          ^~~~~~~~~~~~~~
c_files/pr56980.c:11:17: note: passing argument to parameter here
void foo_sp (B *); /* { dg-message "expected 'B \\*' {aka 'struct A \\*'} but argument is of type .struct B \\*." } */
                ^
c_files/pr56980.c:22:11: warning: incompatible pointer types passing 'union V *' to parameter of type 'V *' (aka 'union U *') [-Wincompatible-pointer-types]
  foo_up ((union V *) v); /* { dg-error "passing argument" } */
          ^~~~~~~~~~~~~
c_files/pr56980.c:12:17: note: passing argument to parameter here
void foo_up (V *); /* { dg-message "expected 'V \\*' {aka 'union U \\*'} but argument is of type .union V \\*." } */
                ^
c_files/pr56980.c:23:11: warning: incompatible pointer types passing 'enum F *' to parameter of type 'F *' (aka 'enum E *') [-Wincompatible-pointer-types]
  foo_ep (__extension__ (enum F *) f); /* { dg-error "passing argument" } */
          ^~~~~~~~~~~~~~~~~~~~~~~~~~
c_files/pr56980.c:13:17: note: passing argument to parameter here
void foo_ep (F *); /* { dg-message "expected 'F \\*' {aka 'enum E \\*'} but argument is of type .enum F \\*." } */
                ^
4 warnings and 2 errors generated.
TranslationUnitDecl 0x7fb948824ad0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fb948825020 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fb948824d40 '__int128'
|-TypedefDecl 0x7fb948825090 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fb948824d60 'unsigned __int128'
|-TypedefDecl 0x7fb948825368 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fb948825170 'struct __NSConstantString_tag'
|   `-Record 0x7fb9488250e8 '__NSConstantString_tag'
|-TypedefDecl 0x7fb948825400 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fb9488253c0 'char *'
|   `-BuiltinType 0x7fb948824b60 'char'
|-TypedefDecl 0x7fb9488256c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fb948825670 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fb9488254e0 'struct __va_list_tag'
|     `-Record 0x7fb948825458 '__va_list_tag'
|-RecordDecl 0x7fb948825720 <c_files/pr56980.c:4:9, col:27> col:16 struct A definition
| `-FieldDecl 0x7fb948876000 <col:20, col:24> col:24 i 'int'
|-TypedefDecl 0x7fb9488760a0 <col:1, col:29> col:29 referenced B 'struct A':'struct A'
| `-ElaboratedType 0x7fb948876050 'struct A' sugar
|   `-RecordType 0x7fb9488257b0 'struct A'
|     `-Record 0x7fb948825720 'A'
|-RecordDecl 0x7fb948876110 <line:5:9, col:26> col:15 union U definition
| `-FieldDecl 0x7fb9488761d8 <col:19, col:23> col:23 i 'int'
|-TypedefDecl 0x7fb948876270 <col:1, col:28> col:28 referenced V 'union U':'union U'
| `-ElaboratedType 0x7fb948876220 'union U' sugar
|   `-RecordType 0x7fb9488761a0 'union U'
|     `-Record 0x7fb948876110 'U'
|-EnumDecl 0x7fb9488762e0 <line:6:9, col:20> col:14 E
| `-EnumConstantDecl 0x7fb9488763a0 <col:18> col:18 G 'int'
|-TypedefDecl 0x7fb948876440 <col:1, col:22> col:22 referenced F 'enum E':'enum E'
| `-ElaboratedType 0x7fb9488763f0 'enum E' sugar
|   `-EnumType 0x7fb948876380 'enum E'
|     `-Enum 0x7fb9488762e0 'E'
|-FunctionDecl 0x7fb9488765d0 <line:8:1, col:21> col:6 used foo_s 'void (struct A)'
| `-ParmVarDecl 0x7fb9488764d0 <col:13, col:20> col:21 'struct A':'struct A'
|-FunctionDecl 0x7fb9488767d0 <line:9:1, col:20> col:6 used foo_u 'void (union U)'
| `-ParmVarDecl 0x7fb9488766d0 <col:13, col:19> col:20 'union U':'union U'
|-FunctionDecl 0x7fb948876990 <line:10:1, col:19> col:6 used foo_e 'void (enum E)'
| `-ParmVarDecl 0x7fb948876890 <col:13, col:18> col:19 'enum E':'enum E'
|-FunctionDecl 0x7fb948876bc0 <line:11:1, col:17> col:6 used foo_sp 'void (B *)'
| `-ParmVarDecl 0x7fb948876ab8 <col:14, col:16> col:17 'B *'
|-FunctionDecl 0x7fb948876df0 <line:12:1, col:17> col:6 used foo_up 'void (V *)'
| `-ParmVarDecl 0x7fb948876ce8 <col:14, col:16> col:17 'V *'
|-FunctionDecl 0x7fb948877028 <line:13:1, col:17> col:6 used foo_ep 'void (F *)'
| `-ParmVarDecl 0x7fb948876f18 <col:14, col:16> col:17 'F *'
`-FunctionDecl 0x7fb9488772f0 <line:15:1, line:24:1> line:16:1 bar 'void (B *, V *, F *)'
  |-ParmVarDecl 0x7fb9488770d8 <col:6, col:9> col:9 used b 'B *'
  |-ParmVarDecl 0x7fb948877148 <col:12, col:15> col:15 used v 'V *'
  |-ParmVarDecl 0x7fb9488771b8 <col:18, col:21> col:21 used f 'F *'
  `-CompoundStmt 0x7fb9488792d8 <line:17:1, line:24:1>
    |-CallExpr 0x7fb948877a30 <line:20:3, col:11> 'void'
    | |-ImplicitCastExpr 0x7fb948877a18 <col:3> 'void (*)(enum E)' <FunctionToPointerDecay>
    | | `-DeclRefExpr 0x7fb948877968 <col:3> 'void (enum E)' Function 0x7fb948876990 'foo_e' 'void (enum E)'
    | `-ImplicitCastExpr 0x7fb948877a78 <col:10> 'enum E':'enum E' <PointerToIntegral>
    |   `-ImplicitCastExpr 0x7fb948877a60 <col:10> 'F *' <LValueToRValue>
    |     `-DeclRefExpr 0x7fb948877990 <col:10> 'F *' lvalue ParmVar 0x7fb9488771b8 'f' 'F *'
    |-CallExpr 0x7fb948877cf0 <line:21:3, col:25> 'void'
    | |-ImplicitCastExpr 0x7fb948877cd8 <col:3> 'void (*)(B *)' <FunctionToPointerDecay>
    | | `-DeclRefExpr 0x7fb948877a90 <col:3> 'void (B *)' Function 0x7fb948876bc0 'foo_sp' 'void (B *)'
    | `-ImplicitCastExpr 0x7fb948877d20 <col:11, col:24> 'B *' <BitCast>
    |   `-CStyleCastExpr 0x7fb948877c58 <col:11, col:24> 'struct B *' <BitCast>
    |     `-ImplicitCastExpr 0x7fb948877c40 <col:24> 'B *' <LValueToRValue>
    |       `-DeclRefExpr 0x7fb948877b60 <col:24> 'B *' lvalue ParmVar 0x7fb9488770d8 'b' 'B *'
    |-CallExpr 0x7fb948877fa0 <line:22:3, col:24> 'void'
    | |-ImplicitCastExpr 0x7fb948877f88 <col:3> 'void (*)(V *)' <FunctionToPointerDecay>
    | | `-DeclRefExpr 0x7fb948877d38 <col:3> 'void (V *)' Function 0x7fb948876df0 'foo_up' 'void (V *)'
    | `-ImplicitCastExpr 0x7fb948877fd0 <col:11, col:23> 'V *' <BitCast>
    |   `-CStyleCastExpr 0x7fb948877f08 <col:11, col:23> 'union V *' <BitCast>
    |     `-ImplicitCastExpr 0x7fb948877ef0 <col:23> 'V *' <LValueToRValue>
    |       `-DeclRefExpr 0x7fb948877e10 <col:23> 'V *' lvalue ParmVar 0x7fb948877148 'v' 'V *'
    `-CallExpr 0x7fb948879290 <line:23:3, col:37> 'void'
      |-ImplicitCastExpr 0x7fb948879278 <col:3> 'void (*)(F *)' <FunctionToPointerDecay>
      | `-DeclRefExpr 0x7fb948879000 <col:3> 'void (F *)' Function 0x7fb948877028 'foo_ep' 'void (F *)'
      `-ImplicitCastExpr 0x7fb9488792c0 <col:11, col:36> 'F *' <BitCast>
        `-UnaryOperator 0x7fb948879200 <col:11, col:36> 'enum F *' prefix '__extension__'
          `-CStyleCastExpr 0x7fb9488791d8 <col:25, col:36> 'enum F *' <BitCast>
            `-ImplicitCastExpr 0x7fb9488791c0 <col:36> 'F *' <LValueToRValue>
              `-DeclRefExpr 0x7fb9488790e0 <col:36> 'F *' lvalue ParmVar 0x7fb9488771b8 'f' 'F *'
