c_files/pr61162-2.c:17:10: error: returning 'struct s' from a function with incompatible result type 'int'
  return s; /* { dg-error "10:incompatible types when returning" } */
         ^
c_files/pr61162-2.c:23:3: error: void function 'fn3' should not return a value [-Wreturn-type]
  return 3; /* { dg-warning "10:in function returning void" } */
  ^      ~
c_files/pr61162-2.c:29:10: warning: incompatible pointer to integer conversion returning 'int *' from a function with result type 'int'; dereference with * [-Wint-conversion]
  return a; /* { dg-warning "10:returning 'int \\*' from a function with return type 'int' makes integer from pointer without a cast" } */
         ^
         *
c_files/pr61162-2.c:35:10: warning: incompatible integer to pointer conversion returning 'int' from a function with result type 'int *'; take the address with & [-Wint-conversion]
  return a; /* { dg-warning "10:returning 'int' from a function with return type 'int \\*' makes pointer from integer without a cast" } */
         ^
         &
c_files/pr61162-2.c:41:10: warning: returning 'int *' from a function with result type 'unsigned int *' converts between pointers to integer types with different sign [-Wpointer-sign]
  return i; /* { dg-warning "10:pointer targets in returning 'int \\*' from a function with return type 'unsigned int \\*' differ" } */
         ^
3 warnings and 2 errors generated.
TranslationUnitDecl 0x7fa0340264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fa034026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fa034026740 '__int128'
|-TypedefDecl 0x7fa034026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fa034026760 'unsigned __int128'
|-TypedefDecl 0x7fa034026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fa034026b70 'struct __NSConstantString_tag'
|   `-Record 0x7fa034026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fa034026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fa034026dc0 'char *'
|   `-BuiltinType 0x7fa034026560 'char'
|-TypedefDecl 0x7fa0340270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fa034027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fa034026ee0 'struct __va_list_tag'
|     `-Record 0x7fa034026e58 '__va_list_tag'
|-EnumDecl 0x7fa034027120 <c_files/pr61162-2.c:5:1, col:12> col:6 e
| `-EnumConstantDecl 0x7fa034077a00 <col:10> col:10 A 'int'
|-RecordDecl 0x7fa034077a48 <line:6:1, col:19> col:8 struct s definition
| `-FieldDecl 0x7fa034077b08 <col:12, col:16> col:16 a 'int'
|-FunctionDecl 0x7fa034077c80 <line:8:1, line:12:1> line:9:1 fn1 'enum e (void)'
| `-CompoundStmt 0x7fa034077da8 <line:10:1, line:12:1>
|   `-ReturnStmt 0x7fa034077d90 <line:11:3, col:10>
|     `-ImplicitCastExpr 0x7fa034077d78 <col:10> 'enum e':'enum e' <IntegralCast>
|       `-IntegerLiteral 0x7fa034077d58 <col:10> 'int' 0
|-FunctionDecl 0x7fa034077f28 <line:14:1, line:18:1> line:15:1 fn2 'int (struct s)'
| |-ParmVarDecl 0x7fa034077e20 <col:6, col:15> col:15 used s 'struct s':'struct s'
| `-CompoundStmt 0x7fa0340783c0 <line:16:1, line:18:1>
|-FunctionDecl 0x7fa0340784a0 <line:20:1, line:24:1> line:21:1 fn3 'void (void)'
| `-CompoundStmt 0x7fa034078588 <line:22:1, line:24:1>
|   `-ReturnStmt 0x7fa034078570 <line:23:3, col:10>
|     `-ImplicitCastExpr 0x7fa034078558 <col:10> 'void' <ToVoid>
|       `-IntegerLiteral 0x7fa034078538 <col:10> 'int' 3
|-FunctionDecl 0x7fa0340786b8 <line:26:1, line:30:1> line:27:1 fn4 'int (int *)'
| |-ParmVarDecl 0x7fa0340785f0 <col:6, col:11> col:11 used a 'int *'
| `-CompoundStmt 0x7fa0340787c8 <line:28:1, line:30:1>
|   `-ReturnStmt 0x7fa0340787b0 <line:29:3, col:10>
|     `-ImplicitCastExpr 0x7fa034078798 <col:10> 'int' <PointerToIntegral>
|       `-ImplicitCastExpr 0x7fa034078780 <col:10> 'int *' <LValueToRValue>
|         `-DeclRefExpr 0x7fa034078758 <col:10> 'int *' lvalue ParmVar 0x7fa0340785f0 'a' 'int *'
|-FunctionDecl 0x7fa0340788c8 <line:32:1, line:36:1> line:33:1 fn5 'int *(int)'
| |-ParmVarDecl 0x7fa034078800 <col:6, col:10> col:10 used a 'int'
| `-CompoundStmt 0x7fa0340789d8 <line:34:1, line:36:1>
|   `-ReturnStmt 0x7fa0340789c0 <line:35:3, col:10>
|     `-ImplicitCastExpr 0x7fa0340789a8 <col:10> 'int *' <IntegralToPointer>
|       `-ImplicitCastExpr 0x7fa034078990 <col:10> 'int' <LValueToRValue>
|         `-DeclRefExpr 0x7fa034078968 <col:10> 'int' lvalue ParmVar 0x7fa034078800 'a' 'int'
|-FunctionDecl 0x7fa034078b18 <line:38:1, line:42:1> line:39:1 fn6 'unsigned int *(int *)'
| |-ParmVarDecl 0x7fa034078a18 <col:6, col:11> col:11 used i 'int *'
| `-CompoundStmt 0x7fa034078c28 <line:40:1, line:42:1>
|   `-ReturnStmt 0x7fa034078c10 <line:41:3, col:10>
|     `-ImplicitCastExpr 0x7fa034078bf8 <col:10> 'unsigned int *' <BitCast>
|       `-ImplicitCastExpr 0x7fa034078be0 <col:10> 'int *' <LValueToRValue>
|         `-DeclRefExpr 0x7fa034078bb8 <col:10> 'int *' lvalue ParmVar 0x7fa034078a18 'i' 'int *'
`-FunctionDecl 0x7fa034078e88 <line:44:1, line:48:1> line:45:1 fn7 'void *(void (*)(void))'
  |-ParmVarDecl 0x7fa034078d78 <col:6, col:22> col:13 used fp 'void (*)(void)'
  `-CompoundStmt 0x7fa034078f98 <line:46:1, line:48:1>
    `-ReturnStmt 0x7fa034078f80 <line:47:3, col:10>
      `-ImplicitCastExpr 0x7fa034078f68 <col:10> 'void *' <BitCast>
        `-ImplicitCastExpr 0x7fa034078f50 <col:10> 'void (*)(void)' <LValueToRValue>
          `-DeclRefExpr 0x7fa034078f28 <col:10> 'void (*)(void)' lvalue ParmVar 0x7fa034078d78 'fp' 'void (*)(void)'
