c_files/pr64778.c:8:13: error: expected expression
  a ^= foo (,); /* { dg-error "expected expression before|too many arguments" } */
            ^
c_files/pr64778.c:8:14: error: expected expression
  a ^= foo (,); /* { dg-error "expected expression before|too many arguments" } */
             ^
2 errors generated.
TranslationUnitDecl 0x7ff38a8264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7ff38a826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7ff38a826740 '__int128'
|-TypedefDecl 0x7ff38a826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7ff38a826760 'unsigned __int128'
|-TypedefDecl 0x7ff38a826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7ff38a826b70 'struct __NSConstantString_tag'
|   `-Record 0x7ff38a826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7ff38a826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7ff38a826dc0 'char *'
|   `-BuiltinType 0x7ff38a826560 'char'
|-TypedefDecl 0x7ff38a8270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7ff38a827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7ff38a826ee0 'struct __va_list_tag'
|     `-Record 0x7ff38a826e58 '__va_list_tag'
`-FunctionDecl 0x7ff38c000030 <c_files/pr64778.c:4:1, line:10:1> line:5:1 referenced foo 'int (int)'
  |-ParmVarDecl 0x7ff38a827138 <col:6, col:10> col:10 p 'int'
  `-CompoundStmt 0x7ff38c000248 <line:6:1, line:10:1>
    |-DeclStmt 0x7ff38c000188 <line:7:3, col:8>
    | `-VarDecl 0x7ff38c000128 <col:3, col:7> col:7 used a 'int'
    `-ReturnStmt 0x7ff38c000230 <line:9:3, col:10>
      `-ImplicitCastExpr 0x7ff38c000218 <col:10> 'int' <LValueToRValue>
        `-DeclRefExpr 0x7ff38c0001f0 <col:10> 'int' lvalue Var 0x7ff38c000128 'a' 'int'
