c_files/pr65228.c:5:17: error: use of undeclared identifier 'b'
__auto_type a = b; /* { dg-error "undeclared" } */
                ^
c_files/pr65228.c:10:19: error: use of undeclared identifier 'd'
  __auto_type c = d; /* { dg-error "undeclared" } */
                  ^
2 errors generated.
TranslationUnitDecl 0x7fa9ef0264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fa9ef026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fa9ef026740 '__int128'
|-TypedefDecl 0x7fa9ef026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fa9ef026760 'unsigned __int128'
|-TypedefDecl 0x7fa9ef026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fa9ef026b70 'struct __NSConstantString_tag'
|   `-Record 0x7fa9ef026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fa9ef026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fa9ef026dc0 'char *'
|   `-BuiltinType 0x7fa9ef026560 'char'
|-TypedefDecl 0x7fa9ef0270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fa9ef027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fa9ef026ee0 'struct __va_list_tag'
|     `-Record 0x7fa9ef026e58 '__va_list_tag'
|-VarDecl 0x7fa9ef027150 <c_files/pr65228.c:5:1, col:13> col:13 invalid a '__auto_type'
`-FunctionDecl 0x7fa9ef825af0 <line:7:1, line:11:1> line:8:1 f 'void (void)'
  `-CompoundStmt 0x7fa9ef825c20 <line:9:1, line:11:1>
    `-DeclStmt 0x7fa9ef825c08 <line:10:3, col:20>
      `-VarDecl 0x7fa9ef825b98 <col:3, col:15> col:15 invalid c '__auto_type'
