c_files/pr67088.c:9:10: error: bit-field 'b1' has non-integral type 'double'
  double b1:1; /* { dg-error "10:has invalid type" } */
         ^  ~
c_files/pr67088.c:10:10: error: expression is not an integer constant expression
  int b2:j; /* { dg-error "7:width not an integer constant" } */
         ^
c_files/pr67088.c:11:10: error: expression is not an integer constant expression
  int b3:f; /* { dg-error "7:width not an integer constant" } */
         ^
c_files/pr67088.c:12:16: error: expression is not an integer constant expression
  int b4:(int) __builtin_log (100); /* { dg-warning "7:width not an integer constant" } */
         ~~~~~~^~~~~~~~~~~~~~~~~~~
c_files/pr67088.c:13:7: error: bit-field 'b5' has negative width (-2)
  int b5:-2; /* { dg-error "7:negative width" } */
      ^
c_files/pr67088.c:14:7: error: named bit-field 'b6' has zero width
  int b6:0; /* { dg-error "7:zero width" } */
      ^
c_files/pr67088.c:16:7: error: width of bit-field 'b8' (64 bits) exceeds width of its type (32 bits)
  int b8:sizeof (int) * __CHAR_BIT__ * 2; /* { dg-error "7:width of" } */
      ^
7 errors generated.
TranslationUnitDecl 0x7f980400d4d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7f980400da20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7f980400d740 '__int128'
|-TypedefDecl 0x7f980400da90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7f980400d760 'unsigned __int128'
|-TypedefDecl 0x7f980400dd68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7f980400db70 'struct __NSConstantString_tag'
|   `-Record 0x7f980400dae8 '__NSConstantString_tag'
|-TypedefDecl 0x7f980400de00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7f980400ddc0 'char *'
|   `-BuiltinType 0x7f980400d560 'char'
|-TypedefDecl 0x7f980400e0c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7f980400e070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7f980400dee0 'struct __va_list_tag'
|     `-Record 0x7f980400de58 '__va_list_tag'
|-EnumDecl 0x7f980400e120 <c_files/pr67088.c:5:1, col:16> col:6 E
| `-EnumConstantDecl 0x7f980302a200 <col:10, col:14> col:10 A 'int'
|   `-IntegerLiteral 0x7f980400e1e0 <col:14> 'int' 2
|-VarDecl 0x7f980302a260 <line:6:1, col:5> col:5 used j 'int'
|-VarDecl 0x7f980302a318 <line:7:1, col:7> col:7 used f 'float'
|-RecordDecl 0x7f980302a378 <line:8:1, line:18:1> line:8:8 invalid struct S1 definition
| |-FieldDecl 0x7f980302a810 <line:9:3, col:10> col:10 invalid b1 'double'
| |-FieldDecl 0x7f980302a898 <line:10:3, col:7> col:7 invalid b2 'int'
| |-FieldDecl 0x7f980302a920 <line:11:3, col:7> col:7 invalid b3 'int'
| |-FieldDecl 0x7f980302ac30 <line:12:3, col:7> col:7 invalid b4 'int'
| |-FieldDecl 0x7f980302acd0 <line:13:3, col:7> col:7 invalid b5 'int'
| |-FieldDecl 0x7f980302ad50 <line:14:3, col:7> col:7 invalid b6 'int'
| |-FieldDecl 0x7f980302add0 <line:15:3, col:15> col:12 b7 'long'
| | `-IntegerLiteral 0x7f980302ad98 <col:15> 'int' 32
| |-FieldDecl 0x7f980302af28 <line:16:3, col:7> col:7 invalid b8 'int'
| `-FieldDecl 0x7f980302afe0 <line:17:17, col:27> col:24 b9 'enum E':'enum E'
|   `-IntegerLiteral 0x7f980302af70 <col:27> 'int' 1
`-FunctionDecl 0x7f980302a9a8 <line:12:16> col:16 implicit used __builtin_log 'double (double)' extern
  |-ParmVarDecl 0x7f980302aa40 <<invalid sloc>> <invalid sloc> 'double'
  |-NoThrowAttr 0x7f980302aaa8 <col:16> Implicit
  `-ConstAttr 0x7f980302aaf0 <col:16> Implicit
