c_files/pr67432.c:4:7: error: use of empty enum
enum {}; /* { dg-error "empty enum is invalid" } */
      ^
c_files/pr67432.c:5:9: error: use of empty enum
enum E {}; /* { dg-error "empty enum is invalid" } */
        ^
c_files/pr67432.c:6:9: error: use of empty enum
enum F {} e; /* { dg-error "empty enum is invalid" } */
        ^
3 errors generated.
TranslationUnitDecl 0x7fd8700264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fd870026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fd870026740 '__int128'
|-TypedefDecl 0x7fd870026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fd870026760 'unsigned __int128'
|-TypedefDecl 0x7fd870026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fd870026b70 'struct __NSConstantString_tag'
|   `-Record 0x7fd870026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fd870026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fd870026dc0 'char *'
|   `-BuiltinType 0x7fd870026560 'char'
|-TypedefDecl 0x7fd8700270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fd870027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fd870026ee0 'struct __va_list_tag'
|     `-Record 0x7fd870026e58 '__va_list_tag'
|-EnumDecl 0x7fd870027120 <c_files/pr67432.c:4:1, col:7> col:1
|-EnumDecl 0x7fd870800000 <line:5:1, col:9> col:6 E
|-EnumDecl 0x7fd8708000c0 <line:6:1, col:9> col:6 F
`-VarDecl 0x7fd8708001d0 <col:1, col:11> col:11 e 'enum F':'enum F'
