c_files/pr69627.c:10:8: error: invalid operands to binary expression ('float' and 'const int *')
  t[1] / s;     /* { dg-error "invalid operands to binary /" } */
  ~~~~ ^ ~
c_files/pr69627.c:22:8: error: invalid operands to binary expression ('float' and 'const int *')
  t[1] / s[0];  /* { dg-error "invalid operands to binary /" } */
  ~~~~ ^ ~~~~
2 errors generated.
TranslationUnitDecl 0x7fa1810264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fa181026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fa181026740 '__int128'
|-TypedefDecl 0x7fa181026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fa181026760 'unsigned __int128'
|-TypedefDecl 0x7fa181026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fa181026b70 'struct __NSConstantString_tag'
|   `-Record 0x7fa181026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fa181026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fa181026dc0 'char *'
|   `-BuiltinType 0x7fa181026560 'char'
|-TypedefDecl 0x7fa1810270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fa181027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fa181026ee0 'struct __va_list_tag'
|     `-Record 0x7fa181026e58 '__va_list_tag'
|-FunctionDecl 0x7fa181027168 <c_files/pr69627.c:5:1, line:15:1> line:6:1 foo 'void ()'
| `-CompoundStmt 0x7fa1818261d0 <line:7:1, line:15:1>
|   |-DeclStmt 0x7fa181825c30 <line:8:3, col:24>
|   | `-VarDecl 0x7fa181825ac0 <col:3, col:23> col:9 used t 'float [2]' cinit
|   |   `-InitListExpr 0x7fa181825bb0 <col:16, col:23> 'float [2]'
|   |     |-ImplicitCastExpr 0x7fa181825c00 <col:18> 'float' <IntegralToFloating>
|   |     | `-IntegerLiteral 0x7fa181825b20 <col:18> 'int' 1
|   |     `-ImplicitCastExpr 0x7fa181825c18 <col:21> 'float' <IntegralToFloating>
|   |       `-IntegerLiteral 0x7fa181825b40 <col:21> 'int' 2
|   `-DeclStmt 0x7fa181825cf8 <line:9:3, col:19>
|     `-VarDecl 0x7fa181825c60 <col:3, col:18> col:14 used s 'const int *' cinit
|       `-ImplicitCastExpr 0x7fa181825ce0 <col:18> 'const int *' <NullToPointer>
|         `-IntegerLiteral 0x7fa181825cc0 <col:18> 'int' 0
`-FunctionDecl 0x7fa181826218 <line:17:1, line:27:1> line:18:1 bar 'void ()'
  `-CompoundStmt 0x7fa1818267f8 <line:19:1, line:27:1>
    |-DeclStmt 0x7fa181826468 <line:20:3, col:24>
    | `-VarDecl 0x7fa1818262f8 <col:3, col:23> col:9 used t 'float [2]' cinit
    |   `-InitListExpr 0x7fa1818263e8 <col:16, col:23> 'float [2]'
    |     |-ImplicitCastExpr 0x7fa181826438 <col:18> 'float' <IntegralToFloating>
    |     | `-IntegerLiteral 0x7fa181826358 <col:18> 'int' 1
    |     `-ImplicitCastExpr 0x7fa181826450 <col:21> 'float' <IntegralToFloating>
    |       `-IntegerLiteral 0x7fa181826378 <col:21> 'int' 2
    `-DeclStmt 0x7fa181826670 <line:21:3, col:29>
      `-VarDecl 0x7fa181826500 <col:3, col:28> col:14 used s 'const int *[2]' cinit
        `-InitListExpr 0x7fa1818265f0 <col:21, col:28> 'const int *[2]'
          |-ImplicitCastExpr 0x7fa181826640 <col:23> 'const int *' <NullToPointer>
          | `-IntegerLiteral 0x7fa181826560 <col:23> 'int' 0
          `-ImplicitCastExpr 0x7fa181826658 <col:26> 'const int *' <NullToPointer>
            `-IntegerLiteral 0x7fa181826580 <col:26> 'int' 0
