c_files/pr69733.c:25:1: error: unknown type name '__seg_fs'
__seg_fs int val3() {return val;} /* { dg-warning "qualifiers ignored" } */
^
c_files/pr69733.c:25:10: error: expected identifier or '('
__seg_fs int val3() {return val;} /* { dg-warning "qualifiers ignored" } */
         ^
2 errors generated.
TranslationUnitDecl 0x7fb7e98264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fb7e9826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fb7e9826740 '__int128'
|-TypedefDecl 0x7fb7e9826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fb7e9826760 'unsigned __int128'
|-TypedefDecl 0x7fb7e9826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fb7e9826b70 'struct __NSConstantString_tag'
|   `-Record 0x7fb7e9826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fb7e9826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fb7e9826dc0 'char *'
|   `-BuiltinType 0x7fb7e9826560 'char'
|-TypedefDecl 0x7fb7e98270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fb7e9827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fb7e9826ee0 'struct __va_list_tag'
|     `-Record 0x7fb7e9826e58 '__va_list_tag'
|-TypedefDecl 0x7fb7e9827138 <c_files/pr69733.c:4:1, col:22> col:22 referenced cd 'const double'
| `-QualType 0x7fb7e98266e1 'const double' const
|   `-BuiltinType 0x7fb7e98266e0 'double'
|-VarDecl 0x7fb7ea025a00 <line:5:1, col:8> col:8 used val 'double'
|-FunctionDecl 0x7fb7ea025af0 <line:7:1, col:33> col:14 val0 'const double ()'
| `-CompoundStmt 0x7fb7ea025be0 <col:21, col:33>
|   `-ReturnStmt 0x7fb7ea025bc8 <col:22, col:29>
|     `-ImplicitCastExpr 0x7fb7ea025bb0 <col:29> 'double' <LValueToRValue>
|       `-DeclRefExpr 0x7fb7ea025b88 <col:29> 'double' lvalue Var 0x7fb7ea025a00 'val' 'double'
|-FunctionDecl 0x7fb7ea025c50 <line:13:1, col:36> col:17 val1 'volatile double ()'
| `-CompoundStmt 0x7fb7ea025d40 <col:24, col:36>
|   `-ReturnStmt 0x7fb7ea025d28 <col:25, col:32>
|     `-ImplicitCastExpr 0x7fb7ea025d10 <col:32> 'double' <LValueToRValue>
|       `-DeclRefExpr 0x7fb7ea025ce8 <col:32> 'double' lvalue Var 0x7fb7ea025a00 'val' 'double'
`-FunctionDecl 0x7fb7ea025dc8 <line:19:1, col:23> col:4 val2 'cd ()'
  `-CompoundStmt 0x7fb7ea025eb8 <col:11, col:23>
    `-ReturnStmt 0x7fb7ea025ea0 <col:12, col:19>
      `-ImplicitCastExpr 0x7fb7ea025e88 <col:19> 'double' <LValueToRValue>
        `-DeclRefExpr 0x7fb7ea025e60 <col:19> 'double' lvalue Var 0x7fb7ea025a00 'val' 'double'
