c_files/pr71362.c:4:22: error: array size is negative
extern void foo (int[-1]); /* { dg-error "21:size of unnamed array is negative" } */
                     ^~
c_files/pr71362.c:9:21: error: array size is negative
  123 + sizeof (int[-1]); /* { dg-error "20:size of unnamed array is negative" } */
                    ^~
2 errors generated.
TranslationUnitDecl 0x7f84870166d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7f8487016c20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7f8487016940 '__int128'
|-TypedefDecl 0x7f8487016c90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7f8487016960 'unsigned __int128'
|-TypedefDecl 0x7f8487016f68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7f8487016d70 'struct __NSConstantString_tag'
|   `-Record 0x7f8487016ce8 '__NSConstantString_tag'
|-TypedefDecl 0x7f8487017000 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7f8487016fc0 'char *'
|   `-BuiltinType 0x7f8487016760 'char'
|-TypedefDecl 0x7f84870172c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7f8487017270 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7f84870170e0 'struct __va_list_tag'
|     `-Record 0x7f8487017058 '__va_list_tag'
|-FunctionDecl 0x7f8486836460 <c_files/pr71362.c:4:1, col:25> col:13 invalid foo 'void (int)' extern
| `-ParmVarDecl 0x7f8487017378 <col:18, <invalid sloc>> col:21 invalid 'int'
`-FunctionDecl 0x7f84868365c8 <line:6:1, line:10:1> line:7:1 bar 'int (void)'
  `-CompoundStmt 0x7f8486836718 <line:8:1, line:10:1>
