c_files/pr71583.c:7:20: error: expected expression
  (int (*)[++i]) { int }; /* { dg-error "expected" } */
                   ^
c_files/pr71583.c:8:18: error: scalar initializer cannot be empty
  (int (*)[++i]) { }; /* { dg-error "empty" } */
                 ^~~
c_files/pr71583.c:9:20: error: expected expression
  (int (*)[++i]) { , }; /* { dg-error "expected" } */
                   ^
c_files/pr71583.c:10:23: error: too few arguments to function call, single argument 'i' was not specified
  (int (*)[++i]) { f () }; /* { dg-error "too few" } */
                   ~  ^
c_files/pr71583.c:4:1: note: 'f' declared here
void
^
4 errors generated.
TranslationUnitDecl 0x7fade7818ed0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fade7819420 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fade7819140 '__int128'
|-TypedefDecl 0x7fade7819490 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fade7819160 'unsigned __int128'
|-TypedefDecl 0x7fade7819768 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fade7819570 'struct __NSConstantString_tag'
|   `-Record 0x7fade78194e8 '__NSConstantString_tag'
|-TypedefDecl 0x7fade7819800 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fade78197c0 'char *'
|   `-BuiltinType 0x7fade7818f60 'char'
|-TypedefDecl 0x7fade7819ac8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fade7819a70 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fade78198e0 'struct __va_list_tag'
|     `-Record 0x7fade7819858 '__va_list_tag'
`-FunctionDecl 0x7fade7033200 <c_files/pr71583.c:4:1, line:11:1> line:5:1 referenced f 'void (int)'
  |-ParmVarDecl 0x7fade7819b38 <col:4, col:8> col:8 used i 'int'
  `-CompoundStmt 0x7fade7033950 <line:6:1, line:11:1>
