c_files/pr79757-2.c:10:23: error: expected ';' at end of declaration
  unsigned long long z = x / y; /* { dg-error "parameter 'z' is initialized" } */
                      ^
                      ;
c_files/pr79757-2.c:16:23: error: expected ';' at end of declaration
  unsigned long long z = x >> y; /* { dg-error "parameter 'z' is initialized" } */
                      ^
                      ;
2 errors generated.
TranslationUnitDecl 0x7fc4df0264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fc4df026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fc4df026740 '__int128'
|-TypedefDecl 0x7fc4df026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fc4df026760 'unsigned __int128'
|-TypedefDecl 0x7fc4df026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fc4df026b70 'struct __NSConstantString_tag'
|   `-Record 0x7fc4df026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fc4df026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fc4df026dc0 'char *'
|   `-BuiltinType 0x7fc4df026560 'char'
|-TypedefDecl 0x7fc4df0270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fc4df027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fc4df026ee0 'struct __va_list_tag'
|     `-Record 0x7fc4df026e58 '__va_list_tag'
|-VarDecl 0x7fc4df027138 <c_files/pr79757-2.c:6:1, col:19> col:19 x 'unsigned __int128'
|-VarDecl 0x7fc4df825a00 <col:1, col:22> col:22 y 'unsigned __int128'
|-FunctionDecl 0x7fc4df825b50 <line:8:1, line:12:1> line:9:1 fn1 'void (unsigned long long)'
| |-ParmVarDecl 0x7fc4df825a90 <line:10:3, col:22> col:22 z 'unsigned long long'
| `-CompoundStmt 0x7fc4df825bf0 <line:11:1, line:12:1>
`-FunctionDecl 0x7fc4df825ca8 <line:14:1, line:18:1> line:15:1 fn2 'void (unsigned long long)'
  |-ParmVarDecl 0x7fc4df825c20 <line:16:3, col:22> col:22 z 'unsigned long long'
  `-CompoundStmt 0x7fc4df825d48 <line:17:1, line:18:1>
