c_files/pr79757-3.c:10:3: error: '__auto_type' not allowed in K&R-style function parameter
  __auto_type z = x / y; /* { dg-error "parameter 'z' is initialized" } */
  ^~~~~~~~~~~
c_files/pr79757-3.c:10:16: error: expected ';' at end of declaration
  __auto_type z = x / y; /* { dg-error "parameter 'z' is initialized" } */
               ^
               ;
c_files/pr79757-3.c:16:3: error: '__auto_type' not allowed in K&R-style function parameter
  __auto_type z = x >> y; /* { dg-error "parameter 'z' is initialized" } */
  ^~~~~~~~~~~
c_files/pr79757-3.c:16:16: error: expected ';' at end of declaration
  __auto_type z = x >> y; /* { dg-error "parameter 'z' is initialized" } */
               ^
               ;
4 errors generated.
TranslationUnitDecl 0x7ffd670234d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7ffd67023a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7ffd67023740 '__int128'
|-TypedefDecl 0x7ffd67023a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7ffd67023760 'unsigned __int128'
|-TypedefDecl 0x7ffd67023d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7ffd67023b70 'struct __NSConstantString_tag'
|   `-Record 0x7ffd67023ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7ffd67023e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7ffd67023dc0 'char *'
|   `-BuiltinType 0x7ffd67023560 'char'
|-TypedefDecl 0x7ffd670240c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7ffd67024070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7ffd67023ee0 'struct __va_list_tag'
|     `-Record 0x7ffd67023e58 '__va_list_tag'
|-VarDecl 0x7ffd67024138 <c_files/pr79757-3.c:6:1, col:19> col:19 x 'unsigned __int128'
|-VarDecl 0x7ffd67074400 <col:1, col:22> col:22 y 'unsigned __int128'
|-FunctionDecl 0x7ffd67074580 <line:8:1, line:12:1> line:9:1 invalid fn1 'void (int)'
| |-ParmVarDecl 0x7ffd670744b8 <line:10:3, col:15> col:15 invalid z 'int'
| `-CompoundStmt 0x7ffd67074620 <line:11:1, line:12:1>
`-FunctionDecl 0x7ffd670746d8 <line:14:1, line:18:1> line:15:1 invalid fn2 'void (int)'
  |-ParmVarDecl 0x7ffd67074650 <line:16:3, col:15> col:15 invalid z 'int'
  `-CompoundStmt 0x7ffd67074778 <line:17:1, line:18:1>
