c_files/pr79730.c:5:1: error: illegal storage class on function
register int x() asm (""); /* { dg-error "invalid storage class" } */
^
c_files/pr79730.c:6:1: error: illegal storage class on function
register float y() asm (""); /* { dg-error "invalid storage class" } */
^
2 errors generated.
TranslationUnitDecl 0x7f8dd30264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7f8dd3026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7f8dd3026740 '__int128'
|-TypedefDecl 0x7f8dd3026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7f8dd3026760 'unsigned __int128'
|-TypedefDecl 0x7f8dd3026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7f8dd3026b70 'struct __NSConstantString_tag'
|   `-Record 0x7f8dd3026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7f8dd3026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7f8dd3026dc0 'char *'
|   `-BuiltinType 0x7f8dd3026560 'char'
|-TypedefDecl 0x7f8dd30270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7f8dd3027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7f8dd3026ee0 'struct __va_list_tag'
|     `-Record 0x7f8dd3026e58 '__va_list_tag'
|-FunctionDecl 0x7f8dd4025a00 <c_files/pr79730.c:5:1, col:25> col:14 invalid x 'int ()'
| `-AsmLabelAttr 0x7f8dd4025a98 <col:23> ""
`-FunctionDecl 0x7f8dd4025b60 <line:6:1, col:27> col:16 invalid y 'float ()'
  `-AsmLabelAttr 0x7f8dd4025bf8 <col:25> ""
