c_files/pr79983.c:7:8: error: redefinition of 'S'
struct S { int i, j; }; /* { dg-error "redefinition of 'struct S'" } */
       ^
c_files/pr79983.c:6:8: note: previous definition is here
struct S { int i; }; /* { dg-message "originally defined here" } */
       ^
c_files/pr79983.c:11:6: error: redefinition of 'E'
enum E { D, F }; /* { dg-error "nested redefinition of 'enum E'|redeclaration of 'enum E'" } */
     ^
c_files/pr79983.c:10:6: note: previous definition is here
enum E { A, B, C }; /* { dg-message "originally defined here" } */
     ^
c_files/pr79983.c:15:7: error: redefinition of 'U'
union U { int i; double d; }; /* { dg-error "redefinition of 'union U'" } */
      ^
c_files/pr79983.c:14:7: note: previous definition is here
union U { int i; }; /* { dg-message "originally defined here" } */
      ^
3 errors generated.
TranslationUnitDecl 0x7fae920264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fae92026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fae92026740 '__int128'
|-TypedefDecl 0x7fae92026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fae92026760 'unsigned __int128'
|-TypedefDecl 0x7fae92026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fae92026b70 'struct __NSConstantString_tag'
|   `-Record 0x7fae92026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fae92026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fae92026dc0 'char *'
|   `-BuiltinType 0x7fae92026560 'char'
|-TypedefDecl 0x7fae920270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fae92027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fae92026ee0 'struct __va_list_tag'
|     `-Record 0x7fae92026e58 '__va_list_tag'
|-RecordDecl 0x7fae92027120 <c_files/pr79983.c:5:1, col:8> col:8 struct S
|-RecordDecl 0x7fae92825a00 prev 0x7fae92027120 <line:6:1, col:19> col:8 struct S definition
| `-FieldDecl 0x7fae92825aa0 <col:12, col:16> col:16 i 'int'
|-RecordDecl 0x7fae92825ae8 <line:7:1, col:22> col:8 invalid struct definition
| |-FieldDecl 0x7fae92825ba8 <col:12, col:16> col:16 i 'int'
| `-FieldDecl 0x7fae92825c08 <col:12, col:19> col:19 j 'int'
|-EnumDecl 0x7fae92825c50 <line:9:1, col:6> col:6 E
|-EnumDecl 0x7fae92825d10 prev 0x7fae92825c50 <line:10:1, col:18> col:6 E
| |-EnumConstantDecl 0x7fae92825da8 <col:10> col:10 A 'int'
| |-EnumConstantDecl 0x7fae92825df0 <col:13> col:13 B 'int'
| `-EnumConstantDecl 0x7fae92825e38 <col:16> col:16 C 'int'
|-EnumDecl 0x7fae92825e80 <line:11:1, col:15> col:6 invalid
| |-EnumConstantDecl 0x7fae92825f40 <col:10> col:10 D 'int'
| `-EnumConstantDecl 0x7fae92825f88 <col:13> col:13 F 'int'
|-RecordDecl 0x7fae92825fd0 <line:13:1, col:7> col:7 union U
|-RecordDecl 0x7fae92826080 prev 0x7fae92825fd0 <line:14:1, col:18> col:7 union U definition
| `-FieldDecl 0x7fae92826120 <col:11, col:15> col:15 i 'int'
`-RecordDecl 0x7fae92826168 <line:15:1, col:28> col:7 invalid union definition
  |-FieldDecl 0x7fae92826228 <col:11, col:15> col:15 i 'int'
  `-FieldDecl 0x7fae92826288 <col:18, col:25> col:25 d 'double'
