c_files/pr83488-3.c:10:8: error: use of unknown builtin '__builtin_ia32_vpdpbusd_v8si' [-Wimplicit-function-declaration]
  *a = __builtin_ia32_vpdpbusd_v8si (*b, *c, *d);       /* { dg-error "needs isa option" } */
       ^
c_files/pr83488-3.c:10:6: error: assigning to '__v8si' (vector of 8 'int' values) from incompatible type 'int'
  *a = __builtin_ia32_vpdpbusd_v8si (*b, *c, *d);       /* { dg-error "needs isa option" } */
     ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2 errors generated.
TranslationUnitDecl 0x7f91850264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7f9185026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7f9185026740 '__int128'
|-TypedefDecl 0x7f9185026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7f9185026760 'unsigned __int128'
|-TypedefDecl 0x7f9185026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7f9185026b70 'struct __NSConstantString_tag'
|   `-Record 0x7f9185026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7f9185026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7f9185026dc0 'char *'
|   `-BuiltinType 0x7f9185026560 'char'
|-TypedefDecl 0x7f91850270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7f9185027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7f9185026ee0 'struct __va_list_tag'
|     `-Record 0x7f9185026e58 '__va_list_tag'
|-TypedefDecl 0x7f9185027178 <c_files/pr83488-3.c:5:1, col:13> col:13 referenced __v8si '__attribute__((__vector_size__(8 * sizeof(int)))) int'
| `-VectorType 0x7f9185027140 '__attribute__((__vector_size__(8 * sizeof(int)))) int' 8
|   `-BuiltinType 0x7f91850265c0 'int'
|-FunctionDecl 0x7f9185826d00 <line:7:1, line:11:1> line:8:1 foo 'void (__v8si *, __v8si *, __v8si *, __v8si *)'
| |-ParmVarDecl 0x7f9185826a68 <col:6, col:14> col:14 used a '__v8si *'
| |-ParmVarDecl 0x7f9185826ad8 <col:17, col:25> col:25 used b '__v8si *'
| |-ParmVarDecl 0x7f9185826b48 <col:28, col:36> col:36 used c '__v8si *'
| |-ParmVarDecl 0x7f9185826bb8 <col:39, col:47> col:47 used d '__v8si *'
| `-CompoundStmt 0x7f9185827520 <line:9:1, line:11:1>
`-FunctionDecl 0x7f9185826ec0 <<invalid sloc>> line:10:8 implicit used __builtin_ia32_vpdpbusd_v8si 'int ()'
