c_files/pr14963.c:1:1: error: unknown type name 'X'
X Y(Z); /* { dg-error "" "error before" } */
^
c_files/pr14963.c:1:5: error: a parameter list without types is only allowed in a function definition
X Y(Z); /* { dg-error "" "error before" } */
    ^
2 errors generated.
TranslationUnitDecl 0x7fba518264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fba51826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fba51826740 '__int128'
|-TypedefDecl 0x7fba51826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fba51826760 'unsigned __int128'
|-TypedefDecl 0x7fba51826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fba51826b70 'struct __NSConstantString_tag'
|   `-Record 0x7fba51826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fba51826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fba51826dc0 'char *'
|   `-BuiltinType 0x7fba51826560 'char'
|-TypedefDecl 0x7fba518270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fba51827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fba51826ee0 'struct __va_list_tag'
|     `-Record 0x7fba51826e58 '__va_list_tag'
`-FunctionDecl 0x7fba5181b200 <c_files/pr14963.c:1:1, col:6> col:3 invalid Y 'int ()'
