c_files/pr15698-2.c:9:6: error: conflicting types for 'g'
void g(a) int a; {} /* { dg-error "number of arguments doesn't match prototype" } */
     ^
c_files/pr15698-2.c:8:17: note: previous declaration is here
void f() { void g(void); } /* { dg-error "prototype declaration" } */
                ^
1 error generated.
TranslationUnitDecl 0x7f8a420264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7f8a42026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7f8a42026740 '__int128'
|-TypedefDecl 0x7f8a42026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7f8a42026760 'unsigned __int128'
|-TypedefDecl 0x7f8a42026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7f8a42026b70 'struct __NSConstantString_tag'
|   `-Record 0x7f8a42026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7f8a42026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7f8a42026dc0 'char *'
|   `-BuiltinType 0x7f8a42026560 'char'
|-TypedefDecl 0x7f8a420270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7f8a42027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7f8a42026ee0 'struct __va_list_tag'
|     `-Record 0x7f8a42026e58 '__va_list_tag'
|-FunctionDecl 0x7f8a42027168 <c_files/pr15698-2.c:8:1, col:26> col:6 f 'void ()'
| `-CompoundStmt 0x7f8a420775c0 <col:10, col:26>
|   `-DeclStmt 0x7f8a420775a8 <col:12, col:24>
|     `-FunctionDecl 0x7f8a42077500 parent 0x7f8a420264d0 <col:12, col:23> col:17 g 'void (void)'
`-FunctionDecl 0x7f8a420776c0 <line:9:1, col:19> col:6 invalid g 'void (int)'
  |-ParmVarDecl 0x7f8a420775f8 <col:11, col:15> col:15 a 'int'
  `-CompoundStmt 0x7f8a42077760 <col:18, col:19>
