c_files/pr15698-7.c:10:7: error: conflicting types for 'strchr'
char *strchr(a) const char *a; { return 0; } /* { dg-error "number of arguments doesn't match prototype" } */
      ^
c_files/pr15698-7.c:9:7: note: previous declaration is here
char *strchr(const char *, int); /* { dg-error "prototype declaration" } */
      ^
1 error generated.
TranslationUnitDecl 0x7ff2380264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7ff238026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7ff238026740 '__int128'
|-TypedefDecl 0x7ff238026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7ff238026760 'unsigned __int128'
|-TypedefDecl 0x7ff238026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7ff238026b70 'struct __NSConstantString_tag'
|   `-Record 0x7ff238026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7ff238026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7ff238026dc0 'char *'
|   `-BuiltinType 0x7ff238026560 'char'
|-TypedefDecl 0x7ff2380270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7ff238027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7ff238026ee0 'struct __va_list_tag'
|     `-Record 0x7ff238026e58 '__va_list_tag'
|-FunctionDecl 0x7ff238825ad8 <c_files/pr15698-7.c:9:7> col:7 implicit strchr 'char *(const char *, int)' extern
| |-ParmVarDecl 0x7ff238825b70 <<invalid sloc>> <invalid sloc> 'const char *'
| `-ParmVarDecl 0x7ff238825bd0 <<invalid sloc>> <invalid sloc> 'int'
|-FunctionDecl 0x7ff238825c40 prev 0x7ff238825ad8 <col:1, col:31> col:7 strchr 'char *(const char *, int)'
| |-ParmVarDecl 0x7ff238027138 <col:14, col:25> col:26 'const char *'
| `-ParmVarDecl 0x7ff238825a00 <col:28> col:31 'int'
`-FunctionDecl 0x7ff238825dc8 <line:10:1, col:44> col:7 invalid strchr 'char *(const char *)'
  |-ParmVarDecl 0x7ff238825d00 <col:17, col:29> col:29 a 'const char *'
  `-CompoundStmt 0x7ff238825eb8 <col:32, col:44>
    `-ReturnStmt 0x7ff238825ea0 <col:34, col:41>
      `-ImplicitCastExpr 0x7ff238825e88 <col:41> 'char *' <NullToPointer>
        `-IntegerLiteral 0x7ff238825e68 <col:41> 'int' 0
