c_files/pr25559.c:5:5: error: vector size not an integral multiple of component size
int vs (-1) a;                  /* { dg-warning "attribute ignored" } */
    ^   ~~
c_files/pr25559.c:4:30: note: expanded from macro 'vs'
#define vs(n) __attribute__((vector_size (n)))
                             ^            ~
c_files/pr25559.c:6:5: error: zero vector size
int vs (0) b;                   /* { dg-error "zero vector size" } */
    ^   ~
c_files/pr25559.c:4:30: note: expanded from macro 'vs'
#define vs(n) __attribute__((vector_size (n)))
                             ^            ~
c_files/pr25559.c:7:5: error: vector size not an integral multiple of component size
int vs (1) c;                   /* { dg-error "multiple of component size" } */
    ^   ~
c_files/pr25559.c:4:30: note: expanded from macro 'vs'
#define vs(n) __attribute__((vector_size (n)))
                             ^            ~
c_files/pr25559.c:8:5: error: vector size not an integral multiple of component size
int vs (sizeof (int) / 2) d;    /* { dg-error "multiple of component size" } */
    ^   ~~~~~~~~~~~~~~~~
c_files/pr25559.c:4:30: note: expanded from macro 'vs'
#define vs(n) __attribute__((vector_size (n)))
                             ^            ~
4 errors generated.
TranslationUnitDecl 0x7fadde0264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fadde026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fadde026740 '__int128'
|-TypedefDecl 0x7fadde026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fadde026760 'unsigned __int128'
|-TypedefDecl 0x7fadde026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fadde026b70 'struct __NSConstantString_tag'
|   `-Record 0x7fadde026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fadde026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fadde026dc0 'char *'
|   `-BuiltinType 0x7fadde026560 'char'
|-TypedefDecl 0x7fadde0270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fadde027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fadde026ee0 'struct __va_list_tag'
|     `-Record 0x7fadde026e58 '__va_list_tag'
|-VarDecl 0x7fadde027178 <c_files/pr25559.c:5:1, col:13> col:13 a 'int'
|-VarDecl 0x7fadde826a78 <line:6:1, col:12> col:12 b 'int'
|-VarDecl 0x7fadde826b10 <line:7:1, col:12> col:12 c 'int'
|-VarDecl 0x7fadde826c20 <line:8:1, col:27> col:27 d 'int'
|-VarDecl 0x7fadde826cf8 <line:9:1, col:23> col:23 e '__attribute__((__vector_size__(1 * sizeof(int)))) int'
`-VarDecl 0x7fadde826e28 <line:10:1, col:27> col:27 f '__attribute__((__vector_size__(2 * sizeof(int)))) int'
