c_files/pr35334.c:16:23: error: called object type '_Complex int' is not a function or function pointer
  ((__complex__ int)i)();               /* { dg-error "is not a function" } */
  ~~~~~~~~~~~~~~~~~~~~^
c_files/pr35334.c:17:23: error: called object type '_Complex int' is not a function or function pointer
  ((__complex__ int)j)();               /* { dg-error "is not a function" } */
  ~~~~~~~~~~~~~~~~~~~~^
c_files/pr35334.c:18:23: error: called object type '_Complex int' is not a function or function pointer
  ((__complex__ int)k)();               /* { dg-error "is not a function" } */
  ~~~~~~~~~~~~~~~~~~~~^
c_files/pr35334.c:19:31: error: called object type '_Complex long double' is not a function or function pointer
  ((__complex__ long double)l)();       /* { dg-error "is not a function" } */
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
c_files/pr35334.c:20:31: error: called object type '_Complex long double' is not a function or function pointer
  ((__complex__ long double)m)();       /* { dg-error "is not a function" } */
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
c_files/pr35334.c:21:31: error: called object type '_Complex long double' is not a function or function pointer
  ((__complex__ long double)n)();       /* { dg-error "is not a function" } */
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
6 errors generated.
TranslationUnitDecl 0x7fb305018ad0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fb305019020 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fb305018d40 '__int128'
|-TypedefDecl 0x7fb305019090 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fb305018d60 'unsigned __int128'
|-TypedefDecl 0x7fb305019368 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fb305019170 'struct __NSConstantString_tag'
|   `-Record 0x7fb3050190e8 '__NSConstantString_tag'
|-TypedefDecl 0x7fb305019400 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fb3050193c0 'char *'
|   `-BuiltinType 0x7fb305018b60 'char'
|-TypedefDecl 0x7fb3050196c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fb305019670 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fb3050194e0 'struct __va_list_tag'
|     `-Record 0x7fb305019458 '__va_list_tag'
|-VarDecl 0x7fb305019758 <c_files/pr35334.c:6:1, col:26> col:26 used i '_Complex unsigned int'
|-VarDecl 0x7fb304833618 <line:7:1, col:5> col:5 used j 'int'
|-VarDecl 0x7fb304833688 <line:8:1, col:6> col:6 used k 'char'
|-VarDecl 0x7fb3048336f8 <line:9:1, col:20> col:20 used l '_Complex double'
|-VarDecl 0x7fb304833770 <line:10:1, col:8> col:8 used m 'double'
|-VarDecl 0x7fb3048337e8 <line:11:1, col:7> col:7 used n 'float'
`-FunctionDecl 0x7fb304833898 <line:13:1, line:22:1> line:14:1 foo 'void ()'
  `-CompoundStmt 0x7fb3048341d8 <line:15:1, line:22:1>
