c_files/pr43690.c:8:20: error: invalid lvalue in asm input for constraint 'm'
  asm ("" : : "m" (x++));       /* { dg-error "is not directly addressable" } */
                   ^~~
c_files/pr43690.c:9:20: error: invalid lvalue in asm input for constraint 'm'
  asm ("" : : "m" (++x));       /* { dg-error "is not directly addressable" "" { target c } } */
                   ^~~
c_files/pr43690.c:10:20: error: invalid lvalue in asm input for constraint 'm'
  asm ("" : : "m" (x--));       /* { dg-error "is not directly addressable" } */
                   ^~~
c_files/pr43690.c:11:20: error: invalid lvalue in asm input for constraint 'm'
  asm ("" : : "m" (--x));       /* { dg-error "is not directly addressable" "" { target c } } */
                   ^~~
c_files/pr43690.c:12:20: error: invalid lvalue in asm input for constraint 'm'
  asm ("" : : "m" (x + 1));     /* { dg-error "is not directly addressable" } */
                   ^~~~~
5 errors generated.
TranslationUnitDecl 0x7fa7b40264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fa7b4026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fa7b4026740 '__int128'
|-TypedefDecl 0x7fa7b4026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fa7b4026760 'unsigned __int128'
|-TypedefDecl 0x7fa7b4026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fa7b4026b70 'struct __NSConstantString_tag'
|   `-Record 0x7fa7b4026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fa7b4026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fa7b4026dc0 'char *'
|   `-BuiltinType 0x7fa7b4026560 'char'
|-TypedefDecl 0x7fa7b40270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fa7b4027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fa7b4026ee0 'struct __va_list_tag'
|     `-Record 0x7fa7b4026e58 '__va_list_tag'
`-FunctionDecl 0x7fa7b4077400 <c_files/pr43690.c:5:1, line:13:1> line:6:1 foo 'void (char *)'
  |-ParmVarDecl 0x7fa7b4027138 <col:6, col:12> col:12 used x 'char *'
  `-CompoundStmt 0x7fa7b40778b0 <line:7:1, line:13:1>
