c_files/pr45750.c:9:5: error: array initializer must be an initializer list or wide string literal
int invalid[] = 0; /* { dg-error "invalid initializer" } */
    ^
c_files/pr45750.c:13:20: error: use of undeclared identifier 'undeclared'
  if (((struct S *)undeclared)->i); /* { dg-error "undeclared" } */
                   ^
2 errors generated.
TranslationUnitDecl 0x7ff2790264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7ff279026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7ff279026740 '__int128'
|-TypedefDecl 0x7ff279026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7ff279026760 'unsigned __int128'
|-TypedefDecl 0x7ff279026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7ff279026b70 'struct __NSConstantString_tag'
|   `-Record 0x7ff279026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7ff279026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7ff279026dc0 'char *'
|   `-BuiltinType 0x7ff279026560 'char'
|-TypedefDecl 0x7ff2790270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7ff279027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7ff279026ee0 'struct __va_list_tag'
|     `-Record 0x7ff279026e58 '__va_list_tag'
|-RecordDecl 0x7ff279027120 <c_files/pr45750.c:4:1, line:7:1> line:4:8 struct S definition
| `-FieldDecl 0x7ff279077400 <line:6:3, col:7> col:7 i 'int'
|-VarDecl 0x7ff2790774a0 <line:9:1, col:13> col:5 invalid invalid 'int []'
`-FunctionDecl 0x7ff279077620 <line:11:1, line:15:1> line:11:6 foo 'void (void)'
  `-CompoundStmt 0x7ff2790777f8 <line:12:1, line:15:1>
    `-IfStmt 0x7ff2790777c0 <line:13:3, col:35>
      |-<<<NULL>>>
      |-<<<NULL>>>
      |-OpaqueValueExpr 0x7ff2790777a0 <<invalid sloc>> '_Bool'
      |-NullStmt 0x7ff279077790 <col:35>
      `-<<<NULL>>>
