c_files/pr48552-2.c:10:22: error: invalid type 'void' in asm input for constraint 'r'
  __asm ("" : : "r" (*x));      /* { dg-warning "dereferencing" "deref" } */
                     ^~
c_files/pr48552-2.c:16:21: error: dereference of pointer to incomplete type 'void'
  __asm ("" : "=r" (*x));       /* { dg-warning "dereferencing" "deref" } */
                    ^
c_files/pr48552-2.c:22:22: error: invalid lvalue in asm input for constraint 'm'
  __asm ("" : : "m" (*x));      /* { dg-warning "dereferencing" } */
                     ^~
c_files/pr48552-2.c:28:21: error: dereference of pointer to incomplete type 'void'
  __asm ("" : "=m" (*x));       /* { dg-warning "dereferencing" } */
                    ^
c_files/pr48552-2.c:34:22: error: invalid type 'void' in asm input for constraint 'g'
  __asm ("" : : "g" (*x));      /* { dg-warning "dereferencing" "deref" } */
                     ^~
c_files/pr48552-2.c:40:21: error: dereference of pointer to incomplete type 'void'
  __asm ("" : "=g" (*x));       /* { dg-warning "dereferencing" "deref" } */
                    ^
c_files/pr48552-2.c:46:22: error: dereference of pointer to incomplete type 'struct S'
  __asm ("" : : "r" (*x));      /* { dg-error "dereferencing pointer to incomplete type" } */
                     ^
c_files/pr48552-2.c:5:8: note: forward declaration of 'struct S'
struct S;
       ^
c_files/pr48552-2.c:52:21: error: dereference of pointer to incomplete type 'struct S'
  __asm ("" : "=r" (*x));       /* { dg-error "invalid lvalue in asm output 0" } */
                    ^
c_files/pr48552-2.c:5:8: note: forward declaration of 'struct S'
struct S;
       ^
8 errors generated.
TranslationUnitDecl 0x7fe9260264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fe926026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fe926026740 '__int128'
|-TypedefDecl 0x7fe926026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fe926026760 'unsigned __int128'
|-TypedefDecl 0x7fe926026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fe926026b70 'struct __NSConstantString_tag'
|   `-Record 0x7fe926026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fe926026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fe926026dc0 'char *'
|   `-BuiltinType 0x7fe926026560 'char'
|-TypedefDecl 0x7fe9260270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fe926027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fe926026ee0 'struct __va_list_tag'
|     `-Record 0x7fe926026e58 '__va_list_tag'
|-RecordDecl 0x7fe926027120 <c_files/pr48552-2.c:5:1, col:8> col:8 struct S
|-FunctionDecl 0x7fe926825ac0 <line:7:1, line:11:1> line:8:1 f1 'void (void *)'
| |-ParmVarDecl 0x7fe926825a00 <col:5, col:11> col:11 used x 'void *'
| `-CompoundStmt 0x7fe926826080 <line:9:1, line:11:1>
|-FunctionDecl 0x7fe926826138 <line:13:1, line:17:1> line:14:1 f2 'void (void *)'
| |-ParmVarDecl 0x7fe9268260b0 <col:5, col:11> col:11 used x 'void *'
| `-CompoundStmt 0x7fe9268262c8 <line:15:1, line:17:1>
|-FunctionDecl 0x7fe926826380 <line:19:1, line:23:1> line:20:1 f3 'void (void *)'
| |-ParmVarDecl 0x7fe9268262f8 <col:5, col:11> col:11 used x 'void *'
| `-CompoundStmt 0x7fe9268264d8 <line:21:1, line:23:1>
|-FunctionDecl 0x7fe926826590 <line:25:1, line:29:1> line:26:1 f4 'void (void *)'
| |-ParmVarDecl 0x7fe926826508 <col:5, col:11> col:11 used x 'void *'
| `-CompoundStmt 0x7fe9268266e8 <line:27:1, line:29:1>
|-FunctionDecl 0x7fe9268267a0 <line:31:1, line:35:1> line:32:1 f5 'void (void *)'
| |-ParmVarDecl 0x7fe926826718 <col:5, col:11> col:11 used x 'void *'
| `-CompoundStmt 0x7fe9268268f8 <line:33:1, line:35:1>
|-FunctionDecl 0x7fe926826a00 <line:37:1, line:41:1> line:38:1 f6 'void (void *)'
| |-ParmVarDecl 0x7fe926826928 <col:5, col:11> col:11 used x 'void *'
| `-CompoundStmt 0x7fe926826b58 <line:39:1, line:41:1>
|-FunctionDecl 0x7fe926826d20 <line:43:1, line:47:1> line:44:1 f7 'void (struct S *)'
| |-ParmVarDecl 0x7fe926826c20 <col:5, col:15> col:15 used x 'struct S *'
| `-CompoundStmt 0x7fe926826e90 <line:45:1, line:47:1>
`-FunctionDecl 0x7fe926826f58 <line:49:1, line:53:1> line:50:1 f8 'void (struct S *)'
  |-ParmVarDecl 0x7fe926826ed0 <col:5, col:15> col:15 used x 'struct S *'
  `-CompoundStmt 0x7fe9268270b0 <line:51:1, line:53:1>
