c_files/pr51034.c:10:23: error: variable has incomplete type 'struct W'
  struct R { typeof (((struct W) {})) x; } r;   /* { dg-error "invalid use of undefined type" } */
                      ^~~~~~~~~~~~~
c_files/pr51034.c:10:31: note: forward declaration of 'struct W'
  struct R { typeof (((struct W) {})) x; } r;   /* { dg-error "invalid use of undefined type" } */
                              ^
c_files/pr51034.c:11:23: error: variable has incomplete type 'struct S'
  struct S { typeof (((struct S) {})) x; } s;   /* { dg-error "invalid use of undefined type" } */
                      ^~~~~~~~~~~~~
c_files/pr51034.c:11:10: note: definition of 'struct S' is not complete until the closing '}'
  struct S { typeof (((struct S) {})) x; } s;   /* { dg-error "invalid use of undefined type" } */
         ^
c_files/pr51034.c:12:28: error: variable has incomplete type 'struct T'
  struct T { int x[sizeof ((struct T) {})]; } t;/* { dg-error "invalid use of undefined type" } */
                           ^~~~~~~~~~~~~
c_files/pr51034.c:12:10: note: definition of 'struct T' is not complete until the closing '}'
  struct T { int x[sizeof ((struct T) {})]; } t;/* { dg-error "invalid use of undefined type" } */
         ^
c_files/pr51034.c:13:27: error: variable has incomplete type 'struct V'
  struct U { int x[sizeof((struct V){})];} u;   /* { dg-error "invalid use of undefined type" } */
                          ^~~~~~~~~~~~
c_files/pr51034.c:13:35: note: forward declaration of 'struct V'
  struct U { int x[sizeof((struct V){})];} u;   /* { dg-error "invalid use of undefined type" } */
                                  ^
4 errors generated.
TranslationUnitDecl 0x7fa39d8264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fa39d826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fa39d826740 '__int128'
|-TypedefDecl 0x7fa39d826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fa39d826760 'unsigned __int128'
|-TypedefDecl 0x7fa39d826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fa39d826b70 'struct __NSConstantString_tag'
|   `-Record 0x7fa39d826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fa39d826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fa39d826dc0 'char *'
|   `-BuiltinType 0x7fa39d826560 'char'
|-TypedefDecl 0x7fa39d8270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fa39d827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fa39d826ee0 'struct __va_list_tag'
|     `-Record 0x7fa39d826e58 '__va_list_tag'
|-RecordDecl 0x7fa39d827120 <c_files/pr51034.c:5:1, col:8> col:8 struct S
`-FunctionDecl 0x7fa39e025a28 <line:7:1, line:14:1> line:8:1 main 'int ()'
  `-CompoundStmt 0x7fa39e0269d0 <line:9:1, line:14:1>
    |-DeclStmt 0x7fa39e0261d8 <line:10:3, col:45>
    | |-RecordDecl 0x7fa39e025b00 <col:3, col:42> col:10 invalid struct R definition
    | | |-RecordDecl 0x7fa39e025bb0 parent 0x7fa39e025a28 <col:24, col:31> col:31 struct W
    | | `-FieldDecl 0x7fa39e0260c8 <col:14, col:39> col:39 invalid x 'int'
    | `-VarDecl 0x7fa39e026160 <col:3, col:44> col:44 r 'struct R':'struct R'
    |-DeclStmt 0x7fa39e026428 <line:11:3, col:45>
    | |-RecordDecl 0x7fa39e0261f0 <col:3, col:42> col:10 invalid struct S definition
    | | `-FieldDecl 0x7fa39e026348 <col:14, col:39> col:39 invalid x 'int'
    | `-VarDecl 0x7fa39e0263b0 <col:3, col:44> col:44 s 'struct S':'struct S'
    |-DeclStmt 0x7fa39e026678 <line:12:3, col:48>
    | |-RecordDecl 0x7fa39e026440 <col:3, col:45> col:10 invalid struct T definition
    | | `-FieldDecl 0x7fa39e026598 <col:14, col:18> col:18 invalid x 'int'
    | `-VarDecl 0x7fa39e026600 <col:3, col:47> col:47 t 'struct T':'struct T'
    `-DeclStmt 0x7fa39e0269b8 <line:13:3, col:45>
      |-RecordDecl 0x7fa39e026690 <col:3, col:42> col:10 invalid struct U definition
      | |-RecordDecl 0x7fa39e026740 parent 0x7fa39e025a28 <col:28, col:35> col:35 struct V
      | `-FieldDecl 0x7fa39e0268a8 <col:14, col:18> col:18 invalid x 'int'
      `-VarDecl 0x7fa39e026940 <col:3, col:44> col:44 u 'struct U':'struct U'
