c_files/pr51768.c:8:7: error: 'asm goto' constructs are not supported yet
  asm goto ("" : : : : lab, lab, lab2, lab);    /* { dg-error "duplicate asm operand name" } */
      ^
c_files/pr51768.c:16:7: error: 'asm goto' constructs are not supported yet
  asm goto ("" : : [lab] "i" (0) : : lab);      /* { dg-error "duplicate asm operand name" } */
      ^
2 errors generated.
TranslationUnitDecl 0x7fa7d901ded0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fa7d901e420 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fa7d901e140 '__int128'
|-TypedefDecl 0x7fa7d901e490 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fa7d901e160 'unsigned __int128'
|-TypedefDecl 0x7fa7d901e768 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fa7d901e570 'struct __NSConstantString_tag'
|   `-Record 0x7fa7d901e4e8 '__NSConstantString_tag'
|-TypedefDecl 0x7fa7d901e800 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fa7d901e7c0 'char *'
|   `-BuiltinType 0x7fa7d901df60 'char'
|-TypedefDecl 0x7fa7d901eac8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fa7d901ea70 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fa7d901e8e0 'struct __va_list_tag'
|     `-Record 0x7fa7d901e858 '__va_list_tag'
|-FunctionDecl 0x7fa7d906ee00 <c_files/pr51768.c:5:1, line:11:1> line:6:1 foo 'void (void)'
| `-CompoundStmt 0x7fa7d906efb8 <line:7:1, line:11:1>
|   |-LabelStmt 0x7fa7d906ef30 <line:9:1, col:5> 'lab'
|   | `-NullStmt 0x7fa7d906eed8 <col:5>
|   `-LabelStmt 0x7fa7d906efa0 <line:10:1, col:6> 'lab2'
|     `-NullStmt 0x7fa7d906ef48 <col:6>
|-FunctionDecl 0x7fa7d906f070 <line:13:1, line:18:1> line:14:1 bar 'void (void)'
| `-CompoundStmt 0x7fa7d906f178 <line:15:1, line:18:1>
|   `-LabelStmt 0x7fa7d906f160 <line:17:1, col:5> 'lab'
|     `-NullStmt 0x7fa7d906f108 <col:5>
`-FunctionDecl 0x7fa7d906f228 <line:20:1, line:25:1> line:21:1 baz 'void (void)'
  `-CompoundStmt 0x7fa7d906f560 <line:22:1, line:25:1>
    |-DeclStmt 0x7fa7d906f338 <line:23:3, col:8>
    | `-VarDecl 0x7fa7d906f2d8 <col:3, col:7> col:7 used x 'int'
    `-GCCAsmStmt 0x7fa7d906f4e8 <line:24:3, col:43>
      |-DeclRefExpr 0x7fa7d906f418 <col:25> 'int' lvalue Var 0x7fa7d906f2d8 'x' 'int'
      `-ImplicitCastExpr 0x7fa7d906f4d0 <col:41> 'int' <LValueToRValue>
        `-DeclRefExpr 0x7fa7d906f4a8 <col:41> 'int' lvalue Var 0x7fa7d906f2d8 'x' 'int'
