c_files/pr56989.c:11:19: error: invalid operands to binary expression ('void' and 'int')
      || voidf () < 0) /* { dg-error "10:void value not ignored as it ought to be" } */
         ~~~~~~~~ ^ ~
c_files/pr56989.c:14:16: error: invalid operands to binary expression ('void' and 'int')
  if (voidf () < 0 /* { dg-error "7:void value not ignored as it ought to be" } */
      ~~~~~~~~ ^ ~
2 errors generated.
TranslationUnitDecl 0x7f80dc0264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7f80dc026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7f80dc026740 '__int128'
|-TypedefDecl 0x7f80dc026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7f80dc026760 'unsigned __int128'
|-TypedefDecl 0x7f80dc026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7f80dc026b70 'struct __NSConstantString_tag'
|   `-Record 0x7f80dc026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7f80dc026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7f80dc026dc0 'char *'
|   `-BuiltinType 0x7f80dc026560 'char'
|-TypedefDecl 0x7f80dc0270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7f80dc027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7f80dc026ee0 'struct __va_list_tag'
|     `-Record 0x7f80dc026e58 '__va_list_tag'
|-FunctionDecl 0x7f80dc077400 <c_files/pr56989.c:4:1, col:24> col:13 used voidf 'void (void)' extern
|-FunctionDecl 0x7f80dc0775a8 <line:5:1, col:22> col:12 used intf 'int (void)' extern
`-FunctionDecl 0x7f80dc0776d8 <line:7:1, line:19:1> line:8:1 f 'int (void)'
  `-CompoundStmt 0x7f80dc077f50 <line:9:1, line:19:1>
    |-IfStmt 0x7f80dc077d18 <line:10:3, line:12:12>
    | |-<<<NULL>>>
    | |-<<<NULL>>>
    | |-OpaqueValueExpr 0x7f80dc077cf8 <<invalid sloc>> '_Bool'
    | |-ReturnStmt 0x7f80dc077ce0 <col:5, col:12>
    | | `-IntegerLiteral 0x7f80dc077cc0 <col:12> 'int' 1
    | `-<<<NULL>>>
    |-IfStmt 0x7f80dc077ee0 <line:14:3, line:16:12>
    | |-<<<NULL>>>
    | |-<<<NULL>>>
    | |-OpaqueValueExpr 0x7f80dc077ec0 <<invalid sloc>> '_Bool'
    | |-ReturnStmt 0x7f80dc077ea8 <col:5, col:12>
    | | `-IntegerLiteral 0x7f80dc077e88 <col:12> 'int' 1
    | `-<<<NULL>>>
    `-ReturnStmt 0x7f80dc077f38 <line:18:3, col:10>
      `-IntegerLiteral 0x7f80dc077f18 <col:10> 'int' 0
