c_files/pr60570.c:15:8: error: division by zero in preprocessor expression
#elif 1/0       /* { dg-error "division by zero" } */
      ~^~
c_files/pr60570.c:20:47: error: expected value in expression
#elif           /* { dg-error "with no expression" } */
                                                       ^
c_files/pr60570.c:40:9: error: division by zero in preprocessor expression
# elif 1/0      /* { dg-error "division by zero" } */
       ~^~
c_files/pr60570.c:46:48: error: expected value in expression
# elif          /* { dg-error "with no expression" } */
                                                       ^
4 errors generated.
TranslationUnitDecl 0x7fa03f0264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fa03f026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fa03f026740 '__int128'
|-TypedefDecl 0x7fa03f026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fa03f026760 'unsigned __int128'
|-TypedefDecl 0x7fa03f026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fa03f026b70 'struct __NSConstantString_tag'
|   `-Record 0x7fa03f026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fa03f026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fa03f026dc0 'char *'
|   `-BuiltinType 0x7fa03f026560 'char'
|-TypedefDecl 0x7fa03f0270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fa03f027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fa03f026ee0 'struct __va_list_tag'
|     `-Record 0x7fa03f026e58 '__va_list_tag'
|-VarDecl 0x7fa03f027138 <c_files/pr60570.c:5:1, col:5> col:5 i 'int'
|-VarDecl 0x7fa03f825a00 <line:10:1, col:5> col:5 j 'int'
|-VarDecl 0x7fa03f825a78 <line:26:1, col:5> col:5 l 'int'
`-VarDecl 0x7fa03f825af0 prev 0x7fa03f825a78 <line:33:1, col:5> col:5 l 'int'
