c_files/pr64744-1.c:8:3: error: non-ASM statement in naked function is not supported
  int aa = 0;
  ^
c_files/pr64744-1.c:5:16: note: attribute is here
__attribute__((naked))
               ^
c_files/pr64744-1.c:14:3: error: non-ASM statement in naked function is not supported
  char aa [ ] = {}; /* { dg-error "cannot allocate stack for variable" } */
  ^
c_files/pr64744-1.c:12:16: note: attribute is here
__attribute__((naked))
               ^
c_files/pr64744-1.c:22:3: error: non-ASM statement in naked function is not supported
  char aa [1] = {0};
  ^
c_files/pr64744-1.c:20:16: note: attribute is here
__attribute__((naked))
               ^
c_files/pr64744-1.c:30:3: error: non-ASM statement in naked function is not supported
  char aa [2] = {0,0}; /* { dg-error "cannot allocate stack for variable" } */
  ^
c_files/pr64744-1.c:28:16: note: attribute is here
__attribute__((naked))
               ^
c_files/pr64744-1.c:34:3: error: non-ASM statement in naked function is not supported
  char aa [3] = {0,0,0}; /* { dg-error "cannot allocate stack for variable" } */
  ^
c_files/pr64744-1.c:32:16: note: attribute is here
__attribute__((naked))
               ^
c_files/pr64744-1.c:39:3: error: non-ASM statement in naked function is not supported
  char aa [4] = {0,0,0,0}; /* { dg-error "cannot allocate stack for variable" } */
  ^
c_files/pr64744-1.c:37:16: note: attribute is here
__attribute__((naked))
               ^
6 errors generated.
TranslationUnitDecl 0x7fc67d8264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fc67d826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fc67d826740 '__int128'
|-TypedefDecl 0x7fc67d826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fc67d826760 'unsigned __int128'
|-TypedefDecl 0x7fc67d826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fc67d826b70 'struct __NSConstantString_tag'
|   `-Record 0x7fc67d826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fc67d826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fc67d826dc0 'char *'
|   `-BuiltinType 0x7fc67d826560 'char'
|-TypedefDecl 0x7fc67d8270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fc67d827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fc67d826ee0 'struct __va_list_tag'
|     `-Record 0x7fc67d826e58 '__va_list_tag'
|-FunctionDecl 0x7fc67d827168 <c_files/pr64744-1.c:5:1, line:10:1> line:6:6 invalid foo1 'void ()'
| |-CompoundStmt 0x7fc67e026c78 <line:7:1, line:10:1>
| | |-DeclStmt 0x7fc67e026b20 <line:8:3, col:13>
| | | `-VarDecl 0x7fc67e026aa0 <col:3, col:12> col:7 aa 'int' cinit
| | |   `-IntegerLiteral 0x7fc67e026b00 <col:12> 'int' 0
| | `-DeclStmt 0x7fc67e026c60 <line:9:3, col:15>
| |   `-VarDecl 0x7fc67e026b50 <col:3, col:14> col:7 ab 'int' cinit
| |     `-InitListExpr 0x7fc67e026c18 <col:12, col:14> 'int'
| |       `-IntegerLiteral 0x7fc67e026bb0 <col:13> 'int' 0
| `-NakedAttr 0x7fc67e026a00 <line:5:16>
|-FunctionDecl 0x7fc67e026cc0 <line:12:1, line:18:1> line:13:6 invalid foo2 'void ()'
| |-CompoundStmt 0x7fc67e0273a0 <col:13, line:18:1>
| | |-DeclStmt 0x7fc67e026f08 <line:14:3, col:19>
| | | `-VarDecl 0x7fc67e026de8 <col:3, col:18> col:8 aa 'char [0]' cinit
| | |   `-InitListExpr 0x7fc67e026e88 <col:17, col:18> 'char [0]'
| | |-DeclStmt 0x7fc67e027088 <line:15:3, col:19>
| | | `-VarDecl 0x7fc67e026f98 <col:3, col:18> col:8 ab 'char [1]' cinit
| | |   `-InitListExpr 0x7fc67e027038 <col:17, col:18> 'char [1]'
| | |     `-array filler
| | |       `-ImplicitValueInitExpr 0x7fc67e027078 <<invalid sloc>> 'char'
| | |-DeclStmt 0x7fc67e027208 <line:16:3, col:19>
| | | `-VarDecl 0x7fc67e027118 <col:3, col:18> col:8 ac 'char [2]' cinit
| | |   `-InitListExpr 0x7fc67e0271b8 <col:17, col:18> 'char [2]'
| | |     `-array filler
| | |       `-ImplicitValueInitExpr 0x7fc67e0271f8 <<invalid sloc>> 'char'
| | `-DeclStmt 0x7fc67e027388 <line:17:3, col:19>
| |   `-VarDecl 0x7fc67e027298 <col:3, col:18> col:8 ad 'char [3]' cinit
| |     `-InitListExpr 0x7fc67e027338 <col:17, col:18> 'char [3]'
| |       `-array filler
| |         `-ImplicitValueInitExpr 0x7fc67e027378 <<invalid sloc>> 'char'
| `-NakedAttr 0x7fc67e026d58 <line:12:16>
|-FunctionDecl 0x7fc67e0273f8 <line:20:1, line:26:1> line:21:6 invalid foo3 'void ()'
| |-CompoundStmt 0x7fc67e027b50 <col:13, line:26:1>
| | |-DeclStmt 0x7fc67e027640 <line:22:3, col:20>
| | | `-VarDecl 0x7fc67e027518 <col:3, col:19> col:8 aa 'char [1]' cinit
| | |   `-InitListExpr 0x7fc67e0275e0 <col:17, col:19> 'char [1]'
| | |     `-ImplicitCastExpr 0x7fc67e027628 <col:18> 'char' <IntegralCast>
| | |       `-IntegerLiteral 0x7fc67e027578 <col:18> 'int' 0
| | |-DeclStmt 0x7fc67e0277d0 <line:23:3, col:20>
| | | `-VarDecl 0x7fc67e027698 <col:3, col:19> col:8 ab 'char [2]' cinit
| | |   `-InitListExpr 0x7fc67e027760 <col:17, col:19> 'char [2]'
| | |     |-array filler
| | |     | `-ImplicitValueInitExpr 0x7fc67e0277c0 <<invalid sloc>> 'char'
| | |     `-ImplicitCastExpr 0x7fc67e0277a0 <col:18> 'char' <IntegralCast>
| | |       `-IntegerLiteral 0x7fc67e0276f8 <col:18> 'int' 0
| | |-DeclStmt 0x7fc67e027960 <line:24:3, col:20>
| | | `-VarDecl 0x7fc67e027828 <col:3, col:19> col:8 ac 'char [3]' cinit
| | |   `-InitListExpr 0x7fc67e0278f0 <col:17, col:19> 'char [3]'
| | |     |-array filler
| | |     | `-ImplicitValueInitExpr 0x7fc67e027950 <<invalid sloc>> 'char'
| | |     `-ImplicitCastExpr 0x7fc67e027930 <col:18> 'char' <IntegralCast>
| | |       `-IntegerLiteral 0x7fc67e027888 <col:18> 'int' 0
| | `-DeclStmt 0x7fc67e027b38 <line:25:3, col:20>
| |   `-VarDecl 0x7fc67e027a00 <col:3, col:19> col:8 ad 'char [4]' cinit
| |     `-InitListExpr 0x7fc67e027ac8 <col:17, col:19> 'char [4]'
| |       |-array filler
| |       | `-ImplicitValueInitExpr 0x7fc67e027b28 <<invalid sloc>> 'char'
| |       `-ImplicitCastExpr 0x7fc67e027b08 <col:18> 'char' <IntegralCast>
| |         `-IntegerLiteral 0x7fc67e027a60 <col:18> 'int' 0
| `-NakedAttr 0x7fc67e027490 <line:20:16>
|-FunctionDecl 0x7fc67e027ba8 <line:28:1, line:31:1> line:29:6 invalid foo4 'void ()'
| |-CompoundStmt 0x7fc67e027e50 <col:13, line:31:1>
| | `-DeclStmt 0x7fc67e027e38 <line:30:3, col:22>
| |   `-VarDecl 0x7fc67e027cc8 <col:3, col:21> col:8 aa 'char [2]' cinit
| |     `-InitListExpr 0x7fc67e027db8 <col:17, col:21> 'char [2]'
| |       |-ImplicitCastExpr 0x7fc67e027e08 <col:18> 'char' <IntegralCast>
| |       | `-IntegerLiteral 0x7fc67e027d28 <col:18> 'int' 0
| |       `-ImplicitCastExpr 0x7fc67e027e20 <col:20> 'char' <IntegralCast>
| |         `-IntegerLiteral 0x7fc67e027d48 <col:20> 'int' 0
| `-NakedAttr 0x7fc67e027c40 <line:28:16>
|-FunctionDecl 0x7fc67e027e90 <line:32:1, line:35:1> line:33:6 invalid foo5 'void ()'
| |-CompoundStmt 0x7fc67e028180 <col:13, line:35:1>
| | `-DeclStmt 0x7fc67e028168 <line:34:3, col:24>
| |   `-VarDecl 0x7fc67e027fb0 <col:3, col:23> col:8 aa 'char [3]' cinit
| |     `-InitListExpr 0x7fc67e0280c8 <col:17, col:23> 'char [3]'
| |       |-ImplicitCastExpr 0x7fc67e028120 <col:18> 'char' <IntegralCast>
| |       | `-IntegerLiteral 0x7fc67e028010 <col:18> 'int' 0
| |       |-ImplicitCastExpr 0x7fc67e028138 <col:20> 'char' <IntegralCast>
| |       | `-IntegerLiteral 0x7fc67e028030 <col:20> 'int' 0
| |       `-ImplicitCastExpr 0x7fc67e028150 <col:22> 'char' <IntegralCast>
| |         `-IntegerLiteral 0x7fc67e028050 <col:22> 'int' 0
| `-NakedAttr 0x7fc67e027f28 <line:32:16>
`-FunctionDecl 0x7fc67e0281c0 <line:37:1, line:40:1> line:38:6 invalid foo6 'void ()'
  |-CompoundStmt 0x7fc67e0284f8 <col:13, line:40:1>
  | `-DeclStmt 0x7fc67e0284e0 <line:39:3, col:26>
  |   `-VarDecl 0x7fc67e0282e0 <col:3, col:25> col:8 aa 'char [4]' cinit
  |     `-InitListExpr 0x7fc67e028420 <col:17, col:25> 'char [4]'
  |       |-ImplicitCastExpr 0x7fc67e028480 <col:18> 'char' <IntegralCast>
  |       | `-IntegerLiteral 0x7fc67e028340 <col:18> 'int' 0
  |       |-ImplicitCastExpr 0x7fc67e028498 <col:20> 'char' <IntegralCast>
  |       | `-IntegerLiteral 0x7fc67e028360 <col:20> 'int' 0
  |       |-ImplicitCastExpr 0x7fc67e0284b0 <col:22> 'char' <IntegralCast>
  |       | `-IntegerLiteral 0x7fc67e028380 <col:22> 'int' 0
  |       `-ImplicitCastExpr 0x7fc67e0284c8 <col:24> 'char' <IntegralCast>
  |         `-IntegerLiteral 0x7fc67e0283a0 <col:24> 'int' 0
  `-NakedAttr 0x7fc67e028258 <line:37:16>
