c_files/pr65050.c:5:22: error: array has incomplete element type 'A' (aka 'int []')
struct S { int i; A a[5]; } s; /* { dg-error "array type has incomplete element type 'A' {aka 'int\\\[\\\]'}" } */
                     ^
c_files/pr65050.c:6:23: error: array has incomplete element type 'int []'
extern void foo (int p[2][]); /* { dg-error "array type has incomplete element type .int\\\[\\\]." } */
                      ^
c_files/pr65050.c:7:21: error: array has incomplete element type 'A' (aka 'int []')
extern void bar (A p[2]); /* { dg-error "array type has incomplete element type 'A' {aka 'int\\\[\\\]'}" } */
                    ^
c_files/pr65050.c:10:11: error: array has incomplete element type 'int []'
foo (int p[2][]) /* { dg-error "array type has incomplete element type .int\\\[\\\]." } */
          ^
c_files/pr65050.c:15:9: error: array has incomplete element type 'A' (aka 'int []')
bar (A p[2]) /* { dg-error "array type has incomplete element type 'A' {aka 'int\\\[\\\]'}" } */
        ^
c_files/pr65050.c:20:11: error: array has incomplete element type 'struct T'
struct T t[5]; /* { dg-error "array type has incomplete element type .struct T." } */
          ^
c_files/pr65050.c:19:8: note: forward declaration of 'struct T'
struct T;
       ^
c_files/pr65050.c:21:11: error: array has incomplete element type 'struct U'
struct U u[] = { { "abc" } }; /* { dg-error "array type has incomplete element type .struct U." } */
          ^
c_files/pr65050.c:21:8: note: forward declaration of 'struct U'
struct U u[] = { { "abc" } }; /* { dg-error "array type has incomplete element type .struct U." } */
       ^
c_files/pr65050.c:23:6: error: array has incomplete element type 'TT' (aka 'struct T')
TT tt[5]; /* { dg-error "array type has incomplete element type 'TT' {aka 'struct T'}" } */
     ^
c_files/pr65050.c:19:8: note: forward declaration of 'struct T'
struct T;
       ^
8 errors generated.
TranslationUnitDecl 0x7fc23f0264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fc23f026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fc23f026740 '__int128'
|-TypedefDecl 0x7fc23f026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fc23f026760 'unsigned __int128'
|-TypedefDecl 0x7fc23f026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fc23f026b70 'struct __NSConstantString_tag'
|   `-Record 0x7fc23f026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fc23f026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fc23f026dc0 'char *'
|   `-BuiltinType 0x7fc23f026560 'char'
|-TypedefDecl 0x7fc23f0270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fc23f027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fc23f026ee0 'struct __va_list_tag'
|     `-Record 0x7fc23f026e58 '__va_list_tag'
|-TypedefDecl 0x7fc23f027170 <c_files/pr65050.c:4:1, col:15> col:13 referenced A 'int []'
| `-IncompleteArrayType 0x7fc23f027120 'int []' 
|   `-BuiltinType 0x7fc23f0265c0 'int'
|-RecordDecl 0x7fc23f825a00 <line:5:1, col:27> col:8 invalid struct S definition
| |-FieldDecl 0x7fc23f825ac8 <col:12, col:16> col:16 i 'int'
| `-FieldDecl 0x7fc23f825f18 <col:19, col:21> col:21 invalid a 'int'
|-VarDecl 0x7fc23f825fb0 <col:1, col:29> col:29 s 'struct S':'struct S'
|-FunctionDecl 0x7fc23f826160 <line:6:1, col:28> col:13 invalid foo 'void (int)' extern
| `-ParmVarDecl 0x7fc23f8260a0 <col:18, col:22> col:22 invalid p 'int'
|-FunctionDecl 0x7fc23f8262c0 <line:7:1, col:24> col:13 invalid bar 'void (int)' extern
| `-ParmVarDecl 0x7fc23f826238 <col:18, col:20> col:20 invalid p 'int'
|-FunctionDecl 0x7fc23f826420 <line:9:1, line:12:1> line:10:1 invalid foo 'void (int)'
| |-ParmVarDecl 0x7fc23f826398 <col:6, col:10> col:10 invalid p 'int'
| `-CompoundStmt 0x7fc23f8264c0 <line:11:1, line:12:1>
|-FunctionDecl 0x7fc23f826598 <line:14:1, line:17:1> line:15:1 invalid bar 'void (int)'
| |-ParmVarDecl 0x7fc23f826510 <col:6, col:8> col:8 invalid p 'int'
| `-CompoundStmt 0x7fc23f826638 <line:16:1, line:17:1>
|-RecordDecl 0x7fc23f826650 <line:19:1, col:8> col:8 struct T
|-VarDecl 0x7fc23f826768 <line:20:1, col:10> col:10 invalid t 'int'
|-RecordDecl 0x7fc23f8267c8 <line:21:1, col:8> col:8 struct U
|-VarDecl 0x7fc23f8268b8 <col:1, col:10> col:10 invalid u 'int'
|-TypedefDecl 0x7fc23f82bc80 <line:22:1, col:18> col:18 referenced TT 'struct T':'struct T'
| `-ElaboratedType 0x7fc23f826720 'struct T' sugar
|   `-RecordType 0x7fc23f8266e0 'struct T'
|     `-Record 0x7fc23f826650 'T'
`-VarDecl 0x7fc23f82bd38 <line:23:1, col:4> col:4 invalid tt 'int'
