c_files/pr67784-2.c:10:20: error: declaration of non-local variable in 'for' loop
  for (typedef int T;;) /* { dg-error "declaration of non-variable" } */
                   ^
c_files/pr67784-2.c:13:6: error: use of undeclared identifier 'x'
  T *x; /* { dg-error "undeclared" } */
     ^
c_files/pr67784-2.c:21:7: error: expected expression
      T = 1; /* { dg-error "expected expression" } */
      ^
c_files/pr67784-2.c:22:6: error: use of undeclared identifier 'x'
  T *x; /* { dg-error "undeclared" } */
     ^
c_files/pr67784-2.c:28:20: error: declaration of non-local variable in 'for' loop
  for (typedef int T;;) /* { dg-error "declaration of non-variable" } */
                   ^
c_files/pr67784-2.c:32:6: error: use of undeclared identifier 'x'
  T *x; /* { dg-error "undeclared" } */
     ^
c_files/pr67784-2.c:38:20: error: declaration of non-local variable in 'for' loop
  for (typedef int T;;) /* { dg-error "declaration of non-variable" } */
                   ^
c_files/pr67784-2.c:42:6: error: use of undeclared identifier 'x'
  T *x; /* { dg-error "undeclared" } */
     ^
c_files/pr67784-2.c:48:20: error: declaration of non-local variable in 'for' loop
  for (typedef int T;;) /* { dg-error "declaration of non-variable" } */
                   ^
c_files/pr67784-2.c:53:6: error: use of undeclared identifier 'x'
  T *x; /* { dg-error "undeclared" } */
     ^
10 errors generated.
TranslationUnitDecl 0x7fe56a0264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fe56a026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fe56a026740 '__int128'
|-TypedefDecl 0x7fe56a026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fe56a026760 'unsigned __int128'
|-TypedefDecl 0x7fe56a026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fe56a026b70 'struct __NSConstantString_tag'
|   `-Record 0x7fe56a026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fe56a026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fe56a026dc0 'char *'
|   `-BuiltinType 0x7fe56a026560 'char'
|-TypedefDecl 0x7fe56a0270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fe56a027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fe56a026ee0 'struct __va_list_tag'
|     `-Record 0x7fe56a026e58 '__va_list_tag'
|-VarDecl 0x7fe56a027138 <c_files/pr67784-2.c:5:1, col:5> col:5 used T 'int'
|-FunctionDecl 0x7fe56a0774b0 <line:7:1, line:14:1> line:8:1 fn1 'void (void)'
| `-CompoundStmt 0x7fe56a0776c0 <line:9:1, line:14:1>
|   `-ForStmt 0x7fe56a077638 <line:10:3, line:12:7>
|     |-DeclStmt 0x7fe56a0775b8 <line:10:8, col:21>
|     | `-TypedefDecl 0x7fe56a077560 <col:8, col:20> col:20 invalid T 'int'
|     |   `-BuiltinType 0x7fe56a0265c0 'int'
|     |-<<<NULL>>>
|     |-<<<NULL>>>
|     |-<<<NULL>>>
|     `-IfStmt 0x7fe56a077600 <line:11:5, line:12:7>
|       |-<<<NULL>>>
|       |-<<<NULL>>>
|       |-IntegerLiteral 0x7fe56a0775d0 <line:11:9> 'int' 1
|       |-NullStmt 0x7fe56a0775f0 <line:12:7>
|       `-<<<NULL>>>
|-FunctionDecl 0x7fe56a077770 <line:16:1, line:23:1> line:17:1 fn2 'void (void)'
| `-CompoundStmt 0x7fe56a077908 <line:18:1, line:23:1>
|-FunctionDecl 0x7fe56a0779b0 <line:25:1, line:33:1> line:26:1 fn3 'void (void)'
| `-CompoundStmt 0x7fe56a077bb0 <line:27:1, line:33:1>
|   `-ForStmt 0x7fe56a077b40 <line:28:3, line:31:7>
|     |-DeclStmt 0x7fe56a077ab8 <line:28:8, col:21>
|     | `-TypedefDecl 0x7fe56a077a60 <col:8, col:20> col:20 invalid T 'int'
|     |   `-BuiltinType 0x7fe56a0265c0 'int'
|     |-<<<NULL>>>
|     |-<<<NULL>>>
|     |-<<<NULL>>>
|     `-IfStmt 0x7fe56a077b08 <line:29:5, line:31:7>
|       |-<<<NULL>>>
|       |-<<<NULL>>>
|       |-IntegerLiteral 0x7fe56a077ad0 <line:29:9> 'int' 1
|       |-CompoundStmt 0x7fe56a077af0 <line:30:7, line:31:7>
|       `-<<<NULL>>>
|-FunctionDecl 0x7fe56a077c60 <line:35:1, line:43:1> line:36:1 fn4 'void (void)'
| `-CompoundStmt 0x7fe56a077eb8 <line:37:1, line:43:1>
|   `-ForStmt 0x7fe56a077e48 <line:38:3, line:41:7>
|     |-DeclStmt 0x7fe56a077d68 <line:38:8, col:21>
|     | `-TypedefDecl 0x7fe56a077d10 <col:8, col:20> col:20 invalid T 'int'
|     |   `-BuiltinType 0x7fe56a0265c0 'int'
|     |-<<<NULL>>>
|     |-<<<NULL>>>
|     |-<<<NULL>>>
|     `-IfStmt 0x7fe56a077e10 <line:39:5, line:41:7>
|       |-<<<NULL>>>
|       |-<<<NULL>>>
|       |-IntegerLiteral 0x7fe56a077d80 <line:39:9> 'int' 1
|       |-LabelStmt 0x7fe56a077df8 <line:40:1, line:41:7> 'L'
|       | `-NullStmt 0x7fe56a077da0 <col:7>
|       `-<<<NULL>>>
`-FunctionDecl 0x7fe56a077f68 <line:45:1, line:54:1> line:46:1 fn5 'void (void)'
  `-CompoundStmt 0x7fe56a078170 <line:47:1, line:54:1>
    `-ForStmt 0x7fe56a078100 <line:48:3, line:52:7>
      |-DeclStmt 0x7fe56a078070 <line:48:8, col:21>
      | `-TypedefDecl 0x7fe56a078018 <col:8, col:20> col:20 invalid T 'int'
      |   `-BuiltinType 0x7fe56a0265c0 'int'
      |-<<<NULL>>>
      |-<<<NULL>>>
      |-<<<NULL>>>
      `-IfStmt 0x7fe56a0780c8 <line:49:5, line:52:7>
        |-<<<NULL>>>
        |-<<<NULL>>>
        |-IntegerLiteral 0x7fe56a078088 <line:49:9> 'int' 1
        |-NullStmt 0x7fe56a0780a8 <line:50:7>
        `-NullStmt 0x7fe56a0780b8 <line:52:7>
