c_files/pr69796.c:9:4: error: expected ';' after expression
  s a;  /* { dg-error "expression statement has incomplete type|expected" } */
   ^
   ;
c_files/pr69796.c:9:3: error: incomplete type 'struct S' where a complete type is required
  s a;  /* { dg-error "expression statement has incomplete type|expected" } */
  ^
c_files/pr69796.c:4:8: note: forward declaration of 'struct S'
struct S s;     /* { dg-error "storage size of 's' isn't known" } */
       ^
c_files/pr69796.c:9:5: error: use of undeclared identifier 'a'
  s a;  /* { dg-error "expression statement has incomplete type|expected" } */
    ^
c_files/pr69796.c:9:3: warning: expression result unused [-Wunused-value]
  s a;  /* { dg-error "expression statement has incomplete type|expected" } */
  ^
c_files/pr69796.c:4:10: error: tentative definition has type 'struct S' that is never completed
struct S s;     /* { dg-error "storage size of 's' isn't known" } */
         ^
c_files/pr69796.c:4:8: note: forward declaration of 'struct S'
struct S s;     /* { dg-error "storage size of 's' isn't known" } */
       ^
1 warning and 4 errors generated.
TranslationUnitDecl 0x7fbae98264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fbae9826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fbae9826740 '__int128'
|-TypedefDecl 0x7fbae9826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fbae9826760 'unsigned __int128'
|-TypedefDecl 0x7fbae9826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fbae9826b70 'struct __NSConstantString_tag'
|   `-Record 0x7fbae9826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fbae9826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fbae9826dc0 'char *'
|   `-BuiltinType 0x7fbae9826560 'char'
|-TypedefDecl 0x7fbae98270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fbae9827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fbae9826ee0 'struct __va_list_tag'
|     `-Record 0x7fbae9826e58 '__va_list_tag'
|-RecordDecl 0x7fbae9827120 <c_files/pr69796.c:4:1, col:8> col:8 struct S
|-VarDecl 0x7fbaea025a20 <col:1, col:10> col:10 used invalid s 'struct S':'struct S'
`-FunctionDecl 0x7fbaea025b28 <line:6:1, line:10:1> line:7:1 foo 'void ()'
  `-CompoundStmt 0x7fbaea025fc8 <line:8:1, line:10:1>
    `-ImplicitCastExpr 0x7fbaea025be8 <line:9:3> 'struct S':'struct S' <LValueToRValue>
      `-DeclRefExpr 0x7fbaea025bc0 <col:3> 'struct S':'struct S' lvalue Var 0x7fbaea025a20 's' 'struct S':'struct S'
