c_files/pr71853.c:6:11: error: expected ':' after 'case'
  case (0) { /* { dg-error "expected" } */
          ^
          :
c_files/pr71853.c:6:3: error: 'case' statement not in switch statement
  case (0) { /* { dg-error "expected" } */
  ^
c_files/pr71853.c:7:12: error: expected '(' after 'switch'
    switch 0: { } /* { dg-error "expected" } */
           ^
3 errors generated.
TranslationUnitDecl 0x7f98868264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7f9886826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7f9886826740 '__int128'
|-TypedefDecl 0x7f9886826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7f9886826760 'unsigned __int128'
|-TypedefDecl 0x7f9886826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7f9886826b70 'struct __NSConstantString_tag'
|   `-Record 0x7f9886826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7f9886826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7f9886826dc0 'char *'
|   `-BuiltinType 0x7f9886826560 'char'
|-TypedefDecl 0x7f98868270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7f9886827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7f9886826ee0 'struct __va_list_tag'
|     `-Record 0x7f9886826e58 '__va_list_tag'
`-FunctionDecl 0x7f9887025a00 <c_files/pr71853.c:4:1, line:9:1> line:4:6 f 'void (void)'
  `-CompoundStmt 0x7f9887025b30 <line:5:1, line:9:1>
    `-CompoundStmt 0x7f9887025b18 <line:6:12, line:8:3>
