c_files/pr77624-2.c:8:32: error: too many arguments to function call, expected 2, have 3
  __atomic_is_lock_free (2, a, 2);      /* { dg-error "too many arguments" } */
  ~~~~~~~~~~~~~~~~~~~~~        ^
c_files/pr77624-2.c:9:27: error: too few arguments to function call, expected 2, have 1
  __atomic_is_lock_free (2);            /* { dg-error "too few arguments" } */
  ~~~~~~~~~~~~~~~~~~~~~   ^
c_files/pr77624-2.c:10:29: error: passing 'double' to parameter of incompatible type 'const volatile void *'
  __atomic_is_lock_free (2, b);         /* { dg-error "incompatible type" "" { target c } } */
                            ^
c_files/pr77624-2.c:20:36: error: too many arguments to function call, expected 2, have 3
  __atomic_always_lock_free (2, a, 2);  /* { dg-error "too many arguments" } */
  ~~~~~~~~~~~~~~~~~~~~~~~~~        ^
c_files/pr77624-2.c:21:31: error: too few arguments to function call, expected 2, have 1
  __atomic_always_lock_free (2);        /* { dg-error "too few arguments" } */
  ~~~~~~~~~~~~~~~~~~~~~~~~~   ^
c_files/pr77624-2.c:22:33: error: passing 'double' to parameter of incompatible type 'const volatile void *'
  __atomic_always_lock_free (2, b);     /* { dg-error "incompatible type" "" { target c } } */
                                ^
6 errors generated.
TranslationUnitDecl 0x7fce228264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fce22826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fce22826740 '__int128'
|-TypedefDecl 0x7fce22826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fce22826760 'unsigned __int128'
|-TypedefDecl 0x7fce22826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fce22826b70 'struct __NSConstantString_tag'
|   `-Record 0x7fce22826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fce22826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fce22826dc0 'char *'
|   `-BuiltinType 0x7fce22826560 'char'
|-TypedefDecl 0x7fce228270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fce22827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fce22826ee0 'struct __va_list_tag'
|     `-Record 0x7fce22826e58 '__va_list_tag'
|-FunctionDecl 0x7fce23025a28 <c_files/pr77624-2.c:4:1, line:14:1> line:5:1 foo 'void (int *)'
| |-ParmVarDecl 0x7fce22827160 <col:6, col:11> col:11 used a 'int *'
| `-CompoundStmt 0x7fce23026578 <line:6:1, line:14:1>
|   |-DeclStmt 0x7fce23025bb8 <line:7:3, col:15>
|   | `-VarDecl 0x7fce23025b20 <col:3, col:14> col:10 used b 'double' cinit
|   |   `-ImplicitCastExpr 0x7fce23025ba0 <col:14> 'double' <IntegralToFloating>
|   |     `-IntegerLiteral 0x7fce23025b80 <col:14> 'int' 0
|   `-CallExpr 0x7fce23026510 <line:13:3, col:30> 'int'
|     |-ImplicitCastExpr 0x7fce230264f8 <col:3> 'int (*)(unsigned long, const volatile void *)' <BuiltinFnToFnPtr>
|     | `-DeclRefExpr 0x7fce23026490 <col:3> '<builtin fn type>' Function 0x7fce23025c40 '__atomic_is_lock_free' 'int (unsigned long, const volatile void *)'
|     |-ImplicitCastExpr 0x7fce23026548 <col:26> 'unsigned long' <IntegralCast>
|     | `-IntegerLiteral 0x7fce230264b8 <col:26> 'int' 2
|     `-ImplicitCastExpr 0x7fce23026560 <col:29> 'const volatile void *' <NullToPointer>
|       `-IntegerLiteral 0x7fce230264d8 <col:29> 'int' 0
|-FunctionDecl 0x7fce23025c40 <line:8:3> col:3 implicit used __atomic_is_lock_free 'int (unsigned long, const volatile void *)' extern
| |-ParmVarDecl 0x7fce23025cd8 <<invalid sloc>> <invalid sloc> 'unsigned long'
| |-ParmVarDecl 0x7fce23025d38 <<invalid sloc>> <invalid sloc> 'const volatile void *'
| `-NoThrowAttr 0x7fce23025da8 <col:3> Implicit
|-FunctionDecl 0x7fce23026640 <line:16:1, line:26:1> line:17:1 bar 'void (int *)'
| |-ParmVarDecl 0x7fce230265b8 <col:6, col:11> col:11 used a 'int *'
| `-CompoundStmt 0x7fce23027cc0 <line:18:1, line:26:1>
|   |-DeclStmt 0x7fce23026790 <line:19:3, col:15>
|   | `-VarDecl 0x7fce230266f8 <col:3, col:14> col:10 used b 'double' cinit
|   |   `-ImplicitCastExpr 0x7fce23026778 <col:14> 'double' <IntegralToFloating>
|   |     `-IntegerLiteral 0x7fce23026758 <col:14> 'int' 0
|   `-CallExpr 0x7fce23027c58 <line:25:3, col:34> 'int'
|     |-ImplicitCastExpr 0x7fce23027c40 <col:3> 'int (*)(unsigned long, const volatile void *)' <BuiltinFnToFnPtr>
|     | `-DeclRefExpr 0x7fce23027bd8 <col:3> '<builtin fn type>' Function 0x7fce230267a8 '__atomic_always_lock_free' 'int (unsigned long, const volatile void *)'
|     |-ImplicitCastExpr 0x7fce23027c90 <col:30> 'unsigned long' <IntegralCast>
|     | `-IntegerLiteral 0x7fce23027c00 <col:30> 'int' 2
|     `-ImplicitCastExpr 0x7fce23027ca8 <col:33> 'const volatile void *' <NullToPointer>
|       `-IntegerLiteral 0x7fce23027c20 <col:33> 'int' 0
`-FunctionDecl 0x7fce230267a8 <line:20:3> col:3 implicit used __atomic_always_lock_free 'int (unsigned long, const volatile void *)' extern
  |-ParmVarDecl 0x7fce23026840 <<invalid sloc>> <invalid sloc> 'unsigned long'
  |-ParmVarDecl 0x7fce230268a0 <<invalid sloc>> <invalid sloc> 'const volatile void *'
  `-NoThrowAttr 0x7fce23026910 <col:3> Implicit
