c_files/pr81622.c:10:3: error: use of unknown builtin '__builtin_vec_ld' [-Wimplicit-function-declaration]
  __builtin_vec_ld (1, 2);      /* { dg-error "invalid parameter combination" } */
  ^
c_files/pr81622.c:11:3: error: use of unknown builtin '__builtin_vec_cmpne' [-Wimplicit-function-declaration]
  __builtin_vec_cmpne (1, 2);   /* { dg-error "invalid parameter combination" } */
  ^
c_files/pr81622.c:11:3: note: did you mean '__builtin_vec_ld'?
c_files/pr81622.c:10:3: note: '__builtin_vec_ld' declared here
  __builtin_vec_ld (1, 2);      /* { dg-error "invalid parameter combination" } */
  ^
c_files/pr81622.c:12:3: error: use of unknown builtin '__builtin_vec_st' [-Wimplicit-function-declaration]
  __builtin_vec_st (1, 0, 5);   /* { dg-error "invalid parameter combination" } */
  ^
c_files/pr81622.c:12:3: note: did you mean '__builtin_vec_ld'?
c_files/pr81622.c:10:3: note: '__builtin_vec_ld' declared here
  __builtin_vec_ld (1, 2);      /* { dg-error "invalid parameter combination" } */
  ^
3 errors generated.
TranslationUnitDecl 0x7fea5c8264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fea5c826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fea5c826740 '__int128'
|-TypedefDecl 0x7fea5c826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fea5c826760 'unsigned __int128'
|-TypedefDecl 0x7fea5c826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fea5c826b70 'struct __NSConstantString_tag'
|   `-Record 0x7fea5c826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fea5c826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fea5c826dc0 'char *'
|   `-BuiltinType 0x7fea5c826560 'char'
|-TypedefDecl 0x7fea5c8270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fea5c827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fea5c826ee0 'struct __va_list_tag'
|     `-Record 0x7fea5c826e58 '__va_list_tag'
|-FunctionDecl 0x7fea5c877400 <c_files/pr81622.c:7:1, line:13:1> line:8:1 foo 'void (void)'
| `-CompoundStmt 0x7fea5c8779d0 <line:9:1, line:13:1>
|   |-CallExpr 0x7fea5c877680 <line:10:3, col:25> 'int'
|   | |-ImplicitCastExpr 0x7fea5c877668 <col:3> 'int (*)()' <FunctionToPointerDecay>
|   | | `-DeclRefExpr 0x7fea5c8775d8 <col:3> 'int ()' Function 0x7fea5c877540 '__builtin_vec_ld' 'int ()'
|   | |-IntegerLiteral 0x7fea5c877600 <col:21> 'int' 1
|   | `-IntegerLiteral 0x7fea5c877620 <col:24> 'int' 2
|   |-CallExpr 0x7fea5c8777f8 <line:11:3, col:28> 'int'
|   | |-ImplicitCastExpr 0x7fea5c8777e0 <col:3> 'int (*)()' <FunctionToPointerDecay>
|   | | `-DeclRefExpr 0x7fea5c877778 <col:3> 'int ()' Function 0x7fea5c8776e0 '__builtin_vec_cmpne' 'int ()'
|   | |-IntegerLiteral 0x7fea5c8777a0 <col:24> 'int' 1
|   | `-IntegerLiteral 0x7fea5c8777c0 <col:27> 'int' 2
|   `-CallExpr 0x7fea5c877990 <line:12:3, col:28> 'int'
|     |-ImplicitCastExpr 0x7fea5c877978 <col:3> 'int (*)()' <FunctionToPointerDecay>
|     | `-DeclRefExpr 0x7fea5c8778f0 <col:3> 'int ()' Function 0x7fea5c877858 '__builtin_vec_st' 'int ()'
|     |-IntegerLiteral 0x7fea5c877918 <col:21> 'int' 1
|     |-IntegerLiteral 0x7fea5c877938 <col:24> 'int' 0
|     `-IntegerLiteral 0x7fea5c877958 <col:27> 'int' 5
|-FunctionDecl 0x7fea5c877540 <<invalid sloc>> line:10:3 implicit used __builtin_vec_ld 'int ()'
|-FunctionDecl 0x7fea5c8776e0 <<invalid sloc>> line:11:3 implicit used __builtin_vec_cmpne 'int ()'
`-FunctionDecl 0x7fea5c877858 <<invalid sloc>> line:12:3 implicit used __builtin_vec_st 'int ()'
