c_files/pr84721.c:5:7: error: statement expression not allowed at file scope
int a[({ int b })];             /* { dg-error "braced-group within expression allowed only inside a function" } */
      ^
c_files/pr84721.c:6:7: error: statement expression not allowed at file scope
int c[({ int d () {}; })];      /* { dg-error "braced-group within expression allowed only inside a function" } */
      ^
2 errors generated.
TranslationUnitDecl 0x7fca6e0166d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fca6e016c20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fca6e016940 '__int128'
|-TypedefDecl 0x7fca6e016c90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fca6e016960 'unsigned __int128'
|-TypedefDecl 0x7fca6e016f68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fca6e016d70 'struct __NSConstantString_tag'
|   `-Record 0x7fca6e016ce8 '__NSConstantString_tag'
|-TypedefDecl 0x7fca6e017000 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fca6e016fc0 'char *'
|   `-BuiltinType 0x7fca6e016760 'char'
|-TypedefDecl 0x7fca6e0172c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fca6e017270 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fca6e0170e0 'struct __va_list_tag'
|     `-Record 0x7fca6e017058 '__va_list_tag'
|-VarDecl 0x7fca6e017338 <c_files/pr84721.c:5:1, col:5> col:5 invalid a 'int'
`-VarDecl 0x7fca6e825a00 <line:6:1, col:5> col:5 invalid c 'int'
