c_files/pr12625-1.c:12:11: warning: using the result of an assignment as a condition without parentheses [-Wparentheses]
  if (s.i = 0) ; /* { dg-warning "parentheses" "warning for bit-field" } */
      ~~~~^~~
c_files/pr12625-1.c:12:11: note: place parentheses around the assignment to silence this warning
  if (s.i = 0) ; /* { dg-warning "parentheses" "warning for bit-field" } */
          ^
      (      )
c_files/pr12625-1.c:12:11: note: use '==' to turn this assignment into an equality comparison
  if (s.i = 0) ; /* { dg-warning "parentheses" "warning for bit-field" } */
          ^
          ==
c_files/pr12625-1.c:12:16: warning: if statement has empty body [-Wempty-body]
  if (s.i = 0) ; /* { dg-warning "parentheses" "warning for bit-field" } */
               ^
c_files/pr12625-1.c:12:16: note: put the semicolon on a separate line to silence this warning
2 warnings generated.
TranslationUnitDecl 0x7ff80501c0d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7ff80501c620 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7ff80501c340 '__int128'
|-TypedefDecl 0x7ff80501c690 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7ff80501c360 'unsigned __int128'
|-TypedefDecl 0x7ff80501c968 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7ff80501c770 'struct __NSConstantString_tag'
|   `-Record 0x7ff80501c6e8 '__NSConstantString_tag'
|-TypedefDecl 0x7ff80501ca00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7ff80501c9c0 'char *'
|   `-BuiltinType 0x7ff80501c160 'char'
|-TypedefDecl 0x7ff80501ccc8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7ff80501cc70 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7ff80501cae0 'struct __va_list_tag'
|     `-Record 0x7ff80501ca58 '__va_list_tag'
|-RecordDecl 0x7ff80501cd20 <c_files/pr12625-1.c:7:8, col:26> col:8 struct definition
| `-FieldDecl 0x7ff80400b418 <col:17, col:23> col:21 referenced i 'int'
|   `-IntegerLiteral 0x7ff80501cdd0 <col:23> 'int' 8
|-VarDecl 0x7ff80400b4b0 <col:1, col:28> col:28 used s 'struct (anonymous struct at c_files/pr12625-1.c:7:8)':'struct (anonymous at c_files/pr12625-1.c:7:8)' static
`-FunctionDecl 0x7ff80400b5b8 <line:9:1, line:13:1> line:10:1 foo 'void ()'
  `-CompoundStmt 0x7ff80400b788 <line:11:1, line:13:1>
    `-IfStmt 0x7ff80400b750 <line:12:3, col:16>
      |-<<<NULL>>>
      |-<<<NULL>>>
      |-BinaryOperator 0x7ff80400b6d0 <col:7, col:13> 'int' '='
      | |-MemberExpr 0x7ff80400b678 <col:7, col:9> 'int' lvalue bitfield .i 0x7ff80400b418
      | | `-DeclRefExpr 0x7ff80400b650 <col:7> 'struct (anonymous struct at c_files/pr12625-1.c:7:8)':'struct (anonymous at c_files/pr12625-1.c:7:8)' lvalue Var 0x7ff80400b4b0 's' 'struct (anonymous struct at c_files/pr12625-1.c:7:8)':'struct (anonymous at c_files/pr12625-1.c:7:8)'
      | `-IntegerLiteral 0x7ff80400b6b0 <col:13> 'int' 0
      |-NullStmt 0x7ff80400b740 <col:16>
      `-<<<NULL>>>
