c_files/pr15443-2.c:8:27: warning: '__malloc__' attribute only applies to return values that are pointers [-Wignored-attributes]
void f () __attribute__ ((__malloc__)); /* { dg-warning "ignored" } */
~~~~                      ^
c_files/pr15443-2.c:9:26: warning: '__malloc__' attribute only applies to return values that are pointers [-Wignored-attributes]
int g () __attribute__ ((__malloc__)); /* { dg-warning "ignored" } */
~~~                      ^
c_files/pr15443-2.c:13:33: warning: '__malloc__' attribute only applies to return values that are pointers [-Wignored-attributes]
struct foo j () __attribute__ ((__malloc__)); /* { dg-warning "ignored" } */
~~~~~~~~~~                      ^
3 warnings generated.
TranslationUnitDecl 0x7fbf5c023ed0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fbf5c024420 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fbf5c024140 '__int128'
|-TypedefDecl 0x7fbf5c024490 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fbf5c024160 'unsigned __int128'
|-TypedefDecl 0x7fbf5c024768 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fbf5c024570 'struct __NSConstantString_tag'
|   `-Record 0x7fbf5c0244e8 '__NSConstantString_tag'
|-TypedefDecl 0x7fbf5c024800 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fbf5c0247c0 'char *'
|   `-BuiltinType 0x7fbf5c023f60 'char'
|-TypedefDecl 0x7fbf5c024ac8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fbf5c024a70 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fbf5c0248e0 'struct __va_list_tag'
|     `-Record 0x7fbf5c024858 '__va_list_tag'
|-RecordDecl 0x7fbf5c024b20 <c_files/pr15443-2.c:3:1, line:5:1> line:3:8 struct foo definition
| `-FieldDecl 0x7fbf5d025a00 <line:4:2, col:6> col:6 bar 'int'
|-TypedefDecl 0x7fbf5d025b00 <line:7:1, col:21> col:21 referenced bar 'struct foo *'
| `-PointerType 0x7fbf5d025ab0 'struct foo *'
|   `-ElaboratedType 0x7fbf5d025a50 'struct foo' sugar
|     `-RecordType 0x7fbf5c024bb0 'struct foo'
|       `-Record 0x7fbf5c024b20 'foo'
|-FunctionDecl 0x7fbf5d025ba8 <line:8:1, col:38> col:6 f 'void ()'
|-FunctionDecl 0x7fbf5d025cd0 <line:9:1, col:37> col:5 g 'int ()'
|-FunctionDecl 0x7fbf5d025df0 <line:10:1, col:38> col:6 h 'int *()'
| `-RestrictAttr 0x7fbf5d025e88 <col:27> malloc
|-FunctionDecl 0x7fbf5d025f20 <line:11:1, col:39> col:7 i 'void *()'
| `-RestrictAttr 0x7fbf5d025fb8 <col:28> malloc
|-FunctionDecl 0x7fbf5d026088 <line:13:1, col:44> col:12 j 'struct foo ()'
|-FunctionDecl 0x7fbf5d0261b0 <line:14:1, col:45> col:13 k 'struct foo *()'
| `-RestrictAttr 0x7fbf5d026248 <col:34> malloc
`-FunctionDecl 0x7fbf5d0262f8 <line:15:1, col:32> col:5 l 'bar ()'
  `-RestrictAttr 0x7fbf5d026390 <col:25> malloc
