c_files/pr24293.c:4:19: warning: tentative definition of variable with internal linkage has incomplete non-array type 'struct foo' [-Wtentative-definition-incomplete-type]
static struct foo x;  /* { dg-error "storage size of 'x' isn't known" } */
                  ^
c_files/pr24293.c:4:15: note: forward declaration of 'struct foo'
static struct foo x;  /* { dg-error "storage size of 'x' isn't known" } */
              ^
c_files/pr24293.c:5:18: warning: tentative definition of variable with internal linkage has incomplete non-array type 'union bar' [-Wtentative-definition-incomplete-type]
static union bar y;  /* { dg-error "storage size of 'y' isn't known" } */
                 ^
c_files/pr24293.c:5:14: note: forward declaration of 'union bar'
static union bar y;  /* { dg-error "storage size of 'y' isn't known" } */
             ^
c_files/pr24293.c:8:10: warning: tentative definition of variable with internal linkage has incomplete non-array type 'p' (aka 'struct P') [-Wtentative-definition-incomplete-type]
static p p_obj;  /* { dg-error "storage size of 'p_obj' isn't known" } */
         ^
c_files/pr24293.c:7:16: note: forward declaration of 'struct P'
typedef struct P p;
               ^
c_files/pr24293.c:10:15: warning: tentative definition of variable with internal linkage has incomplete non-array type 'enum e' [-Wtentative-definition-incomplete-type]
static enum e e_var; /* { dg-error "storage size of 'e_var' isn't known" } */
              ^
c_files/pr24293.c:10:13: note: forward declaration of 'enum e'
static enum e e_var; /* { dg-error "storage size of 'e_var' isn't known" } */
            ^
c_files/pr24293.c:4:19: error: tentative definition has type 'struct foo' that is never completed
static struct foo x;  /* { dg-error "storage size of 'x' isn't known" } */
                  ^
c_files/pr24293.c:4:15: note: forward declaration of 'struct foo'
static struct foo x;  /* { dg-error "storage size of 'x' isn't known" } */
              ^
c_files/pr24293.c:5:18: error: tentative definition has type 'union bar' that is never completed
static union bar y;  /* { dg-error "storage size of 'y' isn't known" } */
                 ^
c_files/pr24293.c:5:14: note: forward declaration of 'union bar'
static union bar y;  /* { dg-error "storage size of 'y' isn't known" } */
             ^
c_files/pr24293.c:8:10: error: tentative definition has type 'p' (aka 'struct P') that is never completed
static p p_obj;  /* { dg-error "storage size of 'p_obj' isn't known" } */
         ^
c_files/pr24293.c:7:16: note: forward declaration of 'struct P'
typedef struct P p;
               ^
c_files/pr24293.c:10:15: error: tentative definition has type 'enum e' that is never completed
static enum e e_var; /* { dg-error "storage size of 'e_var' isn't known" } */
              ^
c_files/pr24293.c:10:13: note: forward declaration of 'enum e'
static enum e e_var; /* { dg-error "storage size of 'e_var' isn't known" } */
            ^
4 warnings and 4 errors generated.
TranslationUnitDecl 0x7fb7970264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fb797026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fb797026740 '__int128'
|-TypedefDecl 0x7fb797026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fb797026760 'unsigned __int128'
|-TypedefDecl 0x7fb797026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fb797026b70 'struct __NSConstantString_tag'
|   `-Record 0x7fb797026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fb797026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fb797026dc0 'char *'
|   `-BuiltinType 0x7fb797026560 'char'
|-TypedefDecl 0x7fb7970270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fb797027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fb797026ee0 'struct __va_list_tag'
|     `-Record 0x7fb797026e58 '__va_list_tag'
|-RecordDecl 0x7fb797027120 <c_files/pr24293.c:4:8, col:15> col:15 struct foo
|-VarDecl 0x7fb797825a20 <col:1, col:19> col:19 invalid x 'struct foo':'struct foo' static
|-RecordDecl 0x7fb797825e88 <line:5:8, col:14> col:14 union bar
|-VarDecl 0x7fb797825f80 <col:1, col:18> col:18 invalid y 'union bar':'union bar' static
|-RecordDecl 0x7fb797825ff8 <line:7:9, col:16> col:16 struct P
|-TypedefDecl 0x7fb7978260f0 <col:1, col:18> col:18 referenced p 'struct P':'struct P'
| `-ElaboratedType 0x7fb7978260a0 'struct P' sugar
|   `-RecordType 0x7fb797826080 'struct P'
|     `-Record 0x7fb797825ff8 'P'
|-VarDecl 0x7fb797826190 <line:8:1, col:10> col:10 invalid p_obj 'p':'struct P' static
|-VarDecl 0x7fb797826300 <line:10:1, col:15> col:15 invalid e_var 'enum e':'enum e' static
|-RecordDecl 0x7fb797826378 <line:12:8, col:15> col:15 struct undefined_object
`-VarDecl 0x7fb797826470 <col:1, col:32> col:32 object 'struct undefined_object':'struct undefined_object' extern
