pr29612-1.c:7:25: warning: multi-character character constant [-Wmultichar]
int foo (void) { return 'ab'; } /* { dg-warning "multi-character" } */
                        ^
pr29612-1.c:15:25: warning: multi-character character constant [-Wmultichar]
int baz (void) { return 'ab'; } /* { dg-warning "multi-character" } */
                        ^
2 warnings generated.
TranslationUnitDecl 0x7f970d8264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7f970d826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7f970d826740 '__int128'
|-TypedefDecl 0x7f970d826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7f970d826760 'unsigned __int128'
|-TypedefDecl 0x7f970d826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7f970d826b70 'struct __NSConstantString_tag'
|   `-Record 0x7f970d826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7f970d826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7f970d826dc0 'char *'
|   `-BuiltinType 0x7f970d826560 'char'
|-TypedefDecl 0x7f970d8270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7f970d827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7f970d826ee0 'struct __va_list_tag'
|     `-Record 0x7f970d826e58 '__va_list_tag'
|-FunctionDecl 0x7f970e025a00 <pr29612-1.c:7:1, col:31> col:5 foo 'int (void)'
| `-CompoundStmt 0x7f970e025b08 <col:16, col:31>
|   `-ReturnStmt 0x7f970e025af0 <col:18, col:25>
|     `-CharacterLiteral 0x7f970e025ad8 <col:25> 'int' 24930
|-FunctionDecl 0x7f970e025bc0 <foo.h:2:1, col:31> col:5 bar 'int (void)'
| `-CompoundStmt 0x7f970e025c88 <col:16, col:31>
|   `-ReturnStmt 0x7f970e025c70 <col:18, col:25>
|     `-CharacterLiteral 0x7f970e025c58 <col:25> 'int' 24930
`-FunctionDecl 0x7f970e025d40 <pr29612-1.c:15:1, col:31> col:5 baz 'int (void)'
  `-CompoundStmt 0x7f970e025e08 <col:16, col:31>
    `-ReturnStmt 0x7f970e025df0 <col:18, col:25>
      `-CharacterLiteral 0x7f970e025dd8 <col:25> 'int' 24930
