c_files/pr30551-3.c:5:1: warning: return type of 'main' is not 'int' [-Wmain-return-type]
void main(char a) {} /* { dg-error "first argument of .main. should be .int." "int" } */
^
c_files/pr30551-3.c:5:1: note: change return type to 'int'
void main(char a) {} /* { dg-error "first argument of .main. should be .int." "int" } */
^~~~
int
c_files/pr30551-3.c:5:6: error: first parameter of 'main' (argument count) must be of type 'int'
void main(char a) {} /* { dg-error "first argument of .main. should be .int." "int" } */
     ^
1 warning and 1 error generated.
TranslationUnitDecl 0x7ff03d0264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7ff03d026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7ff03d026740 '__int128'
|-TypedefDecl 0x7ff03d026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7ff03d026760 'unsigned __int128'
|-TypedefDecl 0x7ff03d026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7ff03d026b70 'struct __NSConstantString_tag'
|   `-Record 0x7ff03d026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7ff03d026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7ff03d026dc0 'char *'
|   `-BuiltinType 0x7ff03d026560 'char'
|-TypedefDecl 0x7ff03d0270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7ff03d027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7ff03d026ee0 'struct __va_list_tag'
|     `-Record 0x7ff03d026e58 '__va_list_tag'
`-FunctionDecl 0x7ff03d825a00 <c_files/pr30551-3.c:5:1, col:20> col:6 invalid main 'void (char)'
  |-ParmVarDecl 0x7ff03d027130 <col:11, col:16> col:16 a 'char'
  `-CompoundStmt 0x7ff03d825e80 <col:19, col:20>
