c_files/pr36489.c:10:26: warning: subobject initialization overrides initialization of other fields within its enclosing subobject [-Winitializer-overrides]
  = { [1][2] = 11, [1] = { [3] = 12 } };        /* { dg-warning "initializ" } */
                         ^~~~~~~~~~~~
c_files/pr36489.c:10:10: note: previous initialization is here
  = { [1][2] = 11, [1] = { [3] = 12 } };        /* { dg-warning "initializ" } */
         ^~~~~~~~
c_files/pr36489.c:12:29: warning: initializer overrides prior initialization of this subobject [-Winitializer-overrides]
  = { [1][2] = 11, [1][2] = 12 };               /* { dg-warning "initializ" } */
                            ^~
c_files/pr36489.c:12:16: note: previous initialization is here
  = { [1][2] = 11, [1][2] = 12 };               /* { dg-warning "initializ" } */
               ^~
c_files/pr36489.c:16:43: warning: subobject initialization overrides initialization of other fields within its enclosing subobject [-Winitializer-overrides]
  = { [0].b[0] = 1, [0].b[1] = 2, [0].b = { 3 } }; /* { dg-warning "initializ" } */
                                          ^~~~~
c_files/pr36489.c:16:12: note: previous initialization is here
  = { [0].b[0] = 1, [0].b[1] = 2, [0].b = { 3 } }; /* { dg-warning "initializ" } */
           ^~~~~~~
c_files/pr36489.c:20:34: warning: subobject initialization overrides initialization of other fields within its enclosing subobject [-Winitializer-overrides]
  = { .b[0] = 1, .b[1] = 2, .b = { 2 } };       /* { dg-warning "initializ" } */
                                 ^~~~~
c_files/pr36489.c:20:9: note: previous initialization is here
  = { .b[0] = 1, .b[1] = 2, .b = { 2 } };       /* { dg-warning "initializ" } */
        ^~~~~~~
c_files/pr36489.c:22:37: warning: initializer overrides prior initialization of this subobject [-Winitializer-overrides]
  = { .b[0] = 1, .b[1] = 2, .b[1] = 3 };        /* { dg-warning "initializ" } */
                                    ^
c_files/pr36489.c:22:26: note: previous initialization is here
  = { .b[0] = 1, .b[1] = 2, .b[1] = 3 };        /* { dg-warning "initializ" } */
                         ^
5 warnings generated.
TranslationUnitDecl 0x7ffa3c8264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7ffa3c826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7ffa3c826740 '__int128'
|-TypedefDecl 0x7ffa3c826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7ffa3c826760 'unsigned __int128'
|-TypedefDecl 0x7ffa3c826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7ffa3c826b70 'struct __NSConstantString_tag'
|   `-Record 0x7ffa3c826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7ffa3c826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7ffa3c826dc0 'char *'
|   `-BuiltinType 0x7ffa3c826560 'char'
|-TypedefDecl 0x7ffa3c8270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7ffa3c827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7ffa3c826ee0 'struct __va_list_tag'
|     `-Record 0x7ffa3c826e58 '__va_list_tag'
|-RecordDecl 0x7ffa3c827120 <c_files/pr36489.c:5:1, col:29> col:8 struct A definition
| |-FieldDecl 0x7ffa3d025a00 <col:12, col:16> col:16 a 'int'
| `-FieldDecl 0x7ffa3d025ad0 <col:19, col:26> col:23 b 'int [3]'
|-RecordDecl 0x7ffa3d025b18 <line:6:1, col:28> col:7 union B definition
| |-FieldDecl 0x7ffa3d025bd8 <col:11, col:15> col:15 a 'int'
| `-FieldDecl 0x7ffa3d025c68 <col:18, col:25> col:22 b 'int [3]'
|-VarDecl 0x7ffa3d025da0 <line:7:1, line:8:32> line:7:5 t1 'int [10][10]' cinit
| `-InitListExpr 0x7ffa3d026020 <line:8:5, col:32> 'int [10][10]'
|   |-array filler
|   | `-ImplicitValueInitExpr 0x7ffa3d026100 <<invalid sloc>> 'int [10]'
|   |-ImplicitValueInitExpr 0x7ffa3d026100 <<invalid sloc>> 'int [10]'
|   `-InitListExpr 0x7ffa3d026070 <col:10, col:16> 'int [10]'
|     |-array filler
|     | `-ImplicitValueInitExpr 0x7ffa3d026110 <<invalid sloc>> 'int'
|     |-ImplicitValueInitExpr 0x7ffa3d026110 <<invalid sloc>> 'int'
|     |-ImplicitValueInitExpr 0x7ffa3d026110 <<invalid sloc>> 'int'
|     |-IntegerLiteral 0x7ffa3d025e80 <col:16> 'int' 11
|     `-IntegerLiteral 0x7ffa3d025f48 <col:29> 'int' 12
|-VarDecl 0x7ffa3d026198 <line:9:1, line:10:39> line:9:5 t2 'int [10][10]' cinit
| `-InitListExpr 0x7ffa3d026448 <line:10:5, col:39> 'int [10][10]'
|   |-array filler
|   | `-ImplicitValueInitExpr 0x7ffa3d026598 <<invalid sloc>> 'int [10]'
|   |-ImplicitValueInitExpr 0x7ffa3d026598 <<invalid sloc>> 'int [10]'
|   `-InitListExpr 0x7ffa3d026528 <col:26, col:37> 'int [10]'
|     |-array filler
|     | `-ImplicitValueInitExpr 0x7ffa3d026588 <<invalid sloc>> 'int'
|     |-ImplicitValueInitExpr 0x7ffa3d026588 <<invalid sloc>> 'int'
|     |-ImplicitValueInitExpr 0x7ffa3d026588 <<invalid sloc>> 'int'
|     |-ImplicitValueInitExpr 0x7ffa3d026588 <<invalid sloc>> 'int'
|     `-IntegerLiteral 0x7ffa3d026300 <col:34> 'int' 12
|-VarDecl 0x7ffa3d026620 <line:11:1, line:12:32> line:11:5 t3 'int [10][10]' cinit
| `-InitListExpr 0x7ffa3d026860 <line:12:5, col:32> 'int [10][10]'
|   |-array filler
|   | `-ImplicitValueInitExpr 0x7ffa3d026940 <<invalid sloc>> 'int [10]'
|   |-ImplicitValueInitExpr 0x7ffa3d026940 <<invalid sloc>> 'int [10]'
|   `-InitListExpr 0x7ffa3d0268b0 <col:10, col:16> 'int [10]'
|     |-array filler
|     | `-ImplicitValueInitExpr 0x7ffa3d026950 <<invalid sloc>> 'int'
|     |-ImplicitValueInitExpr 0x7ffa3d026950 <<invalid sloc>> 'int'
|     |-ImplicitValueInitExpr 0x7ffa3d026950 <<invalid sloc>> 'int'
|     `-IntegerLiteral 0x7ffa3d026788 <col:29> 'int' 12
|-VarDecl 0x7ffa3d026aa8 <line:13:1, line:14:48> line:13:10 t4 'struct A [2]' cinit
| `-InitListExpr 0x7ffa3d026e00 <line:14:5, col:48> 'struct A [2]'
|   |-array filler
|   | `-ImplicitValueInitExpr 0x7ffa3d026f08 <<invalid sloc>> 'struct A':'struct A'
|   `-InitListExpr 0x7ffa3d026e50 <col:10, col:18> 'struct A':'struct A'
|     |-ImplicitValueInitExpr 0x7ffa3d026ef8 <<invalid sloc>> 'int'
|     `-InitListExpr 0x7ffa3d026ea0 <col:12, col:18> 'int [3]'
|       |-IntegerLiteral 0x7ffa3d026b48 <col:18> 'int' 1
|       |-IntegerLiteral 0x7ffa3d026c28 <col:32> 'int' 2
|       `-IntegerLiteral 0x7ffa3d026d08 <col:46> 'int' 3
|-VarDecl 0x7ffa3d026f68 <line:15:1, line:16:49> line:15:10 t5 'struct A [2]' cinit
| `-InitListExpr 0x7ffa3d0272c8 <line:16:5, col:49> 'struct A [2]'
|   |-array filler
|   | `-ImplicitValueInitExpr 0x7ffa3d027428 <<invalid sloc>> 'struct A':'struct A'
|   `-InitListExpr 0x7ffa3d027318 <col:10, col:18> 'struct A':'struct A'
|     |-ImplicitValueInitExpr 0x7ffa3d027418 <<invalid sloc>> 'int'
|     `-InitListExpr 0x7ffa3d0273c0 <col:43, col:47> 'int [3]'
|       |-array filler
|       | `-ImplicitValueInitExpr 0x7ffa3d027408 <<invalid sloc>> 'int'
|       `-IntegerLiteral 0x7ffa3d0271a8 <col:45> 'int' 3
|-VarDecl 0x7ffa3d027490 <line:17:1, line:18:39> line:17:9 t6 'union B':'union B' cinit
| `-InitListExpr 0x7ffa3d027728 <line:18:5, col:39> 'union B':'union B' field Field 0x7ffa3d025c68 'b' 'int [3]'
|   `-InitListExpr 0x7ffa3d027770 <col:9, col:15> 'int [3]'
|     |-IntegerLiteral 0x7ffa3d027510 <col:15> 'int' 1
|     |-IntegerLiteral 0x7ffa3d0275b0 <col:26> 'int' 2
|     `-IntegerLiteral 0x7ffa3d027650 <col:37> 'int' 3
|-VarDecl 0x7ffa3d0277e8 <line:19:1, line:20:40> line:19:9 t7 'union B':'union B' cinit
| `-InitListExpr 0x7ffa3d027a98 <line:20:5, col:40> 'union B':'union B' field Field 0x7ffa3d025c68 'b' 'int [3]'
|   `-InitListExpr 0x7ffa3d027b38 <col:34, col:38> 'int [3]'
|     |-array filler
|     | `-ImplicitValueInitExpr 0x7ffa3d027b80 <<invalid sloc>> 'int'
|     `-IntegerLiteral 0x7ffa3d027988 <col:36> 'int' 2
`-VarDecl 0x7ffa3d027bb0 <line:21:1, line:22:39> line:21:9 t8 'union B':'union B' cinit
  `-InitListExpr 0x7ffa3d027e48 <line:22:5, col:39> 'union B':'union B' field Field 0x7ffa3d025c68 'b' 'int [3]'
    `-InitListExpr 0x7ffa3d027e90 <col:9, col:15> 'int [3]'
      |-array filler
      | `-ImplicitValueInitExpr 0x7ffa3d027ee8 <<invalid sloc>> 'int'
      |-IntegerLiteral 0x7ffa3d027c30 <col:15> 'int' 1
      `-IntegerLiteral 0x7ffa3d027d70 <col:37> 'int' 3
