c_files/pr39013-2.c:6:12: warning: inline function 'foo' is not defined [-Wundefined-inline]
inline int foo (void);          /* { dg-warning "declared but never defined" } */
           ^
c_files/pr39013-2.c:12:10: note: used here
  return foo () + bar ();
         ^
c_files/pr39013-2.c:7:19: warning: inline function 'bar' is not defined [-Wundefined-inline]
extern inline int bar (void);   /* { dg-warning "declared but never defined" } */
                  ^
c_files/pr39013-2.c:12:19: note: used here
  return foo () + bar ();
                  ^
2 warnings generated.
TranslationUnitDecl 0x7f85a70264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7f85a7026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7f85a7026740 '__int128'
|-TypedefDecl 0x7f85a7026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7f85a7026760 'unsigned __int128'
|-TypedefDecl 0x7f85a7026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7f85a7026b70 'struct __NSConstantString_tag'
|   `-Record 0x7f85a7026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7f85a7026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7f85a7026dc0 'char *'
|   `-BuiltinType 0x7f85a7026560 'char'
|-TypedefDecl 0x7f85a70270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7f85a7027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7f85a7026ee0 'struct __va_list_tag'
|     `-Record 0x7f85a7026e58 '__va_list_tag'
|-FunctionDecl 0x7f85a7077400 <c_files/pr39013-2.c:6:1, col:21> col:12 used foo 'int (void)' inline
|-FunctionDecl 0x7f85a7077570 <line:7:1, col:28> col:19 used bar 'int (void)' extern inline
`-FunctionDecl 0x7f85a70776a0 <line:9:1, line:13:1> line:10:1 main 'int (void)'
  `-CompoundStmt 0x7f85a70778a0 <line:11:1, line:13:1>
    `-ReturnStmt 0x7f85a7077888 <line:12:3, col:24>
      `-BinaryOperator 0x7f85a7077860 <col:10, col:24> 'int' '+'
        |-CallExpr 0x7f85a70777d0 <col:10, col:15> 'int'
        | `-ImplicitCastExpr 0x7f85a70777b8 <col:10> 'int (*)(void)' <FunctionToPointerDecay>
        |   `-DeclRefExpr 0x7f85a7077768 <col:10> 'int (void)' Function 0x7f85a7077400 'foo' 'int (void)'
        `-CallExpr 0x7f85a7077838 <col:19, col:24> 'int'
          `-ImplicitCastExpr 0x7f85a7077820 <col:19> 'int (*)(void)' <FunctionToPointerDecay>
            `-DeclRefExpr 0x7f85a70777f8 <col:19> 'int (void)' Function 0x7f85a7077570 'bar' 'int (void)'
