c_files/pr41842.c:7:10: warning: implicit declaration of function 'g' is invalid in C99 [-Wimplicit-function-declaration]
  char x[g (h)];        /* { dg-error "undeclared" } */
         ^
c_files/pr41842.c:7:13: error: use of undeclared identifier 'h'
  char x[g (h)];        /* { dg-error "undeclared" } */
            ^
1 warning and 1 error generated.
TranslationUnitDecl 0x7ff4ff0264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7ff4ff026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7ff4ff026740 '__int128'
|-TypedefDecl 0x7ff4ff026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7ff4ff026760 'unsigned __int128'
|-TypedefDecl 0x7ff4ff026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7ff4ff026b70 'struct __NSConstantString_tag'
|   `-Record 0x7ff4ff026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7ff4ff026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7ff4ff026dc0 'char *'
|   `-BuiltinType 0x7ff4ff026560 'char'
|-TypedefDecl 0x7ff4ff0270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7ff4ff027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7ff4ff026ee0 'struct __va_list_tag'
|     `-Record 0x7ff4ff026e58 '__va_list_tag'
|-FunctionDecl 0x7ff4ff027168 <c_files/pr41842.c:4:1, line:9:1> line:5:1 f 'void ()'
| `-CompoundStmt 0x7ff4ff825c78 <line:6:1, line:9:1>
|   `-DeclStmt 0x7ff4ff825c60 <line:7:3, col:16>
|     `-VarDecl 0x7ff4ff825c00 <col:3, col:8> col:8 invalid x 'char'
`-FunctionDecl 0x7ff4ff825a90 <<invalid sloc>> col:10 implicit used g 'int ()'
