c_files/pr53037-4.c:5:25: warning: unknown attribute 'warn_if_not_aligned' ignored [-Wunknown-attributes]
int foo1 __attribute__((warn_if_not_aligned(8))); /* { dg-error "'warn_if_not_aligned' may not be specified for 'foo1'" } */
                        ^
c_files/pr53037-4.c:7:16: warning: unknown attribute 'warn_if_not_aligned' ignored [-Wunknown-attributes]
__attribute__((warn_if_not_aligned(8)))
               ^
c_files/pr53037-4.c:15:28: warning: unknown attribute 'warn_if_not_aligned' ignored [-Wunknown-attributes]
  int i : 2 __attribute__((warn_if_not_aligned(8))); /* { dg-error "'warn_if_not_aligned' may not be specified for 'i'" } */
                           ^
c_files/pr53037-4.c:19:29: warning: unknown attribute 'warn_if_not_aligned' ignored [-Wunknown-attributes]
  __attribute__((aligned(4),warn_if_not_aligned(8)));
                            ^
4 warnings generated.
TranslationUnitDecl 0x7ffe9880a4d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7ffe9880aa20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7ffe9880a740 '__int128'
|-TypedefDecl 0x7ffe9880aa90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7ffe9880a760 'unsigned __int128'
|-TypedefDecl 0x7ffe9880ad68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7ffe9880ab70 'struct __NSConstantString_tag'
|   `-Record 0x7ffe9880aae8 '__NSConstantString_tag'
|-TypedefDecl 0x7ffe9880ae00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7ffe9880adc0 'char *'
|   `-BuiltinType 0x7ffe9880a560 'char'
|-TypedefDecl 0x7ffe9880b0c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7ffe9880b070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7ffe9880aee0 'struct __va_list_tag'
|     `-Record 0x7ffe9880ae58 '__va_list_tag'
|-VarDecl 0x7ffe9880b158 <c_files/pr53037-4.c:5:1, col:5> col:5 foo1 'int'
|-FunctionDecl 0x7ffe9802a6e0 <line:7:1, line:11:1> line:9:1 foo2 'void (void)'
| `-CompoundStmt 0x7ffe9802a778 <line:10:1, line:11:1>
|-RecordDecl 0x7ffe9802a790 <line:13:1, line:16:1> line:13:8 struct foo3 definition
| `-FieldDecl 0x7ffe9802a898 <line:15:3, col:11> col:7 i 'int'
|   `-IntegerLiteral 0x7ffe9802a840 <col:11> 'int' 2
|-TypedefDecl 0x7ffe9802a938 <line:18:1, col:22> col:22 referenced __u32 'unsigned int'
| |-BuiltinType 0x7ffe9880a660 'unsigned int'
| `-AlignedAttr 0x7ffe9802a990 <line:19:18, col:27> aligned
|   `-IntegerLiteral 0x7ffe9802a8e0 <col:26> 'int' 4
`-RecordDecl 0x7ffe9802a9e0 <line:21:1, line:24:1> line:21:8 struct foo4 definition
  `-FieldDecl 0x7ffe9802aae0 <line:23:3, col:13> col:9 i '__u32':'unsigned int'
    `-IntegerLiteral 0x7ffe9802aab0 <col:13> 'int' 2
