c_files/pr60114.c:8:9: warning: implicit conversion from 'int' to 'signed char' changes value from 256 to 0 [-Wconstant-conversion]
  [0] = 0x100, /* { dg-warning "-Woverflow" } */
        ^~~~~
c_files/pr60114.c:9:9: warning: implicit conversion from 'int' to 'signed char' changes value from 257 to 1 [-Wconstant-conversion]
  [2] = 0x101, /* { dg-warning "-Woverflow" } */
        ^~~~~
c_files/pr60114.c:26:3: warning: expression result unused [-Wunused-value]
  (const signed char []) { 0x100 }; /* { dg-warning "28:-Woverflow" } */
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
c_files/pr60114.c:27:3: warning: expression result unused [-Wunused-value]
  (const float []) { 1e0, 1e1, 1e100 }; /* { dg-warning "32:conversion" } */
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
c_files/pr60114.c:22:28: warning: implicit conversion from 'int' to 'signed char' changes value from 256 to 0 [-Wconstant-conversion]
  signed char a[][3] = { { 0x100, /* { dg-warning "28:-Woverflow" } */
                         ~ ^~~~~
c_files/pr60114.c:23:24: warning: implicit conversion from 'int' to 'signed char' changes value from 256 to 0 [-Wconstant-conversion]
                    1, 0x100 }, /* { dg-warning "24:-Woverflow" } */
                       ^~~~~
c_files/pr60114.c:24:27: warning: implicit conversion from 'int' to 'signed char' changes value from 256 to 0 [-Wconstant-conversion]
                  { '\0', 0x100, '\0' } /* { dg-warning "27:-Woverflow" } */
                  ~       ^~~~~
c_files/pr60114.c:26:28: warning: implicit conversion from 'int' to 'signed char' changes value from 256 to 0 [-Wconstant-conversion]
  (const signed char []) { 0x100 }; /* { dg-warning "28:-Woverflow" } */
                         ~ ^~~~~
8 warnings generated.
TranslationUnitDecl 0x7ff80a8264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7ff80a826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7ff80a826740 '__int128'
|-TypedefDecl 0x7ff80a826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7ff80a826760 'unsigned __int128'
|-TypedefDecl 0x7ff80a826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7ff80a826b70 'struct __NSConstantString_tag'
|   `-Record 0x7ff80a826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7ff80a826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7ff80a826dc0 'char *'
|   `-BuiltinType 0x7ff80a826560 'char'
|-TypedefDecl 0x7ff80a8270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7ff80a827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7ff80a826ee0 'struct __va_list_tag'
|     `-Record 0x7ff80a826e58 '__va_list_tag'
|-RecordDecl 0x7ff80a827120 <c_files/pr60114.c:6:1, col:25> col:8 struct S definition
| |-FieldDecl 0x7ff80a877a00 <col:12, col:16> col:16 n 'int'
| `-FieldDecl 0x7ff80a877ad0 <col:12, col:22> col:19 u 'int [2]'
|-VarDecl 0x7ff80a877ba0 <line:7:1, line:10:1> line:7:19 z 'const signed char [3]' cinit
| `-InitListExpr 0x7ff80a877da0 <col:25, line:10:1> 'const signed char [3]'
|   |-array filler
|   | `-ImplicitValueInitExpr 0x7ff80a877ea8 <<invalid sloc>> 'const signed char'
|   |-ImplicitCastExpr 0x7ff80a877de8 <line:8:9> 'signed char' <IntegralCast>
|   | `-IntegerLiteral 0x7ff80a877c60 <col:9> 'int' 256
|   |-ImplicitValueInitExpr 0x7ff80a877ea8 <<invalid sloc>> 'const signed char'
|   `-ImplicitCastExpr 0x7ff80a877e18 <line:9:9> 'signed char' <IntegralCast>
|     `-IntegerLiteral 0x7ff80a877ce8 <col:9> 'int' 257
|-VarDecl 0x7ff80a8782c0 <line:11:1, line:15:11> line:11:5 A 'int [7]' cinit
| `-InitListExpr 0x7ff80a878478 <col:11, line:15:11> 'int [7]'
|   |-IntegerLiteral 0x7ff80a878320 <line:12:13> 'int' 0
|   |-ImplicitCastExpr 0x7ff80a8784c0 <col:16> 'int' <IntegralCast>
|   | `-IntegerLiteral 0x7ff80a878340 <col:16> 'unsigned int' 2147483648
|   |-IntegerLiteral 0x7ff80a878360 <line:13:13> 'int' 10
|   |-ImplicitCastExpr 0x7ff80a878508 <col:18> 'int' <IntegralCast>
|   | `-IntegerLiteral 0x7ff80a878380 <col:18> 'unsigned int' 2147483648
|   |-IntegerLiteral 0x7ff80a8783a0 <line:14:13> 'int' 10
|   |-IntegerLiteral 0x7ff80a8783c0 <col:18> 'int' 10
|   `-ImplicitCastExpr 0x7ff80a878560 <col:23> 'int' <IntegralCast>
|     `-IntegerLiteral 0x7ff80a8783e0 <col:23> 'unsigned int' 2147483648
|-VarDecl 0x7ff80a878600 <line:16:1, col:32> col:6 p 'int *' cinit
| `-ImplicitCastExpr 0x7ff80a8787b0 <col:10, col:32> 'int *' <ArrayToPointerDecay>
|   `-CompoundLiteralExpr 0x7ff80a878788 <col:10, col:32> 'int [1]' lvalue
|     `-InitListExpr 0x7ff80a8786f0 <col:19, col:32> 'int [1]'
|       `-ImplicitCastExpr 0x7ff80a878730 <col:21> 'int' <IntegralCast>
|         `-IntegerLiteral 0x7ff80a878688 <col:21> 'unsigned int' 2147483648
|-RecordDecl 0x7ff80a8787c8 <line:17:1, col:16> col:1 union definition
| `-FieldDecl 0x7ff80a878888 <col:9, col:13> col:13 k 'int'
|-VarDecl 0x7ff80a878920 <col:1, col:40> col:18 u 'union (anonymous union at c_files/pr60114.c:17:1)':'union (anonymous at c_files/pr60114.c:17:1)' cinit
| `-InitListExpr 0x7ff80a879a48 <col:22, col:40> 'union (anonymous union at c_files/pr60114.c:17:1)':'union (anonymous at c_files/pr60114.c:17:1)' field Field 0x7ff80a878888 'k' 'int'
|   `-ImplicitCastExpr 0x7ff80a879a90 <col:29> 'int' <IntegralCast>
|     `-IntegerLiteral 0x7ff80a878980 <col:29> 'unsigned int' 2147483648
|-TypedefDecl 0x7ff80a879ae8 <line:18:1, col:15> col:13 referenced H 'int []'
| `-IncompleteArrayType 0x7ff80a878270 'int []' 
|   `-BuiltinType 0x7ff80a8265c0 'int'
`-FunctionDecl 0x7ff80a879c00 <line:19:1, line:32:1> line:20:1 foo 'void (void)'
  `-CompoundStmt 0x7ff80a87add0 <line:21:1, line:32:1>
    |-DeclStmt 0x7ff80a87a108 <line:22:3, line:25:18>
    | `-VarDecl 0x7ff80a879d20 <line:22:3, line:25:17> line:22:15 a 'signed char [2][3]' cinit
    |   `-InitListExpr 0x7ff80a879f30 <col:24, line:25:17> 'signed char [2][3]'
    |     |-InitListExpr 0x7ff80a879f70 <line:22:26, line:23:30> 'signed char [3]'
    |     | |-ImplicitCastExpr 0x7ff80a879fd0 <line:22:28> 'signed char' <IntegralCast>
    |     | | `-IntegerLiteral 0x7ff80a879d80 <col:28> 'int' 256
    |     | |-ImplicitCastExpr 0x7ff80a879fe8 <line:23:21> 'signed char' <IntegralCast>
    |     | | `-IntegerLiteral 0x7ff80a879da0 <col:21> 'int' 1
    |     | `-ImplicitCastExpr 0x7ff80a87a000 <col:24> 'signed char' <IntegralCast>
    |     |   `-IntegerLiteral 0x7ff80a879dc0 <col:24> 'int' 256
    |     `-InitListExpr 0x7ff80a87a018 <line:24:19, col:39> 'signed char [3]'
    |       |-ImplicitCastExpr 0x7ff80a87a080 <col:21> 'signed char' <IntegralCast>
    |       | `-CharacterLiteral 0x7ff80a879e38 <col:21> 'int' 0
    |       |-ImplicitCastExpr 0x7ff80a87a098 <col:27> 'signed char' <IntegralCast>
    |       | `-IntegerLiteral 0x7ff80a879e50 <col:27> 'int' 256
    |       `-ImplicitCastExpr 0x7ff80a87a0b0 <col:34> 'signed char' <IntegralCast>
    |         `-CharacterLiteral 0x7ff80a879e70 <col:34> 'int' 0
    |-ImplicitCastExpr 0x7ff80a87a2d8 <line:26:3, col:34> 'const signed char *' <ArrayToPointerDecay>
    | `-CompoundLiteralExpr 0x7ff80a87a288 <col:3, col:34> 'const signed char [1]' lvalue
    |   `-InitListExpr 0x7ff80a87a1b0 <col:26, col:34> 'const signed char [1]'
    |     `-ImplicitCastExpr 0x7ff80a87a1f0 <col:28> 'signed char' <IntegralCast>
    |       `-IntegerLiteral 0x7ff80a87a148 <col:28> 'int' 256
    |-ImplicitCastExpr 0x7ff80a87a5b8 <line:27:3, col:38> 'const float *' <ArrayToPointerDecay>
    | `-CompoundLiteralExpr 0x7ff80a87a568 <col:3, col:38> 'const float [3]' lvalue
    |   `-InitListExpr 0x7ff80a87a428 <col:20, col:38> 'const float [3]'
    |     |-ImplicitCastExpr 0x7ff80a87a468 <col:22> 'float' <FloatingCast>
    |     | `-FloatingLiteral 0x7ff80a87a370 <col:22> 'double' 1.000000e+00
    |     |-ImplicitCastExpr 0x7ff80a87a488 <col:27> 'float' <FloatingCast>
    |     | `-FloatingLiteral 0x7ff80a87a390 <col:27> 'double' 1.000000e+01
    |     `-ImplicitCastExpr 0x7ff80a87a4b0 <col:32> 'float' <FloatingCast>
    |       `-FloatingLiteral 0x7ff80a87a3b0 <col:32> 'double' 1.000000e+100
    |-DeclStmt 0x7ff80a87a760 <line:28:3, col:31>
    | `-VarDecl 0x7ff80a87a620 <col:3, col:30> col:12 s1 'struct S':'struct S' cinit
    |   `-InitListExpr 0x7ff80a87a6e8 <col:17, col:30> 'struct S':'struct S'
    |     |-ImplicitCastExpr 0x7ff80a87a738 <col:19> 'int' <IntegralCast>
    |     | `-IntegerLiteral 0x7ff80a87a680 <col:19> 'unsigned int' 2147483648
    |     `-ImplicitValueInitExpr 0x7ff80a87a750 <<invalid sloc>> 'int [2]'
    |-DeclStmt 0x7ff80a87a918 <line:29:3, col:36>
    | `-VarDecl 0x7ff80a87a798 <col:3, col:35> col:12 s2 'struct S':'struct S' cinit
    |   `-InitListExpr 0x7ff80a87a8a0 <col:17, col:35> 'struct S':'struct S'
    |     |-ImplicitCastExpr 0x7ff80a87a8f0 <col:24> 'int' <IntegralCast>
    |     | `-IntegerLiteral 0x7ff80a87a7f8 <col:24> 'unsigned int' 2147483648
    |     `-ImplicitValueInitExpr 0x7ff80a87a908 <<invalid sloc>> 'int [2]'
    |-DeclStmt 0x7ff80a87ab80 <line:30:3, col:39>
    | `-VarDecl 0x7ff80a87a950 <col:3, col:38> col:12 s3 'struct S':'struct S' cinit
    |   `-InitListExpr 0x7ff80a87aaa8 <col:17, col:38> 'struct S':'struct S'
    |     |-ImplicitValueInitExpr 0x7ff80a87ab60 <<invalid sloc>> 'int'
    |     `-InitListExpr 0x7ff80a87aaf8 <col:21, col:27> 'int [2]'
    |       |-array filler
    |       | `-ImplicitValueInitExpr 0x7ff80a87ab70 <<invalid sloc>> 'int'
    |       |-ImplicitValueInitExpr 0x7ff80a87ab70 <<invalid sloc>> 'int'
    |       `-ImplicitCastExpr 0x7ff80a87ab48 <col:27> 'int' <IntegralCast>
    |         `-IntegerLiteral 0x7ff80a87a9d0 <col:27> 'unsigned int' 2147483648
    `-DeclStmt 0x7ff80a87adb8 <line:31:3, col:29>
      `-VarDecl 0x7ff80a87abd0 <col:3, col:28> col:5 h 'int [3]' cinit
        `-InitListExpr 0x7ff80a87ace8 <col:9, col:28> 'int [3]'
          |-IntegerLiteral 0x7ff80a87ac30 <col:11> 'int' 1
          |-IntegerLiteral 0x7ff80a87ac50 <col:14> 'int' 2
          `-ImplicitCastExpr 0x7ff80a87ad40 <col:17> 'int' <IntegralCast>
            `-IntegerLiteral 0x7ff80a87ac70 <col:17> 'unsigned int' 2147483648
