c_files/pr64637.c:10:28: warning: expression result unused [-Wunused-value]
  for (int i = 0; i < b; i + b) /* { dg-warning "28:statement with no effect" } */
                         ~ ^ ~
c_files/pr64637.c:13:26: warning: expression result unused [-Wunused-value]
  for (int i = 0; i < b; b) /* { dg-warning "statement with no effect" } */
                         ^
c_files/pr64637.c:15:26: warning: expression result unused [-Wunused-value]
  for (int i = 0; i < b; !i) /* { dg-warning "26:statement with no effect" } */
                         ^~
c_files/pr64637.c:17:8: warning: expression result unused [-Wunused-value]
  for (!b;;) /* { dg-warning "8:statement with no effect" } */
       ^~
c_files/pr64637.c:19:13: warning: expression result unused [-Wunused-value]
  for (;; b * 2) /* { dg-warning "13:statement with no effect" } */
          ~ ^ ~
c_files/pr64637.c:22:8: warning: expression result unused [-Wunused-value]
     b / 5; /* { dg-warning "8:statement with no effect" } */
     ~ ^ ~
c_files/pr64637.c:23:8: warning: expression result unused [-Wunused-value]
     b ^ 5;
     ~ ^ ~
7 warnings generated.
TranslationUnitDecl 0x7f9b740264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7f9b74026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7f9b74026740 '__int128'
|-TypedefDecl 0x7f9b74026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7f9b74026760 'unsigned __int128'
|-TypedefDecl 0x7f9b74026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7f9b74026b70 'struct __NSConstantString_tag'
|   `-Record 0x7f9b74026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7f9b74026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7f9b74026dc0 'char *'
|   `-BuiltinType 0x7f9b74026560 'char'
|-TypedefDecl 0x7f9b740270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7f9b74027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7f9b74026ee0 'struct __va_list_tag'
|     `-Record 0x7f9b74026e58 '__va_list_tag'
|-FunctionDecl 0x7f9b74027168 <c_files/pr64637.c:5:1, col:9> col:6 used g 'void ()'
`-FunctionDecl 0x7f9b74825b20 <line:7:1, line:25:1> line:8:1 f 'void (int)'
  |-ParmVarDecl 0x7f9b74825a58 <col:4, col:8> col:8 used b 'int'
  `-CompoundStmt 0x7f9b748266a0 <line:9:1, line:25:1>
    |-ForStmt 0x7f9b74825e58 <line:10:3, line:11:8>
    | |-DeclStmt 0x7f9b74825c58 <line:10:8, col:17>
    | | `-VarDecl 0x7f9b74825bd8 <col:8, col:16> col:12 used i 'int' cinit
    | |   `-IntegerLiteral 0x7f9b74825c38 <col:16> 'int' 0
    | |-<<<NULL>>>
    | |-BinaryOperator 0x7f9b74825cf0 <col:19, col:23> 'int' '<'
    | | |-ImplicitCastExpr 0x7f9b74825cc0 <col:19> 'int' <LValueToRValue>
    | | | `-DeclRefExpr 0x7f9b74825c70 <col:19> 'int' lvalue Var 0x7f9b74825bd8 'i' 'int'
    | | `-ImplicitCastExpr 0x7f9b74825cd8 <col:23> 'int' <LValueToRValue>
    | |   `-DeclRefExpr 0x7f9b74825c98 <col:23> 'int' lvalue ParmVar 0x7f9b74825a58 'b' 'int'
    | |-BinaryOperator 0x7f9b74825d98 <col:26, col:30> 'int' '+'
    | | |-ImplicitCastExpr 0x7f9b74825d68 <col:26> 'int' <LValueToRValue>
    | | | `-DeclRefExpr 0x7f9b74825d18 <col:26> 'int' lvalue Var 0x7f9b74825bd8 'i' 'int'
    | | `-ImplicitCastExpr 0x7f9b74825d80 <col:30> 'int' <LValueToRValue>
    | |   `-DeclRefExpr 0x7f9b74825d40 <col:30> 'int' lvalue ParmVar 0x7f9b74825a58 'b' 'int'
    | `-CallExpr 0x7f9b74825e30 <line:11:5, col:8> 'void'
    |   `-ImplicitCastExpr 0x7f9b74825e18 <col:5> 'void (*)()' <FunctionToPointerDecay>
    |     `-DeclRefExpr 0x7f9b74825dc0 <col:5> 'void ()' Function 0x7f9b74027168 'g' 'void ()'
    |-ForStmt 0x7f9b74826090 <line:13:3, line:14:8>
    | |-DeclStmt 0x7f9b74825f28 <line:13:8, col:17>
    | | `-VarDecl 0x7f9b74825ea8 <col:8, col:16> col:12 used i 'int' cinit
    | |   `-IntegerLiteral 0x7f9b74825f08 <col:16> 'int' 0
    | |-<<<NULL>>>
    | |-BinaryOperator 0x7f9b74825fc0 <col:19, col:23> 'int' '<'
    | | |-ImplicitCastExpr 0x7f9b74825f90 <col:19> 'int' <LValueToRValue>
    | | | `-DeclRefExpr 0x7f9b74825f40 <col:19> 'int' lvalue Var 0x7f9b74825ea8 'i' 'int'
    | | `-ImplicitCastExpr 0x7f9b74825fa8 <col:23> 'int' <LValueToRValue>
    | |   `-DeclRefExpr 0x7f9b74825f68 <col:23> 'int' lvalue ParmVar 0x7f9b74825a58 'b' 'int'
    | |-ImplicitCastExpr 0x7f9b74826010 <col:26> 'int' <LValueToRValue>
    | | `-DeclRefExpr 0x7f9b74825fe8 <col:26> 'int' lvalue ParmVar 0x7f9b74825a58 'b' 'int'
    | `-CallExpr 0x7f9b74826068 <line:14:5, col:8> 'void'
    |   `-ImplicitCastExpr 0x7f9b74826050 <col:5> 'void (*)()' <FunctionToPointerDecay>
    |     `-DeclRefExpr 0x7f9b74826028 <col:5> 'void ()' Function 0x7f9b74027168 'g' 'void ()'
    |-ForStmt 0x7f9b748262e8 <line:15:3, line:16:8>
    | |-DeclStmt 0x7f9b74826160 <line:15:8, col:17>
    | | `-VarDecl 0x7f9b748260e0 <col:8, col:16> col:12 used i 'int' cinit
    | |   `-IntegerLiteral 0x7f9b74826140 <col:16> 'int' 0
    | |-<<<NULL>>>
    | |-BinaryOperator 0x7f9b748261f8 <col:19, col:23> 'int' '<'
    | | |-ImplicitCastExpr 0x7f9b748261c8 <col:19> 'int' <LValueToRValue>
    | | | `-DeclRefExpr 0x7f9b74826178 <col:19> 'int' lvalue Var 0x7f9b748260e0 'i' 'int'
    | | `-ImplicitCastExpr 0x7f9b748261e0 <col:23> 'int' <LValueToRValue>
    | |   `-DeclRefExpr 0x7f9b748261a0 <col:23> 'int' lvalue ParmVar 0x7f9b74825a58 'b' 'int'
    | |-UnaryOperator 0x7f9b74826260 <col:26, col:27> 'int' prefix '!'
    | | `-ImplicitCastExpr 0x7f9b74826248 <col:27> 'int' <LValueToRValue>
    | |   `-DeclRefExpr 0x7f9b74826220 <col:27> 'int' lvalue Var 0x7f9b748260e0 'i' 'int'
    | `-CallExpr 0x7f9b748262c0 <line:16:5, col:8> 'void'
    |   `-ImplicitCastExpr 0x7f9b748262a8 <col:5> 'void (*)()' <FunctionToPointerDecay>
    |     `-DeclRefExpr 0x7f9b74826280 <col:5> 'void ()' Function 0x7f9b74027168 'g' 'void ()'
    |-ForStmt 0x7f9b748263e8 <line:17:3, line:18:8>
    | |-UnaryOperator 0x7f9b74826360 <line:17:8, col:9> 'int' prefix '!'
    | | `-ImplicitCastExpr 0x7f9b74826348 <col:9> 'int' <LValueToRValue>
    | |   `-DeclRefExpr 0x7f9b74826320 <col:9> 'int' lvalue ParmVar 0x7f9b74825a58 'b' 'int'
    | |-<<<NULL>>>
    | |-<<<NULL>>>
    | |-<<<NULL>>>
    | `-CallExpr 0x7f9b748263c0 <line:18:5, col:8> 'void'
    |   `-ImplicitCastExpr 0x7f9b748263a8 <col:5> 'void (*)()' <FunctionToPointerDecay>
    |     `-DeclRefExpr 0x7f9b74826380 <col:5> 'void ()' Function 0x7f9b74027168 'g' 'void ()'
    |-ForStmt 0x7f9b74826510 <line:19:3, line:20:8>
    | |-<<<NULL>>>
    | |-<<<NULL>>>
    | |-<<<NULL>>>
    | |-BinaryOperator 0x7f9b74826480 <line:19:11, col:15> 'int' '*'
    | | |-ImplicitCastExpr 0x7f9b74826468 <col:11> 'int' <LValueToRValue>
    | | | `-DeclRefExpr 0x7f9b74826420 <col:11> 'int' lvalue ParmVar 0x7f9b74825a58 'b' 'int'
    | | `-IntegerLiteral 0x7f9b74826448 <col:15> 'int' 2
    | `-CallExpr 0x7f9b748264e8 <line:20:5, col:8> 'void'
    |   `-ImplicitCastExpr 0x7f9b748264d0 <col:5> 'void (*)()' <FunctionToPointerDecay>
    |     `-DeclRefExpr 0x7f9b748264a8 <col:5> 'void ()' Function 0x7f9b74027168 'g' 'void ()'
    `-StmtExpr 0x7f9b74826680 <line:21:3, line:24:5> 'int'
      `-CompoundStmt 0x7f9b74826658 <line:21:4, line:24:4>
        |-BinaryOperator 0x7f9b748265a8 <line:22:6, col:10> 'int' '/'
        | |-ImplicitCastExpr 0x7f9b74826590 <col:6> 'int' <LValueToRValue>
        | | `-DeclRefExpr 0x7f9b74826548 <col:6> 'int' lvalue ParmVar 0x7f9b74825a58 'b' 'int'
        | `-IntegerLiteral 0x7f9b74826570 <col:10> 'int' 5
        `-BinaryOperator 0x7f9b74826630 <line:23:6, col:10> 'int' '^'
          |-ImplicitCastExpr 0x7f9b74826618 <col:6> 'int' <LValueToRValue>
          | `-DeclRefExpr 0x7f9b748265d0 <col:6> 'int' lvalue ParmVar 0x7f9b74825a58 'b' 'int'
          `-IntegerLiteral 0x7f9b748265f8 <col:10> 'int' 5
