c_files/pr65345-3.c:7:19: warning: expression with side effects has no effect in an unevaluated context [-Wunevaluated-expression]
float a2 = sizeof (i = 0);
                  ^
c_files/pr65345-3.c:8:19: warning: expression with side effects has no effect in an unevaluated context [-Wunevaluated-expression]
float a3 = sizeof (i++);
                  ^
c_files/pr65345-3.c:9:19: warning: expression with side effects has no effect in an unevaluated context [-Wunevaluated-expression]
float a4 = sizeof (i--);
                  ^
c_files/pr65345-3.c:12:12: warning: '_Alignof' applied to an expression is a GNU extension [-Wgnu-alignof-expression]
float b1 = _Alignof (i + 1);
           ^
c_files/pr65345-3.c:13:12: warning: '_Alignof' applied to an expression is a GNU extension [-Wgnu-alignof-expression]
float b2 = _Alignof (i = 0);
           ^
c_files/pr65345-3.c:13:24: warning: expression with side effects has no effect in an unevaluated context [-Wunevaluated-expression]
float b2 = _Alignof (i = 0);
                       ^
c_files/pr65345-3.c:14:12: warning: '_Alignof' applied to an expression is a GNU extension [-Wgnu-alignof-expression]
float b3 = _Alignof (i++);
           ^
c_files/pr65345-3.c:14:23: warning: expression with side effects has no effect in an unevaluated context [-Wunevaluated-expression]
float b3 = _Alignof (i++);
                      ^
c_files/pr65345-3.c:15:12: warning: '_Alignof' applied to an expression is a GNU extension [-Wgnu-alignof-expression]
float b4 = _Alignof (i--);
           ^
c_files/pr65345-3.c:15:23: warning: expression with side effects has no effect in an unevaluated context [-Wunevaluated-expression]
float b4 = _Alignof (i--);
                      ^
c_files/pr65345-3.c:16:12: warning: '_Alignof' applied to an expression is a GNU extension [-Wgnu-alignof-expression]
float b5 = _Alignof (-i);
           ^
c_files/pr65345-3.c:18:12: error: initializer element is not a compile-time constant
float c1 = i; /* { dg-error "initializer element is not constant" } */
           ^
c_files/pr65345-3.c:19:12: error: initializer element is not a compile-time constant
float c2 = (i ? 1 : 2); /* { dg-error "initializer element is not constant" } */
           ^~~~~~~~~~~
c_files/pr65345-3.c:20:7: error: variable length array declaration not allowed at file scope
float c3[(int) i]; /* { dg-error "variably modified" } */
      ^  ~~~~~~~
c_files/pr65345-3.c:21:12: error: initializer element is not a compile-time constant
float c4 = 0 || i; /* { dg-error "initializer element is not constant" } */
           ^~~~~~
c_files/pr65345-3.c:22:7: error: initializing 'float' with an expression of incompatible type '_Atomic(float)'
float c5 = (i += 10); /* { dg-error "initializer element is not constant" } */
      ^    ~~~~~~~~~
c_files/pr65345-3.c:26:29: warning: expression with side effects has no effect in an unevaluated context [-Wunevaluated-expression]
_Static_assert (_Generic (i = 0, float: 1, default: 0) == 1, "3");
                            ^
c_files/pr65345-3.c:26:1: error: static_assert failed "3"
_Static_assert (_Generic (i = 0, float: 1, default: 0) == 1, "3");
^               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
c_files/pr65345-3.c:27:28: warning: expression with side effects has no effect in an unevaluated context [-Wunevaluated-expression]
_Static_assert (_Generic (i++, float: 1, default: 0) == 1, "4");
                           ^
c_files/pr65345-3.c:28:28: warning: expression with side effects has no effect in an unevaluated context [-Wunevaluated-expression]
_Static_assert (_Generic (i--, float: 1, default: 0) == 1, "5");
                           ^
c_files/pr65345-3.c:32:19: error: size of array has non-integer type '_Atomic(int)'
void fn2 (float a[sz = 0]);
                  ^~~~~~
14 warnings and 7 errors generated.
TranslationUnitDecl 0x7f9f6d8264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7f9f6d826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7f9f6d826740 '__int128'
|-TypedefDecl 0x7f9f6d826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7f9f6d826760 'unsigned __int128'
|-TypedefDecl 0x7f9f6d826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7f9f6d826b70 'struct __NSConstantString_tag'
|   `-Record 0x7f9f6d826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7f9f6d826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7f9f6d826dc0 'char *'
|   `-BuiltinType 0x7f9f6d826560 'char'
|-TypedefDecl 0x7f9f6d8270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7f9f6d827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7f9f6d826ee0 'struct __va_list_tag'
|     `-Record 0x7f9f6d826e58 '__va_list_tag'
|-VarDecl 0x7f9f6d827168 <c_files/pr65345-3.c:4:1, col:19> col:15 used i '_Atomic(float)' cinit
| `-ImplicitCastExpr 0x7f9f6d877a60 <col:19> '_Atomic(float)' <NonAtomicToAtomic>
|   `-FloatingLiteral 0x7f9f6d877a40 <col:19> 'float' 3.000000e+00
|-VarDecl 0x7f9f6d877a90 <line:6:1, col:27> col:7 a1 'float' cinit
| `-ImplicitCastExpr 0x7f9f6d877be8 <col:12, col:27> 'float' <IntegralToFloating>
|   `-UnaryExprOrTypeTraitExpr 0x7f9f6d877bc8 <col:12, col:27> 'unsigned long' sizeof
|     `-ParenExpr 0x7f9f6d877ba8 <col:19, col:27> 'double'
|       `-BinaryOperator 0x7f9f6d877b80 <col:20, col:24> 'double' '+'
|         |-ImplicitCastExpr 0x7f9f6d877b68 <col:20> 'double' <FloatingCast>
|         | `-ImplicitCastExpr 0x7f9f6d877b50 <col:20> 'float' <AtomicToNonAtomic>
|         |   `-ImplicitCastExpr 0x7f9f6d877b38 <col:20> '_Atomic(float)' <LValueToRValue>
|         |     `-DeclRefExpr 0x7f9f6d877af0 <col:20> '_Atomic(float)' lvalue Var 0x7f9f6d827168 'i' '_Atomic(float)'
|         `-FloatingLiteral 0x7f9f6d877b18 <col:24> 'double' 1.200000e+00
|-VarDecl 0x7f9f6d877c18 <line:7:1, col:25> col:7 a2 'float' cinit
| `-ImplicitCastExpr 0x7f9f6d877d58 <col:12, col:25> 'float' <IntegralToFloating>
|   `-UnaryExprOrTypeTraitExpr 0x7f9f6d877d38 <col:12, col:25> 'unsigned long' sizeof
|     `-ParenExpr 0x7f9f6d877d18 <col:19, col:25> '_Atomic(float)'
|       `-BinaryOperator 0x7f9f6d877cf0 <col:20, col:24> '_Atomic(float)' '='
|         |-DeclRefExpr 0x7f9f6d877c78 <col:20> '_Atomic(float)' lvalue Var 0x7f9f6d827168 'i' '_Atomic(float)'
|         `-ImplicitCastExpr 0x7f9f6d877cd8 <col:24> '_Atomic(float)' <NonAtomicToAtomic>
|           `-ImplicitCastExpr 0x7f9f6d877cc0 <col:24> 'float' <IntegralToFloating>
|             `-IntegerLiteral 0x7f9f6d877ca0 <col:24> 'int' 0
|-VarDecl 0x7f9f6d877d88 <line:8:1, col:23> col:7 a3 'float' cinit
| `-ImplicitCastExpr 0x7f9f6d877e70 <col:12, col:23> 'float' <IntegralToFloating>
|   `-UnaryExprOrTypeTraitExpr 0x7f9f6d877e50 <col:12, col:23> 'unsigned long' sizeof
|     `-ParenExpr 0x7f9f6d877e30 <col:19, col:23> 'float'
|       `-UnaryOperator 0x7f9f6d877e10 <col:20, col:21> 'float' postfix '++'
|         `-DeclRefExpr 0x7f9f6d877de8 <col:20> '_Atomic(float)' lvalue Var 0x7f9f6d827168 'i' '_Atomic(float)'
|-VarDecl 0x7f9f6d877ea0 <line:9:1, col:23> col:7 a4 'float' cinit
| `-ImplicitCastExpr 0x7f9f6d877f88 <col:12, col:23> 'float' <IntegralToFloating>
|   `-UnaryExprOrTypeTraitExpr 0x7f9f6d877f68 <col:12, col:23> 'unsigned long' sizeof
|     `-ParenExpr 0x7f9f6d877f48 <col:19, col:23> 'float'
|       `-UnaryOperator 0x7f9f6d877f28 <col:20, col:21> 'float' postfix '--'
|         `-DeclRefExpr 0x7f9f6d877f00 <col:20> '_Atomic(float)' lvalue Var 0x7f9f6d827168 'i' '_Atomic(float)'
|-VarDecl 0x7f9f6d877fb8 <line:10:1, col:22> col:7 a5 'float' cinit
| `-ImplicitCastExpr 0x7f9f6d8780d0 <col:12, col:22> 'float' <IntegralToFloating>
|   `-UnaryExprOrTypeTraitExpr 0x7f9f6d8780b0 <col:12, col:22> 'unsigned long' sizeof
|     `-ParenExpr 0x7f9f6d878090 <col:19, col:22> 'float'
|       `-UnaryOperator 0x7f9f6d878070 <col:20, col:21> 'float' prefix '-'
|         `-ImplicitCastExpr 0x7f9f6d878058 <col:21> 'float' <AtomicToNonAtomic>
|           `-ImplicitCastExpr 0x7f9f6d878040 <col:21> '_Atomic(float)' <LValueToRValue>
|             `-DeclRefExpr 0x7f9f6d878018 <col:21> '_Atomic(float)' lvalue Var 0x7f9f6d827168 'i' '_Atomic(float)'
|-VarDecl 0x7f9f6d878100 <line:12:1, col:27> col:7 b1 'float' cinit
| `-ImplicitCastExpr 0x7f9f6d878258 <col:12, col:27> 'float' <IntegralToFloating>
|   `-UnaryExprOrTypeTraitExpr 0x7f9f6d878238 <col:12, col:27> 'unsigned long' alignof
|     `-ParenExpr 0x7f9f6d878218 <col:21, col:27> 'float'
|       `-BinaryOperator 0x7f9f6d8781f0 <col:22, col:26> 'float' '+'
|         |-ImplicitCastExpr 0x7f9f6d8781c0 <col:22> 'float' <AtomicToNonAtomic>
|         | `-ImplicitCastExpr 0x7f9f6d8781a8 <col:22> '_Atomic(float)' <LValueToRValue>
|         |   `-DeclRefExpr 0x7f9f6d878160 <col:22> '_Atomic(float)' lvalue Var 0x7f9f6d827168 'i' '_Atomic(float)'
|         `-ImplicitCastExpr 0x7f9f6d8781d8 <col:26> 'float' <IntegralToFloating>
|           `-IntegerLiteral 0x7f9f6d878188 <col:26> 'int' 1
|-VarDecl 0x7f9f6d878288 <line:13:1, col:27> col:7 b2 'float' cinit
| `-ImplicitCastExpr 0x7f9f6d8783c8 <col:12, col:27> 'float' <IntegralToFloating>
|   `-UnaryExprOrTypeTraitExpr 0x7f9f6d8783a8 <col:12, col:27> 'unsigned long' alignof
|     `-ParenExpr 0x7f9f6d878388 <col:21, col:27> '_Atomic(float)'
|       `-BinaryOperator 0x7f9f6d878360 <col:22, col:26> '_Atomic(float)' '='
|         |-DeclRefExpr 0x7f9f6d8782e8 <col:22> '_Atomic(float)' lvalue Var 0x7f9f6d827168 'i' '_Atomic(float)'
|         `-ImplicitCastExpr 0x7f9f6d878348 <col:26> '_Atomic(float)' <NonAtomicToAtomic>
|           `-ImplicitCastExpr 0x7f9f6d878330 <col:26> 'float' <IntegralToFloating>
|             `-IntegerLiteral 0x7f9f6d878310 <col:26> 'int' 0
|-VarDecl 0x7f9f6d8783f8 <line:14:1, col:25> col:7 b3 'float' cinit
| `-ImplicitCastExpr 0x7f9f6d8784e0 <col:12, col:25> 'float' <IntegralToFloating>
|   `-UnaryExprOrTypeTraitExpr 0x7f9f6d8784c0 <col:12, col:25> 'unsigned long' alignof
|     `-ParenExpr 0x7f9f6d8784a0 <col:21, col:25> 'float'
|       `-UnaryOperator 0x7f9f6d878480 <col:22, col:23> 'float' postfix '++'
|         `-DeclRefExpr 0x7f9f6d878458 <col:22> '_Atomic(float)' lvalue Var 0x7f9f6d827168 'i' '_Atomic(float)'
|-VarDecl 0x7f9f6d878510 <line:15:1, col:25> col:7 b4 'float' cinit
| `-ImplicitCastExpr 0x7f9f6d8785f8 <col:12, col:25> 'float' <IntegralToFloating>
|   `-UnaryExprOrTypeTraitExpr 0x7f9f6d8785d8 <col:12, col:25> 'unsigned long' alignof
|     `-ParenExpr 0x7f9f6d8785b8 <col:21, col:25> 'float'
|       `-UnaryOperator 0x7f9f6d878598 <col:22, col:23> 'float' postfix '--'
|         `-DeclRefExpr 0x7f9f6d878570 <col:22> '_Atomic(float)' lvalue Var 0x7f9f6d827168 'i' '_Atomic(float)'
|-VarDecl 0x7f9f6d878628 <line:16:1, col:24> col:7 b5 'float' cinit
| `-ImplicitCastExpr 0x7f9f6d878740 <col:12, col:24> 'float' <IntegralToFloating>
|   `-UnaryExprOrTypeTraitExpr 0x7f9f6d878720 <col:12, col:24> 'unsigned long' alignof
|     `-ParenExpr 0x7f9f6d878700 <col:21, col:24> 'float'
|       `-UnaryOperator 0x7f9f6d8786e0 <col:22, col:23> 'float' prefix '-'
|         `-ImplicitCastExpr 0x7f9f6d8786c8 <col:23> 'float' <AtomicToNonAtomic>
|           `-ImplicitCastExpr 0x7f9f6d8786b0 <col:23> '_Atomic(float)' <LValueToRValue>
|             `-DeclRefExpr 0x7f9f6d878688 <col:23> '_Atomic(float)' lvalue Var 0x7f9f6d827168 'i' '_Atomic(float)'
|-VarDecl 0x7f9f6d878770 <line:18:1, col:12> col:7 c1 'float' cinit
| `-ImplicitCastExpr 0x7f9f6d878810 <col:12> 'float' <AtomicToNonAtomic>
|   `-ImplicitCastExpr 0x7f9f6d8787f8 <col:12> '_Atomic(float)' <LValueToRValue>
|     `-DeclRefExpr 0x7f9f6d8787d0 <col:12> '_Atomic(float)' lvalue Var 0x7f9f6d827168 'i' '_Atomic(float)'
|-VarDecl 0x7f9f6d878840 <line:19:1, col:22> col:7 c2 'float' cinit
| `-ImplicitCastExpr 0x7f9f6d878988 <col:12, col:22> 'float' <IntegralToFloating>
|   `-ParenExpr 0x7f9f6d878968 <col:12, col:22> 'int'
|     `-ConditionalOperator 0x7f9f6d878938 <col:13, col:21> 'int'
|       |-ImplicitCastExpr 0x7f9f6d878920 <col:13> 'float' <AtomicToNonAtomic>
|       | `-ImplicitCastExpr 0x7f9f6d878908 <col:13> '_Atomic(float)' <LValueToRValue>
|       |   `-DeclRefExpr 0x7f9f6d8788a0 <col:13> '_Atomic(float)' lvalue Var 0x7f9f6d827168 'i' '_Atomic(float)'
|       |-IntegerLiteral 0x7f9f6d8788c8 <col:17> 'int' 1
|       `-IntegerLiteral 0x7f9f6d8788e8 <col:21> 'int' 2
|-VarDecl 0x7f9f6d879aa0 <line:20:1, col:17> col:7 invalid c3 'float [(int)i]'
|-VarDecl 0x7f9f6d879b18 <line:21:1, col:17> col:7 c4 'float' cinit
| `-ImplicitCastExpr 0x7f9f6d879c18 <col:12, col:17> 'float' <IntegralToFloating>
|   `-BinaryOperator 0x7f9f6d879bf0 <col:12, col:17> 'int' '||'
|     |-IntegerLiteral 0x7f9f6d879b78 <col:12> 'int' 0
|     `-ImplicitCastExpr 0x7f9f6d879bd8 <col:17> 'float' <AtomicToNonAtomic>
|       `-ImplicitCastExpr 0x7f9f6d879bc0 <col:17> '_Atomic(float)' <LValueToRValue>
|         `-DeclRefExpr 0x7f9f6d879b98 <col:17> '_Atomic(float)' lvalue Var 0x7f9f6d827168 'i' '_Atomic(float)'
|-VarDecl 0x7f9f6d879c48 <line:22:1, col:7> col:7 invalid c5 'float'
|-StaticAssertDecl 0x7f9f6d87a2f0 <line:24:1, col:61> col:1
| |-BinaryOperator 0x7f9f6d87a240 <col:17, col:55> 'int' '=='
| | |-GenericSelectionExpr 0x7f9f6d87a1c0 <col:17, col:50> 'int'
| | | |-ImplicitCastExpr 0x7f9f6d87a1a8 <col:27> 'float' <AtomicToNonAtomic>
| | | | `-ImplicitCastExpr 0x7f9f6d87a190 <col:27> '_Atomic(float)' <LValueToRValue>
| | | |   `-DeclRefExpr 0x7f9f6d87a110 <col:27> '_Atomic(float)' lvalue Var 0x7f9f6d827168 'i' '_Atomic(float)'
| | | |-IntegerLiteral 0x7f9f6d87a150 <col:37> 'int' 1
| | | `-IntegerLiteral 0x7f9f6d87a170 <col:49> 'int' 0
| | `-IntegerLiteral 0x7f9f6d87a220 <col:55> 'int' 1
| `-StringLiteral 0x7f9f6d87a2a8 <col:58> 'char [2]' lvalue "1"
|-StaticAssertDecl 0x7f9f6d87a528 <line:25:1, col:65> col:1
| |-BinaryOperator 0x7f9f6d87a4b8 <col:17, col:59> 'int' '=='
| | |-GenericSelectionExpr 0x7f9f6d87a438 <col:17, col:54> 'int'
| | | |-BinaryOperator 0x7f9f6d87a3b8 <col:27, col:31> 'float' '+'
| | | | |-ImplicitCastExpr 0x7f9f6d87a388 <col:27> 'float' <AtomicToNonAtomic>
| | | | | `-ImplicitCastExpr 0x7f9f6d87a370 <col:27> '_Atomic(float)' <LValueToRValue>
| | | | |   `-DeclRefExpr 0x7f9f6d87a328 <col:27> '_Atomic(float)' lvalue Var 0x7f9f6d827168 'i' '_Atomic(float)'
| | | | `-ImplicitCastExpr 0x7f9f6d87a3a0 <col:31> 'float' <IntegralToFloating>
| | | |   `-IntegerLiteral 0x7f9f6d87a350 <col:31> 'int' 1
| | | |-IntegerLiteral 0x7f9f6d87a3f8 <col:41> 'int' 1
| | | `-IntegerLiteral 0x7f9f6d87a418 <col:53> 'int' 0
| | `-IntegerLiteral 0x7f9f6d87a498 <col:59> 'int' 1
| `-StringLiteral 0x7f9f6d87a4e0 <col:62> 'char [2]' lvalue "2"
|-StaticAssertDecl 0x7f9f6d87a748 <line:26:1, col:65> col:1
| |-BinaryOperator 0x7f9f6d87a6d8 <col:17, col:59> 'int' '=='
| | |-GenericSelectionExpr 0x7f9f6d87a658 <col:17, col:54> 'int'
| | | |-BinaryOperator 0x7f9f6d87a5d8 <col:27, col:31> '_Atomic(float)' '='
| | | | |-DeclRefExpr 0x7f9f6d87a560 <col:27> '_Atomic(float)' lvalue Var 0x7f9f6d827168 'i' '_Atomic(float)'
| | | | `-ImplicitCastExpr 0x7f9f6d87a5c0 <col:31> '_Atomic(float)' <NonAtomicToAtomic>
| | | |   `-ImplicitCastExpr 0x7f9f6d87a5a8 <col:31> 'float' <IntegralToFloating>
| | | |     `-IntegerLiteral 0x7f9f6d87a588 <col:31> 'int' 0
| | | |-IntegerLiteral 0x7f9f6d87a618 <col:41> 'int' 1
| | | `-IntegerLiteral 0x7f9f6d87a638 <col:53> 'int' 0
| | `-IntegerLiteral 0x7f9f6d87a6b8 <col:59> 'int' 1
| `-StringLiteral 0x7f9f6d87a700 <col:62> 'char [2]' lvalue "3"
|-StaticAssertDecl 0x7f9f6d87a910 <line:27:1, col:63> col:1
| |-BinaryOperator 0x7f9f6d87a8a0 <col:17, col:57> 'int' '=='
| | |-GenericSelectionExpr 0x7f9f6d87a820 <col:17, col:52> 'int'
| | | |-UnaryOperator 0x7f9f6d87a7a8 <col:27, col:28> 'float' postfix '++'
| | | | `-DeclRefExpr 0x7f9f6d87a780 <col:27> '_Atomic(float)' lvalue Var 0x7f9f6d827168 'i' '_Atomic(float)'
| | | |-IntegerLiteral 0x7f9f6d87a7e0 <col:39> 'int' 1
| | | `-IntegerLiteral 0x7f9f6d87a800 <col:51> 'int' 0
| | `-IntegerLiteral 0x7f9f6d87a880 <col:57> 'int' 1
| `-StringLiteral 0x7f9f6d87a8c8 <col:60> 'char [2]' lvalue "4"
|-StaticAssertDecl 0x7f9f6d87aaf0 <line:28:1, col:63> col:1
| |-BinaryOperator 0x7f9f6d87aa80 <col:17, col:57> 'int' '=='
| | |-GenericSelectionExpr 0x7f9f6d87aa00 <col:17, col:52> 'int'
| | | |-UnaryOperator 0x7f9f6d87a970 <col:27, col:28> 'float' postfix '--'
| | | | `-DeclRefExpr 0x7f9f6d87a948 <col:27> '_Atomic(float)' lvalue Var 0x7f9f6d827168 'i' '_Atomic(float)'
| | | |-IntegerLiteral 0x7f9f6d87a9a8 <col:39> 'int' 1
| | | `-IntegerLiteral 0x7f9f6d87a9c8 <col:51> 'int' 0
| | `-IntegerLiteral 0x7f9f6d87aa60 <col:57> 'int' 1
| `-StringLiteral 0x7f9f6d87aaa8 <col:60> 'char [2]' lvalue "5"
|-VarDecl 0x7f9f6d87ab78 <line:30:1, col:18> col:13 used sz '_Atomic(int)' cinit
| `-ImplicitCastExpr 0x7f9f6d87abf8 <col:18> '_Atomic(int)' <NonAtomicToAtomic>
|   `-IntegerLiteral 0x7f9f6d87abd8 <col:18> 'int' 2
|-FunctionDecl 0x7f9f6d87ae70 <line:31:1, col:26> col:6 fn1 'void (float *)'
| `-ParmVarDecl 0x7f9f6d87ad70 <col:11, col:25> col:17 a 'float *':'float *'
|-FunctionDecl 0x7f9f6d87b070 <line:32:1, col:26> col:6 invalid fn2 'void (int)'
| `-ParmVarDecl 0x7f9f6d87afb0 <col:11, col:17> col:17 invalid a 'int'
|-FunctionDecl 0x7f9f6d87b2b0 <line:33:1, col:24> col:6 fn3 'void (float *)'
| `-ParmVarDecl 0x7f9f6d87b1f0 <col:11, col:23> col:17 a 'float *':'float *'
|-FunctionDecl 0x7f9f6d87b4f0 <line:34:1, col:24> col:6 fn4 'void (float *)'
| `-ParmVarDecl 0x7f9f6d87b430 <col:11, col:23> col:17 a 'float *':'float *'
`-FunctionDecl 0x7f9f6d87b760 <line:35:1, col:23> col:6 fn5 'void (float *)'
  `-ParmVarDecl 0x7f9f6d87b6a0 <col:11, col:22> col:17 a 'float *':'float *'
