c_files/pr71514.c:15:26: warning: incompatible pointer types passing 'void (*)()' to parameter of type 'int *' [-Wincompatible-pointer-types]
  __atomic_exchange (&a, &foo, &b, __ATOMIC_RELAXED); /* { dg-error "must not be a pointer to a function" } */
                         ^~~~
c_files/pr71514.c:22:26: warning: incompatible pointer types passing 'int (*)[n]' to parameter of type 'int *' [-Wincompatible-pointer-types]
  __atomic_exchange (&a, &arr, &b, __ATOMIC_RELAXED); /* { dg-error "must be a pointer to a constant size type" } */
                         ^~~~
2 warnings generated.
TranslationUnitDecl 0x7fc80901f0d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fc80901f620 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fc80901f340 '__int128'
|-TypedefDecl 0x7fc80901f690 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fc80901f360 'unsigned __int128'
|-TypedefDecl 0x7fc80901f968 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fc80901f770 'struct __NSConstantString_tag'
|   `-Record 0x7fc80901f6e8 '__NSConstantString_tag'
|-TypedefDecl 0x7fc80901fa00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fc80901f9c0 'char *'
|   `-BuiltinType 0x7fc80901f160 'char'
|-TypedefDecl 0x7fc80901fcc8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fc80901fc70 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fc80901fae0 'struct __va_list_tag'
|     `-Record 0x7fc80901fa58 '__va_list_tag'
|-FunctionDecl 0x7fc80901fd68 <c_files/pr71514.c:5:1, line:8:1> line:6:1 used foo 'void ()'
| `-CompoundStmt 0x7fc80a025a40 <line:7:1, line:8:1>
|-VarDecl 0x7fc80a025a70 <line:10:1, col:5> col:5 used a 'int'
|-VarDecl 0x7fc80a025ae8 <col:1, col:8> col:8 used b 'int'
|-FunctionDecl 0x7fc80a025c20 <line:12:1, line:16:1> line:13:1 fn1 'void (void)'
| `-CompoundStmt 0x7fc80a026340 <line:14:1, line:16:1>
|   `-AtomicExpr 0x7fc80a0262f0 <line:15:3, col:52> 'void'
|     |-UnaryOperator 0x7fc80a025dc8 <col:22, col:23> 'int *' prefix '&'
|     | `-DeclRefExpr 0x7fc80a025d78 <col:23> 'int' lvalue Var 0x7fc80a025a70 'a' 'int'
|     |-IntegerLiteral 0x7fc80a025ea0 <<built-in>:11:26> 'int' 0
|     |-ImplicitCastExpr 0x7fc80a025f20 <c_files/pr71514.c:15:26, col:27> 'int *' <BitCast>
|     | `-UnaryOperator 0x7fc80a025e38 <col:26, col:27> 'void (*)()' prefix '&'
|     |   `-DeclRefExpr 0x7fc80a025de8 <col:27> 'void ()' Function 0x7fc80901fd68 'foo' 'void ()'
|     `-UnaryOperator 0x7fc80a025e80 <col:32, col:33> 'int *' prefix '&'
|       `-DeclRefExpr 0x7fc80a025e58 <col:33> 'int' lvalue Var 0x7fc80a025ae8 'b' 'int'
|-FunctionDecl 0x7fc80a025cb8 <col:3> col:3 implicit used __atomic_exchange 'void ()' extern
`-FunctionDecl 0x7fc80a026440 <line:18:1, line:23:1> line:19:1 fn2 'void (int)'
  |-ParmVarDecl 0x7fc80a026378 <col:6, col:10> col:10 used n 'int'
  `-CompoundStmt 0x7fc80a026808 <line:20:1, line:23:1>
    |-DeclStmt 0x7fc80a0265e0 <line:21:3, col:13>
    | `-VarDecl 0x7fc80a026580 <col:3, col:12> col:7 used arr 'int [n]'
    `-AtomicExpr 0x7fc80a0267b8 <line:22:3, col:52> 'void'
      |-UnaryOperator 0x7fc80a026648 <col:22, col:23> 'int *' prefix '&'
      | `-DeclRefExpr 0x7fc80a026620 <col:23> 'int' lvalue Var 0x7fc80a025a70 'a' 'int'
      |-IntegerLiteral 0x7fc80a026720 <<built-in>:11:26> 'int' 0
      |-ImplicitCastExpr 0x7fc80a0267a0 <c_files/pr71514.c:22:26, col:27> 'int *' <BitCast>
      | `-UnaryOperator 0x7fc80a0266b8 <col:26, col:27> 'int (*)[n]' prefix '&'
      |   `-DeclRefExpr 0x7fc80a026668 <col:27> 'int [n]' lvalue Var 0x7fc80a026580 'arr' 'int [n]'
      `-UnaryOperator 0x7fc80a026700 <col:32, col:33> 'int *' prefix '&'
        `-DeclRefExpr 0x7fc80a0266d8 <col:33> 'int' lvalue Var 0x7fc80a025ae8 'b' 'int'
