c_files/pr71265.c:5:11: warning: predefined identifier is only valid inside function [-Wpredefined-identifier-outside-function]
  int ID [__func__]; /* { dg-error "size of array .ID. has non-integer type" } */
          ^
c_files/pr71265.c:5:11: error: size of array has non-integer type 'const char [1]'
  int ID [__func__]; /* { dg-error "size of array .ID. has non-integer type" } */
          ^~~~~~~~
1 warning and 1 error generated.
TranslationUnitDecl 0x7fe1df8264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fe1df826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fe1df826740 '__int128'
|-TypedefDecl 0x7fe1df826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fe1df826760 'unsigned __int128'
|-TypedefDecl 0x7fe1df826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fe1df826b70 'struct __NSConstantString_tag'
|   `-Record 0x7fe1df826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fe1df826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fe1df826dc0 'char *'
|   `-BuiltinType 0x7fe1df826560 'char'
|-TypedefDecl 0x7fe1df8270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fe1df827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fe1df826ee0 'struct __va_list_tag'
|     `-Record 0x7fe1df826e58 '__va_list_tag'
`-FunctionDecl 0x7fe1df877890 <c_files/pr71265.c:4:1, line:7:1> line:4:6 invalid ID 'void (int)'
  |-ParmVarDecl 0x7fe1df8777c8 <line:5:3, col:7> col:7 invalid ID 'int'
  `-CompoundStmt 0x7fe1df877930 <line:6:1, line:7:1>
