c_files/pr71574.c:5:32: warning: 'alloc_align' attribute only applies to return values that are pointers or references [-Wignored-attributes]
int fn2 (void) __attribute__ ((alloc_align (fn1))); /* { dg-warning "parameter outside range" } */
~~~                            ^~~~~~~~~~~~~~~~~
c_files/pr71574.c:6:32: warning: 'alloc_size' attribute only applies to return values that are pointers [-Wignored-attributes]
int fn3 (void) __attribute__ ((alloc_size (fn1))); /* { dg-warning "parameter outside range" } */
                               ^
c_files/pr71574.c:7:32: warning: 'assume_aligned' attribute only applies to return values that are pointers or references [-Wignored-attributes]
int fn4 (void) __attribute__ ((assume_aligned (fn1))); /* { dg-warning "not integer constant" } */
~~~                            ^~~~~~~~~~~~~~~~~~~~
c_files/pr71574.c:8:41: error: 'nonnull' attribute requires parameter 1 to be an integer constant
int fn5 (char *, char *) __attribute__((nonnull (fn1))); /* { dg-error "nonnull argument has invalid operand" } */
                                        ^        ~~~
c_files/pr71574.c:9:45: error: 'sentinel' attribute requires parameter 1 to be an integer constant
int fn6 (const char *, ...) __attribute__ ((sentinel (fn1))); /* { dg-warning "not an integer constant" } */
                                            ^         ~~~
c_files/pr71574.c:11:28: error: 'vector_size' attribute requires an integer constant
typedef int __attribute__((vector_size (fn1))) v4si; /* { dg-warning "attribute ignored" } */
                           ^            ~~~
c_files/pr71574.c:12:39: error: 'aligned' attribute requires integer constant
typedef int T __attribute__((aligned (fn1))); /* { dg-error "requested alignment is not" } */
                                      ^~~
3 warnings and 4 errors generated.
TranslationUnitDecl 0x7f8d62018ed0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7f8d62019420 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7f8d62019140 '__int128'
|-TypedefDecl 0x7f8d62019490 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7f8d62019160 'unsigned __int128'
|-TypedefDecl 0x7f8d62019768 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7f8d62019570 'struct __NSConstantString_tag'
|   `-Record 0x7f8d620194e8 '__NSConstantString_tag'
|-TypedefDecl 0x7f8d62019800 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7f8d620197c0 'char *'
|   `-BuiltinType 0x7f8d62018f60 'char'
|-TypedefDecl 0x7f8d62019ac8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7f8d62019a70 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7f8d620198e0 'struct __va_list_tag'
|     `-Record 0x7f8d62019858 '__va_list_tag'
|-FunctionDecl 0x7f8d62825a00 <c_files/pr71574.c:4:1, col:14> col:5 used fn1 'int (void)'
|-FunctionDecl 0x7f8d62825b98 <line:5:1, col:50> col:5 fn2 'int (void)'
|-FunctionDecl 0x7f8d62825cf0 <line:6:1, col:49> col:5 fn3 'int (void)'
|-FunctionDecl 0x7f8d62825e48 <line:7:1, col:53> col:5 fn4 'int (void)'
|-FunctionDecl 0x7f8d62826078 <line:8:1, col:55> col:5 fn5 'int (char *, char *)'
| |-ParmVarDecl 0x7f8d62825ef8 <col:10, col:15> col:16 'char *'
| `-ParmVarDecl 0x7f8d62825f70 <col:18, col:23> col:24 'char *'
|-FunctionDecl 0x7f8d62826228 <line:9:1, col:60> col:5 fn6 'int (const char *, ...)'
| `-ParmVarDecl 0x7f8d62826138 <col:10, col:21> col:22 'const char *'
|-TypedefDecl 0x7f8d62826308 <line:11:1, col:48> col:48 v4si 'int'
| `-BuiltinType 0x7f8d62018fc0 'int'
`-TypedefDecl 0x7f8d628263a0 <line:12:1, col:13> col:13 T 'int'
  `-BuiltinType 0x7f8d62018fc0 'int'
