c_files/pr71573.c:13:3: warning: use of out-of-scope declaration of 't'
  t (0); /* { dg-error "called object .t. is not a function or function pointer" } */
  ^
c_files/pr71573.c:7:14: note: previous declaration is here
  extern int t;
             ^
c_files/pr71573.c:13:5: error: called object type 'int' is not a function or function pointer
  t (0); /* { dg-error "called object .t. is not a function or function pointer" } */
  ~ ^
1 warning and 1 error generated.
TranslationUnitDecl 0x7f95e80264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7f95e8026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7f95e8026740 '__int128'
|-TypedefDecl 0x7f95e8026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7f95e8026760 'unsigned __int128'
|-TypedefDecl 0x7f95e8026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7f95e8026b70 'struct __NSConstantString_tag'
|   `-Record 0x7f95e8026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7f95e8026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7f95e8026dc0 'char *'
|   `-BuiltinType 0x7f95e8026560 'char'
|-TypedefDecl 0x7f95e80270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7f95e8027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7f95e8026ee0 'struct __va_list_tag'
|     `-Record 0x7f95e8026e58 '__va_list_tag'
|-FunctionDecl 0x7f95e8077400 <c_files/pr71573.c:4:1, line:8:1> line:5:1 f1 'void (void)'
| `-CompoundStmt 0x7f95e8077578 <line:6:1, line:8:1>
|   `-DeclStmt 0x7f95e8077560 <line:7:3, col:15>
|     `-VarDecl 0x7f95e80774f0 parent 0x7f95e80264d0 <col:3, col:14> col:14 used t 'int' extern
`-FunctionDecl 0x7f95e8077628 <line:10:1, line:14:1> line:11:1 f2 'void (void)'
  `-CompoundStmt 0x7f95e8077b00 <line:12:1, line:14:1>
