c_files/pr77624-1.c:7:36: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'const volatile void *' [-Wint-conversion]
  return __atomic_is_lock_free (2, a);          /* { dg-warning "pointer from integer" "" { target c } } */
                                   ^
c_files/pr77624-1.c:13:40: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'const volatile void *' [-Wint-conversion]
  return __atomic_always_lock_free (2, a);      /* { dg-warning "pointer from integer" "" { target c } } */
                                       ^
2 warnings generated.
TranslationUnitDecl 0x7fba248264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fba24826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fba24826740 '__int128'
|-TypedefDecl 0x7fba24826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fba24826760 'unsigned __int128'
|-TypedefDecl 0x7fba24826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fba24826b70 'struct __NSConstantString_tag'
|   `-Record 0x7fba24826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fba24826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fba24826dc0 'char *'
|   `-BuiltinType 0x7fba24826560 'char'
|-TypedefDecl 0x7fba248270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fba24827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fba24826ee0 'struct __va_list_tag'
|     `-Record 0x7fba24826e58 '__va_list_tag'
|-FunctionDecl 0x7fba25025a30 <c_files/pr77624-1.c:4:1, line:8:1> line:5:1 foo 'int (int)'
| |-ParmVarDecl 0x7fba24827138 <col:6, col:10> col:10 used a 'int'
| `-CompoundStmt 0x7fba25026278 <line:6:1, line:8:1>
|   `-ReturnStmt 0x7fba25026260 <line:7:3, col:37>
|     `-CallExpr 0x7fba25025e00 <col:10, col:37> 'int'
|       |-ImplicitCastExpr 0x7fba25025de8 <col:10> 'int (*)(unsigned long, const volatile void *)' <BuiltinFnToFnPtr>
|       | `-DeclRefExpr 0x7fba25025d30 <col:10> '<builtin fn type>' Function 0x7fba25025b80 '__atomic_is_lock_free' 'int (unsigned long, const volatile void *)'
|       |-ImplicitCastExpr 0x7fba25025e38 <col:33> 'unsigned long' <IntegralCast>
|       | `-IntegerLiteral 0x7fba25025d58 <col:33> 'int' 2
|       `-ImplicitCastExpr 0x7fba25025e68 <col:36> 'const volatile void *' <IntegralToPointer>
|         `-ImplicitCastExpr 0x7fba25025e50 <col:36> 'int' <LValueToRValue>
|           `-DeclRefExpr 0x7fba25025d78 <col:36> 'int' lvalue ParmVar 0x7fba24827138 'a' 'int'
|-FunctionDecl 0x7fba25025b80 <col:10> col:10 implicit used __atomic_is_lock_free 'int (unsigned long, const volatile void *)' extern
| |-ParmVarDecl 0x7fba25025c18 <<invalid sloc>> <invalid sloc> 'unsigned long'
| |-ParmVarDecl 0x7fba25025c78 <<invalid sloc>> <invalid sloc> 'const volatile void *'
| `-NoThrowAttr 0x7fba25025ce8 <col:10> Implicit
|-FunctionDecl 0x7fba25026340 <line:10:1, line:14:1> line:11:1 bar 'int (int)'
| |-ParmVarDecl 0x7fba250262b0 <col:6, col:10> col:10 used a 'int'
| `-CompoundStmt 0x7fba250266c8 <line:12:1, line:14:1>
|   `-ReturnStmt 0x7fba250266b0 <line:13:3, col:41>
|     `-CallExpr 0x7fba25026630 <col:10, col:41> 'int'
|       |-ImplicitCastExpr 0x7fba25026618 <col:10> 'int (*)(unsigned long, const volatile void *)' <BuiltinFnToFnPtr>
|       | `-DeclRefExpr 0x7fba25026590 <col:10> '<builtin fn type>' Function 0x7fba250263e0 '__atomic_always_lock_free' 'int (unsigned long, const volatile void *)'
|       |-ImplicitCastExpr 0x7fba25026668 <col:37> 'unsigned long' <IntegralCast>
|       | `-IntegerLiteral 0x7fba250265b8 <col:37> 'int' 2
|       `-ImplicitCastExpr 0x7fba25026698 <col:40> 'const volatile void *' <IntegralToPointer>
|         `-ImplicitCastExpr 0x7fba25026680 <col:40> 'int' <LValueToRValue>
|           `-DeclRefExpr 0x7fba250265d8 <col:40> 'int' lvalue ParmVar 0x7fba250262b0 'a' 'int'
`-FunctionDecl 0x7fba250263e0 <col:10> col:10 implicit used __atomic_always_lock_free 'int (unsigned long, const volatile void *)' extern
  |-ParmVarDecl 0x7fba25026478 <<invalid sloc>> <invalid sloc> 'unsigned long'
  |-ParmVarDecl 0x7fba250264d8 <<invalid sloc>> <invalid sloc> 'const volatile void *'
  `-NoThrowAttr 0x7fba25026548 <col:10> Implicit
