c_files/pr80919.c:9:28: warning: format specifies type 'int' but the argument has type 'int (*)[0]' [-Wformat]
  __builtin_printf("%d\n", &a); /* { dg-warning "expects argument of type" } */
                    ~~     ^~
c_files/pr80919.c:10:28: warning: format specifies type 'int' but the argument has type 'int (*)[0]' [-Wformat]
  __builtin_printf("%i\n", &a); /* { dg-warning "expects argument of type" } */
                    ~~     ^~
c_files/pr80919.c:12:28: warning: format specifies type 'unsigned int' but the argument has type 'int (*)[0]' [-Wformat]
  __builtin_printf("%o\n", &a); /* { dg-warning "expects argument of type" } */
                    ~~     ^~
c_files/pr80919.c:13:28: warning: format specifies type 'unsigned int' but the argument has type 'int (*)[0]' [-Wformat]
  __builtin_printf("%u\n", &a); /* { dg-warning "expects argument of type" } */
                    ~~     ^~
c_files/pr80919.c:14:28: warning: format specifies type 'unsigned int' but the argument has type 'int (*)[0]' [-Wformat]
  __builtin_printf("%x\n", &a); /* { dg-warning "expects argument of type" } */
                    ~~     ^~
c_files/pr80919.c:15:28: warning: format specifies type 'unsigned int' but the argument has type 'int (*)[0]' [-Wformat]
  __builtin_printf("%X\n", &a); /* { dg-warning "expects argument of type" } */
                    ~~     ^~
6 warnings generated.
TranslationUnitDecl 0x7fe8850264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fe885026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fe885026740 '__int128'
|-TypedefDecl 0x7fe885026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fe885026760 'unsigned __int128'
|-TypedefDecl 0x7fe885026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fe885026b70 'struct __NSConstantString_tag'
|   `-Record 0x7fe885026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fe885026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fe885026dc0 'char *'
|   `-BuiltinType 0x7fe885026560 'char'
|-TypedefDecl 0x7fe8850270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fe885027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fe885026ee0 'struct __va_list_tag'
|     `-Record 0x7fe885026e58 '__va_list_tag'
|-FunctionDecl 0x7fe885825a00 <c_files/pr80919.c:5:1, line:16:1> line:6:1 fn 'void (void)'
| `-CompoundStmt 0x7fe885826880 <line:7:1, line:16:1>
|   |-DeclStmt 0x7fe885825bc0 <line:8:3, col:11>
|   | `-VarDecl 0x7fe885825b60 <col:3, col:10> col:7 used a 'int [0]'
|   |-CallExpr 0x7fe885825ec0 <line:9:3, col:30> 'int'
|   | |-ImplicitCastExpr 0x7fe885825ea8 <col:3> 'int (*)(const char *, ...)' <BuiltinFnToFnPtr>
|   | | `-DeclRefExpr 0x7fe885825d70 <col:3> '<builtin fn type>' Function 0x7fe885825c18 '__builtin_printf' 'int (const char *, ...)'
|   | |-ImplicitCastExpr 0x7fe885825f10 <col:20> 'const char *' <BitCast>
|   | | `-ImplicitCastExpr 0x7fe885825ef8 <col:20> 'char *' <ArrayToPointerDecay>
|   | |   `-StringLiteral 0x7fe885825dd8 <col:20> 'char [4]' lvalue "%d\n"
|   | `-UnaryOperator 0x7fe885825e58 <col:28, col:29> 'int (*)[0]' prefix '&'
|   |   `-DeclRefExpr 0x7fe885825e08 <col:29> 'int [0]' lvalue Var 0x7fe885825b60 'a' 'int [0]'
|   |-CallExpr 0x7fe885826398 <line:10:3, col:30> 'int'
|   | |-ImplicitCastExpr 0x7fe885826380 <col:3> 'int (*)(const char *, ...)' <BuiltinFnToFnPtr>
|   | | `-DeclRefExpr 0x7fe8858262e0 <col:3> '<builtin fn type>' Function 0x7fe885825c18 '__builtin_printf' 'int (const char *, ...)'
|   | |-ImplicitCastExpr 0x7fe8858263e8 <col:20> 'const char *' <BitCast>
|   | | `-ImplicitCastExpr 0x7fe8858263d0 <col:20> 'char *' <ArrayToPointerDecay>
|   | |   `-StringLiteral 0x7fe885826308 <col:20> 'char [4]' lvalue "%i\n"
|   | `-UnaryOperator 0x7fe885826360 <col:28, col:29> 'int (*)[0]' prefix '&'
|   |   `-DeclRefExpr 0x7fe885826338 <col:29> 'int [0]' lvalue Var 0x7fe885825b60 'a' 'int [0]'
|   |-CallExpr 0x7fe8858264b8 <line:12:3, col:30> 'int'
|   | |-ImplicitCastExpr 0x7fe8858264a0 <col:3> 'int (*)(const char *, ...)' <BuiltinFnToFnPtr>
|   | | `-DeclRefExpr 0x7fe885826400 <col:3> '<builtin fn type>' Function 0x7fe885825c18 '__builtin_printf' 'int (const char *, ...)'
|   | |-ImplicitCastExpr 0x7fe885826508 <col:20> 'const char *' <BitCast>
|   | | `-ImplicitCastExpr 0x7fe8858264f0 <col:20> 'char *' <ArrayToPointerDecay>
|   | |   `-StringLiteral 0x7fe885826428 <col:20> 'char [4]' lvalue "%o\n"
|   | `-UnaryOperator 0x7fe885826480 <col:28, col:29> 'int (*)[0]' prefix '&'
|   |   `-DeclRefExpr 0x7fe885826458 <col:29> 'int [0]' lvalue Var 0x7fe885825b60 'a' 'int [0]'
|   |-CallExpr 0x7fe8858265d8 <line:13:3, col:30> 'int'
|   | |-ImplicitCastExpr 0x7fe8858265c0 <col:3> 'int (*)(const char *, ...)' <BuiltinFnToFnPtr>
|   | | `-DeclRefExpr 0x7fe885826520 <col:3> '<builtin fn type>' Function 0x7fe885825c18 '__builtin_printf' 'int (const char *, ...)'
|   | |-ImplicitCastExpr 0x7fe885826628 <col:20> 'const char *' <BitCast>
|   | | `-ImplicitCastExpr 0x7fe885826610 <col:20> 'char *' <ArrayToPointerDecay>
|   | |   `-StringLiteral 0x7fe885826548 <col:20> 'char [4]' lvalue "%u\n"
|   | `-UnaryOperator 0x7fe8858265a0 <col:28, col:29> 'int (*)[0]' prefix '&'
|   |   `-DeclRefExpr 0x7fe885826578 <col:29> 'int [0]' lvalue Var 0x7fe885825b60 'a' 'int [0]'
|   |-CallExpr 0x7fe8858266f8 <line:14:3, col:30> 'int'
|   | |-ImplicitCastExpr 0x7fe8858266e0 <col:3> 'int (*)(const char *, ...)' <BuiltinFnToFnPtr>
|   | | `-DeclRefExpr 0x7fe885826640 <col:3> '<builtin fn type>' Function 0x7fe885825c18 '__builtin_printf' 'int (const char *, ...)'
|   | |-ImplicitCastExpr 0x7fe885826748 <col:20> 'const char *' <BitCast>
|   | | `-ImplicitCastExpr 0x7fe885826730 <col:20> 'char *' <ArrayToPointerDecay>
|   | |   `-StringLiteral 0x7fe885826668 <col:20> 'char [4]' lvalue "%x\n"
|   | `-UnaryOperator 0x7fe8858266c0 <col:28, col:29> 'int (*)[0]' prefix '&'
|   |   `-DeclRefExpr 0x7fe885826698 <col:29> 'int [0]' lvalue Var 0x7fe885825b60 'a' 'int [0]'
|   `-CallExpr 0x7fe885826818 <line:15:3, col:30> 'int'
|     |-ImplicitCastExpr 0x7fe885826800 <col:3> 'int (*)(const char *, ...)' <BuiltinFnToFnPtr>
|     | `-DeclRefExpr 0x7fe885826760 <col:3> '<builtin fn type>' Function 0x7fe885825c18 '__builtin_printf' 'int (const char *, ...)'
|     |-ImplicitCastExpr 0x7fe885826868 <col:20> 'const char *' <BitCast>
|     | `-ImplicitCastExpr 0x7fe885826850 <col:20> 'char *' <ArrayToPointerDecay>
|     |   `-StringLiteral 0x7fe885826788 <col:20> 'char [4]' lvalue "%X\n"
|     `-UnaryOperator 0x7fe8858267e0 <col:28, col:29> 'int (*)[0]' prefix '&'
|       `-DeclRefExpr 0x7fe8858267b8 <col:29> 'int [0]' lvalue Var 0x7fe885825b60 'a' 'int [0]'
`-FunctionDecl 0x7fe885825c18 <line:9:3> col:3 implicit used __builtin_printf 'int (const char *, ...)' extern
  |-ParmVarDecl 0x7fe885825cb0 <<invalid sloc>> <invalid sloc> 'const char *'
  `-FormatAttr 0x7fe885825d18 <col:3> Implicit printf 1 2
