c_files/pr82596.c:16:6: warning: array index -9223372036854775808 is before the beginning of the array [-Warray-bounds]
  T (""[SSIZE_MIN]);                      /* { dg-warning "below array bounds" "string" { xfail lp64 } } */
  ~~~^~~~~~~~~~~~~~
c_files/pr82596.c:12:22: note: expanded from macro 'T'
#define T(arg) sink (arg)
                     ^~~
c_files/pr82596.c:17:6: warning: array index -9223372036854775807 is before the beginning of the array [-Warray-bounds]
  T (""[SSIZE_MIN + 1]);                  /* { dg-warning "below array bounds" "string" } */
  ~~~^~~~~~~~~~~~~~~~~~
c_files/pr82596.c:12:22: note: expanded from macro 'T'
#define T(arg) sink (arg)
                     ^~~
c_files/pr82596.c:18:6: warning: array index -1 is before the beginning of the array [-Warray-bounds]
  T (""[-1]);                             /* { dg-warning "below array bounds" "string" } */
     ^  ~~
c_files/pr82596.c:12:22: note: expanded from macro 'T'
#define T(arg) sink (arg)
                     ^~~
c_files/pr82596.c:20:6: warning: array index 1 is past the end of the array (which contains 1 element) [-Warray-bounds]
  T (""[1]);                              /* { dg-warning "above array bounds" "string" } */
     ^  ~
c_files/pr82596.c:12:22: note: expanded from macro 'T'
#define T(arg) sink (arg)
                     ^~~
c_files/pr82596.c:21:6: warning: array index 2 is past the end of the array (which contains 2 elements) [-Warray-bounds]
  T ("0"[2]);                             /* { dg-warning "above array bounds" "string" } */
     ^   ~
c_files/pr82596.c:12:22: note: expanded from macro 'T'
#define T(arg) sink (arg)
                     ^~~
c_files/pr82596.c:24:6: warning: array index 4 is past the end of the array (which contains 4 elements) [-Warray-bounds]
  T ("012"[4]);                           /* { dg-warning "above array bounds" "string" } */
     ^     ~
c_files/pr82596.c:12:22: note: expanded from macro 'T'
#define T(arg) sink (arg)
                     ^~~
c_files/pr82596.c:25:6: warning: array index 9223372036854775807 is past the end of the array (which contains 5 elements) [-Warray-bounds]
  T ("0123"[SSIZE_MAX]);                  /* { dg-warning "above array bounds" "string" } */
     ^      ~~~~~~~~~
c_files/pr82596.c:12:22: note: expanded from macro 'T'
#define T(arg) sink (arg)
                     ^~~
c_files/pr82596.c:26:6: warning: array index -1 is past the end of the array (which contains 5 elements) [-Warray-bounds]
  T ("0123"[SIZE_MAX]);                   /* { dg-warning "above array bounds" "string" } */
     ^      ~~~~~~~~
c_files/pr82596.c:12:22: note: expanded from macro 'T'
#define T(arg) sink (arg)
                     ^~~
8 warnings generated.
TranslationUnitDecl 0x7fa9ab0264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fa9ab026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fa9ab026740 '__int128'
|-TypedefDecl 0x7fa9ab026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fa9ab026760 'unsigned __int128'
|-TypedefDecl 0x7fa9ab026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fa9ab026b70 'struct __NSConstantString_tag'
|   `-Record 0x7fa9ab026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fa9ab026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fa9ab026dc0 'char *'
|   `-BuiltinType 0x7fa9ab026560 'char'
|-TypedefDecl 0x7fa9ab0270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fa9ab027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fa9ab026ee0 'struct __va_list_tag'
|     `-Record 0x7fa9ab026e58 '__va_list_tag'
|-FunctionDecl 0x7fa9ab826a00 <c_files/pr82596.c:10:1, col:20> col:6 used sink 'void (int, ...)'
| `-ParmVarDecl 0x7fa9ab027138 <col:12> col:15 'int'
`-FunctionDecl 0x7fa9ab826bc0 <line:14:1, line:27:1> line:14:6 test_cststring 'void (int)'
  |-ParmVarDecl 0x7fa9ab826af8 <col:22, col:26> col:26 i 'int'
  `-CompoundStmt 0x7fa9ab827c38 <line:15:1, line:27:1>
    |-CallExpr 0x7fa9ab826e30 <line:12:16, col:25> 'void'
    | |-ImplicitCastExpr 0x7fa9ab826e18 <col:16> 'void (*)(int, ...)' <FunctionToPointerDecay>
    | | `-DeclRefExpr 0x7fa9ab826c60 <col:16> 'void (int, ...)' Function 0x7fa9ab826a00 'sink' 'void (int, ...)'
    | `-ImplicitCastExpr 0x7fa9ab826e78 <line:16:6, col:18> 'int' <IntegralCast>
    |   `-ImplicitCastExpr 0x7fa9ab826e60 <col:6, col:18> 'char' <LValueToRValue>
    |     `-ArraySubscriptExpr 0x7fa9ab826dc8 <col:6, col:18> 'char' lvalue
    |       |-ImplicitCastExpr 0x7fa9ab826db0 <col:6> 'char *' <ArrayToPointerDecay>
    |       | `-StringLiteral 0x7fa9ab826cc8 <col:6> 'char [1]' lvalue ""
    |       `-ParenExpr 0x7fa9ab826d90 <line:8:19, col:34> 'long'
    |         `-BinaryOperator 0x7fa9ab826d68 <col:20, col:33> 'long' '-'
    |           |-UnaryOperator 0x7fa9ab826d10 <col:20, <built-in>:40:25> 'long' prefix '-'
    |           | `-IntegerLiteral 0x7fa9ab826cf0 <col:25> 'long' 9223372036854775807
    |           `-ImplicitCastExpr 0x7fa9ab826d50 <c_files/pr82596.c:8:33> 'long' <IntegralCast>
    |             `-IntegerLiteral 0x7fa9ab826d30 <col:33> 'int' 1
    |-CallExpr 0x7fa9ab827058 <line:12:16, col:25> 'void'
    | |-ImplicitCastExpr 0x7fa9ab827040 <col:16> 'void (*)(int, ...)' <FunctionToPointerDecay>
    | | `-DeclRefExpr 0x7fa9ab826e90 <col:16> 'void (int, ...)' Function 0x7fa9ab826a00 'sink' 'void (int, ...)'
    | `-ImplicitCastExpr 0x7fa9ab8270a0 <line:17:6, col:22> 'int' <IntegralCast>
    |   `-ImplicitCastExpr 0x7fa9ab827088 <col:6, col:22> 'char' <LValueToRValue>
    |     `-ArraySubscriptExpr 0x7fa9ab827018 <col:6, col:22> 'char' lvalue
    |       |-ImplicitCastExpr 0x7fa9ab827000 <col:6> 'char *' <ArrayToPointerDecay>
    |       | `-StringLiteral 0x7fa9ab826eb8 <col:6> 'char [1]' lvalue ""
    |       `-BinaryOperator 0x7fa9ab826fd8 <line:8:19, line:17:21> 'long' '+'
    |         |-ParenExpr 0x7fa9ab826f80 <line:8:19, col:34> 'long'
    |         | `-BinaryOperator 0x7fa9ab826f58 <col:20, col:33> 'long' '-'
    |         |   |-UnaryOperator 0x7fa9ab826f00 <col:20, <built-in>:40:25> 'long' prefix '-'
    |         |   | `-IntegerLiteral 0x7fa9ab826ee0 <col:25> 'long' 9223372036854775807
    |         |   `-ImplicitCastExpr 0x7fa9ab826f40 <c_files/pr82596.c:8:33> 'long' <IntegralCast>
    |         |     `-IntegerLiteral 0x7fa9ab826f20 <col:33> 'int' 1
    |         `-ImplicitCastExpr 0x7fa9ab826fc0 <line:17:21> 'long' <IntegralCast>
    |           `-IntegerLiteral 0x7fa9ab826fa0 <col:21> 'int' 1
    |-CallExpr 0x7fa9ab8271a0 <line:12:16, col:25> 'void'
    | |-ImplicitCastExpr 0x7fa9ab827188 <col:16> 'void (*)(int, ...)' <FunctionToPointerDecay>
    | | `-DeclRefExpr 0x7fa9ab8270b8 <col:16> 'void (int, ...)' Function 0x7fa9ab826a00 'sink' 'void (int, ...)'
    | `-ImplicitCastExpr 0x7fa9ab8271e8 <line:18:6, col:11> 'int' <IntegralCast>
    |   `-ImplicitCastExpr 0x7fa9ab8271d0 <col:6, col:11> 'char' <LValueToRValue>
    |     `-ArraySubscriptExpr 0x7fa9ab827160 <col:6, col:11> 'char' lvalue
    |       |-ImplicitCastExpr 0x7fa9ab827148 <col:6> 'char *' <ArrayToPointerDecay>
    |       | `-StringLiteral 0x7fa9ab8270e0 <col:6> 'char [1]' lvalue ""
    |       `-UnaryOperator 0x7fa9ab827128 <col:9, col:10> 'int' prefix '-'
    |         `-IntegerLiteral 0x7fa9ab827108 <col:10> 'int' 1
    |-CallExpr 0x7fa9ab8272c8 <line:12:16, col:25> 'void'
    | |-ImplicitCastExpr 0x7fa9ab8272b0 <col:16> 'void (*)(int, ...)' <FunctionToPointerDecay>
    | | `-DeclRefExpr 0x7fa9ab827200 <col:16> 'void (int, ...)' Function 0x7fa9ab826a00 'sink' 'void (int, ...)'
    | `-ImplicitCastExpr 0x7fa9ab827310 <line:19:6, col:10> 'int' <IntegralCast>
    |   `-ImplicitCastExpr 0x7fa9ab8272f8 <col:6, col:10> 'char' <LValueToRValue>
    |     `-ArraySubscriptExpr 0x7fa9ab827288 <col:6, col:10> 'char' lvalue
    |       |-ImplicitCastExpr 0x7fa9ab827270 <col:6> 'char *' <ArrayToPointerDecay>
    |       | `-StringLiteral 0x7fa9ab827228 <col:6> 'char [1]' lvalue ""
    |       `-IntegerLiteral 0x7fa9ab827250 <col:9> 'int' 0
    |-CallExpr 0x7fa9ab8273f0 <line:12:16, col:25> 'void'
    | |-ImplicitCastExpr 0x7fa9ab8273d8 <col:16> 'void (*)(int, ...)' <FunctionToPointerDecay>
    | | `-DeclRefExpr 0x7fa9ab827328 <col:16> 'void (int, ...)' Function 0x7fa9ab826a00 'sink' 'void (int, ...)'
    | `-ImplicitCastExpr 0x7fa9ab827438 <line:20:6, col:10> 'int' <IntegralCast>
    |   `-ImplicitCastExpr 0x7fa9ab827420 <col:6, col:10> 'char' <LValueToRValue>
    |     `-ArraySubscriptExpr 0x7fa9ab8273b0 <col:6, col:10> 'char' lvalue
    |       |-ImplicitCastExpr 0x7fa9ab827398 <col:6> 'char *' <ArrayToPointerDecay>
    |       | `-StringLiteral 0x7fa9ab827350 <col:6> 'char [1]' lvalue ""
    |       `-IntegerLiteral 0x7fa9ab827378 <col:9> 'int' 1
    |-CallExpr 0x7fa9ab827560 <line:12:16, col:25> 'void'
    | |-ImplicitCastExpr 0x7fa9ab827548 <col:16> 'void (*)(int, ...)' <FunctionToPointerDecay>
    | | `-DeclRefExpr 0x7fa9ab827450 <col:16> 'void (int, ...)' Function 0x7fa9ab826a00 'sink' 'void (int, ...)'
    | `-ImplicitCastExpr 0x7fa9ab8275a8 <line:21:6, col:11> 'int' <IntegralCast>
    |   `-ImplicitCastExpr 0x7fa9ab827590 <col:6, col:11> 'char' <LValueToRValue>
    |     `-ArraySubscriptExpr 0x7fa9ab827520 <col:6, col:11> 'char' lvalue
    |       |-ImplicitCastExpr 0x7fa9ab827508 <col:6> 'char *' <ArrayToPointerDecay>
    |       | `-StringLiteral 0x7fa9ab8274b8 <col:6> 'char [2]' lvalue "0"
    |       `-IntegerLiteral 0x7fa9ab8274e8 <col:10> 'int' 2
    |-CallExpr 0x7fa9ab8276d0 <line:12:16, col:25> 'void'
    | |-ImplicitCastExpr 0x7fa9ab8276b8 <col:16> 'void (*)(int, ...)' <FunctionToPointerDecay>
    | | `-DeclRefExpr 0x7fa9ab8275c0 <col:16> 'void (int, ...)' Function 0x7fa9ab826a00 'sink' 'void (int, ...)'
    | `-ImplicitCastExpr 0x7fa9ab827718 <line:22:6, col:13> 'int' <IntegralCast>
    |   `-ImplicitCastExpr 0x7fa9ab827700 <col:6, col:13> 'char' <LValueToRValue>
    |     `-ArraySubscriptExpr 0x7fa9ab827690 <col:6, col:13> 'char' lvalue
    |       |-ImplicitCastExpr 0x7fa9ab827678 <col:6> 'char *' <ArrayToPointerDecay>
    |       | `-StringLiteral 0x7fa9ab827628 <col:6> 'char [4]' lvalue "012"
    |       `-IntegerLiteral 0x7fa9ab827658 <col:12> 'int' 2
    |-CallExpr 0x7fa9ab827800 <line:12:16, col:25> 'void'
    | |-ImplicitCastExpr 0x7fa9ab8277e8 <col:16> 'void (*)(int, ...)' <FunctionToPointerDecay>
    | | `-DeclRefExpr 0x7fa9ab827730 <col:16> 'void (int, ...)' Function 0x7fa9ab826a00 'sink' 'void (int, ...)'
    | `-ImplicitCastExpr 0x7fa9ab827848 <line:23:6, col:13> 'int' <IntegralCast>
    |   `-ImplicitCastExpr 0x7fa9ab827830 <col:6, col:13> 'char' <LValueToRValue>
    |     `-ArraySubscriptExpr 0x7fa9ab8277c0 <col:6, col:13> 'char' lvalue
    |       |-ImplicitCastExpr 0x7fa9ab8277a8 <col:6> 'char *' <ArrayToPointerDecay>
    |       | `-StringLiteral 0x7fa9ab827758 <col:6> 'char [4]' lvalue "012"
    |       `-IntegerLiteral 0x7fa9ab827788 <col:12> 'int' 3
    |-CallExpr 0x7fa9ab827930 <line:12:16, col:25> 'void'
    | |-ImplicitCastExpr 0x7fa9ab827918 <col:16> 'void (*)(int, ...)' <FunctionToPointerDecay>
    | | `-DeclRefExpr 0x7fa9ab827860 <col:16> 'void (int, ...)' Function 0x7fa9ab826a00 'sink' 'void (int, ...)'
    | `-ImplicitCastExpr 0x7fa9ab827978 <line:24:6, col:13> 'int' <IntegralCast>
    |   `-ImplicitCastExpr 0x7fa9ab827960 <col:6, col:13> 'char' <LValueToRValue>
    |     `-ArraySubscriptExpr 0x7fa9ab8278f0 <col:6, col:13> 'char' lvalue
    |       |-ImplicitCastExpr 0x7fa9ab8278d8 <col:6> 'char *' <ArrayToPointerDecay>
    |       | `-StringLiteral 0x7fa9ab827888 <col:6> 'char [4]' lvalue "012"
    |       `-IntegerLiteral 0x7fa9ab8278b8 <col:12> 'int' 4
    |-CallExpr 0x7fa9ab827aa8 <line:12:16, col:25> 'void'
    | |-ImplicitCastExpr 0x7fa9ab827a90 <col:16> 'void (*)(int, ...)' <FunctionToPointerDecay>
    | | `-DeclRefExpr 0x7fa9ab827990 <col:16> 'void (int, ...)' Function 0x7fa9ab826a00 'sink' 'void (int, ...)'
    | `-ImplicitCastExpr 0x7fa9ab827af0 <line:25:6, col:22> 'int' <IntegralCast>
    |   `-ImplicitCastExpr 0x7fa9ab827ad8 <col:6, col:22> 'char' <LValueToRValue>
    |     `-ArraySubscriptExpr 0x7fa9ab827a68 <col:6, col:22> 'char' lvalue
    |       |-ImplicitCastExpr 0x7fa9ab827a50 <col:6> 'char *' <ArrayToPointerDecay>
    |       | `-StringLiteral 0x7fa9ab827a00 <col:6> 'char [5]' lvalue "0123"
    |       `-IntegerLiteral 0x7fa9ab827a30 <<built-in>:40:25> 'long' 9223372036854775807
    `-CallExpr 0x7fa9ab827bd8 <c_files/pr82596.c:12:16, col:25> 'void'
      |-ImplicitCastExpr 0x7fa9ab827bc0 <col:16> 'void (*)(int, ...)' <FunctionToPointerDecay>
      | `-DeclRefExpr 0x7fa9ab827b08 <col:16> 'void (int, ...)' Function 0x7fa9ab826a00 'sink' 'void (int, ...)'
      `-ImplicitCastExpr 0x7fa9ab827c20 <line:26:6, col:21> 'int' <IntegralCast>
        `-ImplicitCastExpr 0x7fa9ab827c08 <col:6, col:21> 'char' <LValueToRValue>
          `-ArraySubscriptExpr 0x7fa9ab827b98 <col:6, col:21> 'char' lvalue
            |-ImplicitCastExpr 0x7fa9ab827b80 <col:6> 'char *' <ArrayToPointerDecay>
            | `-StringLiteral 0x7fa9ab827b30 <col:6> 'char [5]' lvalue "0123"
            `-IntegerLiteral 0x7fa9ab827b60 <<built-in>:38:22> 'unsigned long' 18446744073709551615
