c_files/pr82848.c:9:6: warning: unknown attribute 'altivec' ignored [-Wunknown-attributes]
foo (vector float a)
     ^
c_files/pr82848.c:6:32: note: expanded from macro 'vector'
#define vector __attribute__ ((altivec (vector__)))
                               ^
c_files/pr82848.c:8:1: warning: unknown attribute 'altivec' ignored [-Wunknown-attributes]
vector float
^
c_files/pr82848.c:6:32: note: expanded from macro 'vector'
#define vector __attribute__ ((altivec (vector__)))
                               ^
c_files/pr82848.c:11:3: warning: unknown attribute 'altivec' ignored [-Wunknown-attributes]
  vector float b = __builtin_vec_madd (b, a, (vector float) { C, C, C, C });    /* { dg-error "requires the '-maltivec' option" } */
  ^
c_files/pr82848.c:6:32: note: expanded from macro 'vector'
#define vector __attribute__ ((altivec (vector__)))
                               ^
c_files/pr82848.c:11:20: error: use of unknown builtin '__builtin_vec_madd' [-Wimplicit-function-declaration]
  vector float b = __builtin_vec_madd (b, a, (vector float) { C, C, C, C });    /* { dg-error "requires the '-maltivec' option" } */
                   ^
c_files/pr82848.c:11:47: warning: unknown attribute 'altivec' ignored [-Wunknown-attributes]
  vector float b = __builtin_vec_madd (b, a, (vector float) { C, C, C, C });    /* { dg-error "requires the '-maltivec' option" } */
                                              ^~~~~~
c_files/pr82848.c:6:32: note: expanded from macro 'vector'
#define vector __attribute__ ((altivec (vector__)))
                               ^~~~~~~~~~~~~~~~~~
c_files/pr82848.c:11:66: warning: excess elements in scalar initializer
  vector float b = __builtin_vec_madd (b, a, (vector float) { C, C, C, C });    /* { dg-error "requires the '-maltivec' option" } */
                                                                 ^
c_files/pr82848.c:5:11: note: expanded from macro 'C'
#define C 3.68249351546114573519399405666776E-44f
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
5 warnings and 1 error generated.
TranslationUnitDecl 0x7fd0618264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fd061826a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fd061826740 '__int128'
|-TypedefDecl 0x7fd061826a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fd061826760 'unsigned __int128'
|-TypedefDecl 0x7fd061826d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fd061826b70 'struct __NSConstantString_tag'
|   `-Record 0x7fd061826ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fd061826e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fd061826dc0 'char *'
|   `-BuiltinType 0x7fd061826560 'char'
|-TypedefDecl 0x7fd0618270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fd061827070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fd061826ee0 'struct __va_list_tag'
|     `-Record 0x7fd061826e58 '__va_list_tag'
|-FunctionDecl 0x7fd061879430 <c_files/pr82848.c:6:16, line:13:1> line:9:1 foo 'float (float)'
| |-ParmVarDecl 0x7fd061827158 <line:6:16, line:9:19> col:19 used a 'float'
| `-CompoundStmt 0x7fd061879a28 <line:10:1, line:13:1>
|   |-DeclStmt 0x7fd0618799b8 <line:6:16, line:11:76>
|   | `-VarDecl 0x7fd061879538 <line:6:16, line:11:75> col:16 used b 'float' cinit
|   |   `-ImplicitCastExpr 0x7fd0618799a0 <col:20, col:75> 'float' <IntegralToFloating>
|   |     `-CallExpr 0x7fd0618798d0 <col:20, col:75> 'int'
|   |       |-ImplicitCastExpr 0x7fd0618798b8 <col:20> 'int (*)()' <FunctionToPointerDecay>
|   |       | `-DeclRefExpr 0x7fd061879698 <col:20> 'int ()' Function 0x7fd061879600 '__builtin_vec_madd' 'int ()'
|   |       |-ImplicitCastExpr 0x7fd061879928 <col:40> 'double' <FloatingCast>
|   |       | `-ImplicitCastExpr 0x7fd061879910 <col:40> 'float' <LValueToRValue>
|   |       |   `-DeclRefExpr 0x7fd0618796c0 <col:40> 'float' lvalue Var 0x7fd061879538 'b' 'float'
|   |       |-ImplicitCastExpr 0x7fd061879958 <col:43> 'double' <FloatingCast>
|   |       | `-ImplicitCastExpr 0x7fd061879940 <col:43> 'float' <LValueToRValue>
|   |       |   `-DeclRefExpr 0x7fd0618796e8 <col:43> 'float' lvalue ParmVar 0x7fd061827158 'a' 'float'
|   |       `-ImplicitCastExpr 0x7fd061879988 <col:46, col:74> 'double' <FloatingCast>
|   |         `-ImplicitCastExpr 0x7fd061879970 <col:46, col:74> 'float' <LValueToRValue>
|   |           `-CompoundLiteralExpr 0x7fd061879860 <col:46, col:74> 'float' lvalue
|   |             `-InitListExpr 0x7fd061879818 <col:61, col:74> 'float'
|   |               `-FloatingLiteral 0x7fd061879738 <line:5:11> 'float' 3.643376e-44
|   `-ReturnStmt 0x7fd061879a10 <line:12:3, col:10>
|     `-ImplicitCastExpr 0x7fd0618799f8 <col:10> 'float' <LValueToRValue>
|       `-DeclRefExpr 0x7fd0618799d0 <col:10> 'float' lvalue Var 0x7fd061879538 'b' 'float'
`-FunctionDecl 0x7fd061879600 <<invalid sloc>> line:11:20 implicit used __builtin_vec_madd 'int ()'
