c_files/pragma-float-const-decimal64-5.c:14:14: warning: unknown pragma in STDC namespace [-Wunknown-pragmas]
#pragma STDC FLOAT_CONST_DECIMAL64 ON           /* { dg-warning "ISO C" } */
             ^
c_files/pragma-float-const-decimal64-5.c:20:14: warning: unknown pragma in STDC namespace [-Wunknown-pragmas]
#pragma STDC FLOAT_CONST_DECIMAL64 OFF          /* { dg-warning "ISO C" } */
             ^
c_files/pragma-float-const-decimal64-5.c:26:14: warning: unknown pragma in STDC namespace [-Wunknown-pragmas]
#pragma STDC FLOAT_CONST_DECIMAL64 DEFAULT      /* { dg-warning "ISO C" } */
             ^
c_files/pragma-float-const-decimal64-5.c:32:3: warning: unknown pragma in STDC namespace [-Wunknown-pragmas]
  _Pragma ("STDC FLOAT_CONST_DECIMAL64 ON")     /* { dg-warning "ISO C" } */
  ^
<scratch space>:2:7: note: expanded from here
 STDC FLOAT_CONST_DECIMAL64 ON
      ^
c_files/pragma-float-const-decimal64-5.c:38:3: warning: unknown pragma in STDC namespace [-Wunknown-pragmas]
  _Pragma ("STDC FLOAT_CONST_DECIMAL64 OFF")    /* { dg-warning "ISO C" } */
  ^
<scratch space>:4:7: note: expanded from here
 STDC FLOAT_CONST_DECIMAL64 OFF
      ^
c_files/pragma-float-const-decimal64-5.c:44:3: warning: unknown pragma in STDC namespace [-Wunknown-pragmas]
  _Pragma ("STDC FLOAT_CONST_DECIMAL64 DEFAULT")        /* { dg-warning "ISO C" } */
  ^
<scratch space>:6:7: note: expanded from here
 STDC FLOAT_CONST_DECIMAL64 DEFAULT
      ^
6 warnings generated.
TranslationUnitDecl 0x7ff15d8130d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7ff15d813620 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7ff15d813340 '__int128'
|-TypedefDecl 0x7ff15d813690 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7ff15d813360 'unsigned __int128'
|-TypedefDecl 0x7ff15d813968 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7ff15d813770 'struct __NSConstantString_tag'
|   `-Record 0x7ff15d8136e8 '__NSConstantString_tag'
|-TypedefDecl 0x7ff15d813a00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7ff15d8139c0 'char *'
|   `-BuiltinType 0x7ff15d813160 'char'
|-TypedefDecl 0x7ff15d813cc8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7ff15d813c70 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7ff15d813ae0 'struct __va_list_tag'
|     `-Record 0x7ff15d813a58 '__va_list_tag'
|-VarDecl 0x7ff15d813d38 <c_files/pragma-float-const-decimal64-5.c:10:1, col:8> col:8 used a 'double'
|-FunctionDecl 0x7ff15e842eb0 <line:12:1, line:16:1> line:12:6 f1 'void (void)'
| `-CompoundStmt 0x7ff15e842fb8 <line:13:1, line:16:1>
|   `-BinaryOperator 0x7ff15e842f90 <line:15:3, col:7> 'double' '='
|     |-DeclRefExpr 0x7ff15e842f48 <col:3> 'double' lvalue Var 0x7ff15d813d38 'a' 'double'
|     `-FloatingLiteral 0x7ff15e842f70 <col:7> 'double' 1.000000e+00
|-FunctionDecl 0x7ff15e843068 <line:18:1, line:22:1> line:18:6 f2 'void (void)'
| `-CompoundStmt 0x7ff15e843170 <line:19:1, line:22:1>
|   `-BinaryOperator 0x7ff15e843148 <line:21:3, col:7> 'double' '='
|     |-DeclRefExpr 0x7ff15e843100 <col:3> 'double' lvalue Var 0x7ff15d813d38 'a' 'double'
|     `-FloatingLiteral 0x7ff15e843128 <col:7> 'double' 2.000000e+00
|-FunctionDecl 0x7ff15e843220 <line:24:1, line:28:1> line:24:6 f3 'void (void)'
| `-CompoundStmt 0x7ff15e843328 <line:25:1, line:28:1>
|   `-BinaryOperator 0x7ff15e843300 <line:27:3, col:7> 'double' '='
|     |-DeclRefExpr 0x7ff15e8432b8 <col:3> 'double' lvalue Var 0x7ff15d813d38 'a' 'double'
|     `-FloatingLiteral 0x7ff15e8432e0 <col:7> 'double' 3.000000e+00
|-FunctionDecl 0x7ff15e8433d8 <line:30:1, line:34:1> line:30:6 f4 'void (void)'
| `-CompoundStmt 0x7ff15e8434e0 <line:31:1, line:34:1>
|   `-BinaryOperator 0x7ff15e8434b8 <line:33:3, col:7> 'double' '='
|     |-DeclRefExpr 0x7ff15e843470 <col:3> 'double' lvalue Var 0x7ff15d813d38 'a' 'double'
|     `-FloatingLiteral 0x7ff15e843498 <col:7> 'double' 1.000000e+00
|-FunctionDecl 0x7ff15e843590 <line:36:1, line:40:1> line:36:6 f5 'void (void)'
| `-CompoundStmt 0x7ff15e843698 <line:37:1, line:40:1>
|   `-BinaryOperator 0x7ff15e843670 <line:39:3, col:7> 'double' '='
|     |-DeclRefExpr 0x7ff15e843628 <col:3> 'double' lvalue Var 0x7ff15d813d38 'a' 'double'
|     `-FloatingLiteral 0x7ff15e843650 <col:7> 'double' 2.000000e+00
`-FunctionDecl 0x7ff15e843748 <line:42:1, line:46:1> line:42:6 f6 'void (void)'
  `-CompoundStmt 0x7ff15e843850 <line:43:1, line:46:1>
    `-BinaryOperator 0x7ff15e843828 <line:45:3, col:7> 'double' '='
      |-DeclRefExpr 0x7ff15e8437e0 <col:3> 'double' lvalue Var 0x7ff15d813d38 'a' 'double'
      `-FloatingLiteral 0x7ff15e843808 <col:7> 'double' 3.000000e+00
