c_files/pragma-ms_struct.c:12:51: warning: incorrect use of '#pragma ms_struct on|off' - ignored [-Wignored-pragmas]
#pragma ms_struct /* { dg-warning "malformed" } */
                                                  ^
c_files/pragma-ms_struct.c:14:22: warning: extra tokens at end of '#pragma ms_struct' - ignored [-Wignored-pragmas]
#pragma ms_struct on top of spaghetti /* { dg-warning "junk" } */
                     ^
2 warnings generated.
TranslationUnitDecl 0x7fa0620264d0 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x7fa062026a20 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x7fa062026740 '__int128'
|-TypedefDecl 0x7fa062026a90 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x7fa062026760 'unsigned __int128'
|-TypedefDecl 0x7fa062026d68 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x7fa062026b70 'struct __NSConstantString_tag'
|   `-Record 0x7fa062026ae8 '__NSConstantString_tag'
|-TypedefDecl 0x7fa062026e00 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x7fa062026dc0 'char *'
|   `-BuiltinType 0x7fa062026560 'char'
|-TypedefDecl 0x7fa0620270c8 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x7fa062027070 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x7fa062026ee0 'struct __va_list_tag'
|     `-Record 0x7fa062026e58 '__va_list_tag'
`-RecordDecl 0x7fa062027120 <c_files/pragma-ms_struct.c:16:1, line:21:1> line:16:8 struct foo definition
  |-FieldDecl 0x7fa062077400 <line:18:3, col:7> col:7 a 'int'
  |-FieldDecl 0x7fa062077460 <line:19:3, col:7> col:7 b 'int'
  `-FieldDecl 0x7fa0620774b8 <line:20:3, col:8> col:8 c 'char'
